# *TIR1/AFB* genes {#AFBs} 

```{r TIR1AFBdivstats, fig.cap="(ref:TIR1AFBdivstats)"}
gene_info <- geneInfoFromFile(
  fname = system.file("shiny-app", "gene_family_data",
                      "AFB_gene_ids.csv", package = "r1001genomes"))

VCF <- readRDS(system.file("shiny-app", "gene_family_data",
                      "AFBs_VCF.rds", package = "r1001genomes"))
div_stats <- ldply(VCF, diversityStats, geneInfo=gene_info, .id="transcript_ID")
div_stats$tair_symbol %<>% fct_reorder(.x =  div_stats$tair_symbol, 
                                       .fun = function(x) 
                                         {as.integer(str_remove(x, 
                                              pattern = "AFB|TIR"))})
div_stats$tair_symbol %<>% fct_relevel("TIR1")
div_stats %<>% arrange(tair_symbol)
div_stats %<>% dplyr::select(tair_symbol, everything())

if (knitr::is_html_output()) {
  formatRound(datatable(div_stats,
                  colnames = c("transcript",
                               "symbol",
                               "&pi;<sub>N</sub>",
                               "&pi;<sub>S</sub>",
                               "&pi;<sub>N</sub>/&pi;<sub>S</sub>",
                               "&pi; coding",
                               "&pi; transcript"),
                  rownames = FALSE, escape = FALSE,
                  options = list(paging=FALSE, searching=FALSE)),
                columns = 2:7, digits = 6)
  } else 
{
  dt <- div_stats
  names(dt) <- c("transcript",
                                 "symbol",
                                 "&pi;~N~",
                                 "&pi;~S~",
                                 "&pi;~N~/&pi;~S~",
                                 "&pi; coding",
                                 "&pi; transcript")
  dt %<>% mutate_if(is.numeric, round, 6)
  names(dt) <- paste0("**", names(dt), "**")
  kable(dt) 
}
```

```{r AFB_ alignment}
alignment <- alignCDS(gene_info$transcript_ID)
# make an alignment data frame
aln_df <- makeAlnDF(alignment[[2]])

# make a collapsed alignment for translating positions
collapsed_aln_df <- 
  aln_df %>% dplyr::select(aln_pos, seq_pos, seq_name) %>%
  tidyr::spread(key = seq_name, value = seq_pos)
```

```{r Mask_AFB_alignment, eval=FALSE}
aliscore(as.AAbin(alignment[[2]]), path = "/Volumes/SSD/Applications/ALISCORE_v2.0/Aliscore_v.2.0/",  o = "AT2G39940")
id <- scan(file = "/Volumes/SSD/Applications/ALISCORE_v2.0/Aliscore_v.2.0/input.fas_List_random.txt", sep = " ", quiet = TRUE)
AAAlign_no_gaps <- as.matrix(alignment[[2]])[, id]
write.nexus.data(AAAlign_no_gaps, file = "trees/AFBs.nex", format = "protein")
```
```{bash AFB_phylogeny_construction, eval=FALSE}
cd ~/Desktop/natural-variation-webtool/manuscript/trees

/Applications/MrBayes/mb

execute AFBs.nex
# set IAA33 as the outgroup
outgroup AT2G39940.1

lset nucmodel=protein Rates=invgamma
# or include Nst=6  ngammacat=4 omegavar=M3 
prset aamodelpr=fixed(jones) statefreqpr=fixed(empirical) 
# initial run with aamodelpr=mixed yielded posterior prob of 1 for jones model
propset ExtTBR$prob=0

mcmcp ngen=250000 nchains=4 printfreq=1000 samplefreq=100 burnin=200

mcmc
sump
sumt

```

```{r Load AFB phylogeny and data}
tree <- treeio::read.beast(file = "trees/AFBs.nex.con.tre")

AFB_data <- read.csv(system.file("shiny-app", "gene_family_data",
                      "AFB_gene_ids.csv", package = "r1001genomes"))
AFB_data %<>% left_join(y = div_stats, by = c("name" = "tair_symbol"))
#tip_labels need to be in the first column
AFB_data %<>% dplyr::select(name, everything())
AFB_data$tair_locus %<>% as.character()

AFB_data <- AFB_data[match(gsub(x = tree@phylo$tip.label, pattern = "\\.1",replacement = ""), AFB_data$tair_locus), ] 

tree@phylo$tip.label <- AFB_data$name
```

(ref:AFBtree) **AFB protein sequence tree mapped with ${\pi_N/\pi_S}$.** Protein sequences were aligned [@wright_decipher_2015] and low information content regions were masked [@kuck_parametric_2010] prior to infering a phylogeny [@ronquist_mrbayes_2003]. Tips of the tree are mapped with circles of diameter proportional to ${\pi_N/\pi_S}$ and also are colored according to ${\pi_N/\pi_S}$. Nodes are labeled with the poster probability of monophyly.

```{r AFBtree, fig.height=2, fig.width=3, fig.cap="(ref:AFBtree)"}
p <- ggtree(tree, layout = "rectangular", color = "grey") %<+% AFB_data + theme_tree() + scale_x_continuous(expand = c(0, 1.2)) + 
  geom_tippoint(mapping = aes(size = 4*Pi_NS_Ratio, fill = Pi_NS_Ratio), shape = 21, color = "white", show.legend = TRUE) + 
  scale_fill_viridis(values = rescale(c(min(AFB_data$Pi_NS_Ratio),
                                        max(AFB_data$Pi_NS_Ratio)), 
                                      to = c(0,1)), guide = "colourbar") + scale_size(guide = "none") +
  labs(fill = expression(pi [N] / pi [S])) + 
  theme(legend.position = c(.1,.7)) + 
  #geom_nodelab(mapping = aes(label = node)) +
  geom_nodelab(mapping = aes(label = round(as.numeric(prob),digits = 2)), nudge_x = -.2, nudge_y = .2, size = 3) + 
  geom_tiplab(hjust = -0.25) + 
  scale_color_viridis(discrete = TRUE)  
p <- p %>% ggtree::rotate(12) %>% ggtree::rotate(9)
p
#ggsave("AFBs_tree.pdf", height = 3, width = 4)
```

```{r full-AFBs-alignment}
AFB_order <- fct_inorder(na.exclude(p$data$label[base::order(p$data$y, decreasing = TRUE)]), ordered = TRUE)

aln_df <- left_join(aln_df, dplyr::select(gene_info, "tair_locus",
                                       "tair_symbol", "transcript_ID"),
                        by = c("transcript_ID" = "transcript_ID"))
aln_df$tair_symbol %<>% factor(levels = AFB_order, 
                                       ordered = TRUE)
aln_df$tair_symbol %<>% fct_rev()
VCF <- ldply(.data = VCF, .fun = subset, gt_GT != "0|0")
VCF <- getCodingDiv(VCF)

aln_df <- addSNPsToAlnDF(aln_df, VCF, effect_order = SNPeff_order)
aln_df$gap <- aln_df$seq_pos == "-"

p <-ggplot(subset(aln_df, tair_symbol != "COI1"), aes(x = aln_pos, y = tair_symbol)) +
  geom_raster(mapping = aes(fill = strength,
                            alpha = gap)) +
  scale_alpha_manual(values = c("TRUE" = 0, "FALSE" = 1), 
                     labels = c("aligned", "gap"), 
                     guide = guide_legend(override.aes = 
                                            list(fill = c("grey85", "white")),
                       title = "alignment",
                       direction = "vertical")) + 
   scale_fill_viridis(breaks = range(SNPeff_order$strength),
                      limits = range(SNPeff_order$strength),
                      na.value = "grey85",
                      direction = -1,
                      labels = c("neutral", "deleterious"),
                      option = "A", 
                      guide = guide_colorbar(ticks = FALSE, 
                                             title.vjust = .8, 
                                             title.position = "top", 
                                             title.hjust = 0.5)) +
  scale_y_discrete(expand = c(0,0)) +
  scale_x_continuous(expand = c(0,0)) +  # expand increases distance from axis
  labs(x = "codon position in alignment", y = "", fill = "variant effect") +
  theme_logo(base_family = "Helvetica") +
  theme(panel.grid = element_blank(), 
        panel.grid.minor = element_blank(),
        strip.background = element_blank(),
        strip.text.x = element_blank(),
        #panel.background = element_rect(fill = "grey85", color = "white"),
        legend.position = c("bottom"),
        legend.key = element_rect(linetype = "solid"),
        legend.spacing = unit(4, "char"), 
        legend.box.spacing = unit(0.1,"line")) 
p

```
