# Introduction {#intro}

The sequencing of the first *Arabidopsis thaliana* genome ushered in a new era of tool development and systematic functional annotation of plant genes [@thearabidopsisgenomeinitiative_analysis_2000]. Since that landmark effort, massive scaling of sequencing technology has allowed for the survey of genomic variation in natural *Arabidopsis thaliana* populations [@nordborg_pattern_2005; @borevitz_genomewide_2007; @weigel_1001_2009]. This valuable population genetics resource has led to several associations of genetic loci with phenotypic traits and provided insights into how selective pressure has influenced the evolution of plant genomes [@long_massive_2013; @atwell_genomewide_2010; @clark_common_2007]. 

Beyond its utility in gene discovery and understanding genome evolution, natural genetic variation provides a catalog of permissible polymorphisms that can facilitate the connection of genotype to phenotype at the gene, gene family and network scales [@nieduszynski_sequence_2011; @joly-lopez_developing_2016]. This is an especially critical resource in large gene families where loss of function in individual genes may have little or no phenotypic effect [@dharmasiri_plant_2005; @moore_evolutionary_2005; @guo_gene_2013] and directed allele replacement remains time and resource-intensive [@chen_crispr_2019]. Natural variation datasets provide novel alleles and germplasm which can be examined with biochemical and genetic approaches to map sequence to function and genotype to phenotype. In human clinical medicine, massively parallel assays of variant effects  stand to revolutionize genetic diagnostics and personalized medicine [@starita_variant_2017; @gasperini_power_2016; @matreyek_platform_2017]. Similarly, we envision the use of plant natural variation datasets as a tool to revolutionize breeding and genetic engineering of crop plants by rapidly advancing our understanding of genotype/function/phenotype relationships. A proof-of-principle survey of a relatively small subset of natural variants paired with a synthetic assay of gene function successfully mapped critical functional domains of auxin receptors and identified new alleles which affect plant phenotype [@wright_insights_2017]. 

Why is the survey of natural variants not as routine as a BLAST search or ordering T-DNA insertion mutants? One reason may be the current requirement for a fairly high level of bioinformatics expertise to extract the desired information from whole genome resequencing datasets. While existing resources such as the 1001 Proteomes website [@joshi_1001_2012] and ePlant [@waese_eplant_2017] facilitate access to this data at the gene scale, they do not provide comparative summaries or visualizations of variation at the gene family or network scales. To address this concern, we created ViVa: a webtool and R-package for **Vi**sualizing **Va**riation, which allows plant molecular biologists of any level access to gene-level data from the 1001 Genomes database. Using ViVa researchers may: 1) Identify polymorphisms to facilitate biochemical assays of variant effects [@starita_variant_2017; @wright_insights_2017]; 2) Produce family-wise alignments of variants to facilitate *de novo* functional domain identification [@melamed_combining_2015]; 3) Generate lists of accessions containing polymorphisms to facilitate phenotypic analysis of gene variant effects [@park_distributions_2017]; and 4) Quantify metrics of genetic diversity to facilitate the study of gene, gene family, and network evolution [@delker_natural_2010; @kliebenstein_role_2008]. Here we present a summary of the functionality of ViVa along with an analysis of the natural variation in the nuclear auxin signaling pathway using ViVa. To demonstrate ViVa usage we focus on the analysis of the *Aux/IAA* family and reserve the remainder of our analysis for the interested reader as supplemental material. 

