# Introduction {#intro}

The first genome sequence of *Arabidopsis thaliana* facilitated rapid advancement of plant biology [@thearabidopsisgenomeinitiative_analysis_2000]. Since this initial genome, massive scaling of sequencing technology has allowed the survey the genomic variation in natural *Arabidopsis thaliana* populations [@nordborg_pattern_2005; @borevitz_genomewide_2007; @weigel_1001_2009]. This valuable population genetics resource has led to several associations of genetic loci with phenotypic traits and provided insights into how selective pressure has influenced the evolution of plant genomes [@long_massive_2013; @atwell_genomewide_2010; @clark_common_2007]. Despite the high value of this compendium of genetic variation, its use has been largely limited to investigations of population genetics and for genome-wide association studies. 

Natural genetic variation provides a catalog of permissible polymorphisms which may facilitate exploration and prediction of genotype/function/phenotype relationships at the gene, gene family and network scales [@joly-lopez_developing_2016]. Outside of large effect variants, these relationships are difficult to map. Such studies in plants have the added obstacles of large gene families and the still quite time and resource-intensive process of allele replacement. Massively parallel assays of variant effects have revolutionized genetic diagnostics and personalized medicine [@starita_variant_2017; @gasperini_power_2016; @matreyek_platform_2017]. We envision the use of plant natural variation datasets as a tool to similarly revolutionize breeding and genetic engineering of crop plants by rapidly advancing our understanding of genotype/function/phenotype relationships. Biochemical or synthetic assays of gene function and quantitative phenotyping of the germplasm of collections of natural variation can facilitate functional and phenotypic quantification [@wright_insights_2017]. However, the high dimensionality of natural variation datasets often limits access to those with bioinformatics expertise. 

Here, we present ViVa: a web application and R-package for Visualizing Variation, which allows plant molecular biologists of any level access to the 1001 genomes database. ViVa facilitates formulation of hypotheses as to the genotype/function/phenotype relationships within any gene, family, or network. Using ViVa researchers may: 1. Identify polymorphisms to facilitate biochemical assays of variant effects [@starita_variant_2017; @wright_insights_2017]; 2. Produce family-wise alignments of variants to facilitate de novo functional domain identification [@melamed_combining_2015]; 3. Generate lists of accessions containing polymorphisms to facilitate phenotypic analysis of gene variant effects [@park_distributions_2017]; and 4. Quantify metrics of genetic diversity to facilitate the study of gene, gene family and network evolution [@delker_natural_2010; @kliebenstein_role_2008].

