# Results and Discussion {#rd}

## An overview of ViVa

ViVa, in this first iteration, is meant to visualize natural variation in the coding sequences of genes or gene families. Non-coding sequence variation is intentionally excluded from the analysis tools. This reflects challenges both in alignment of non-coding sequences and the increased difficulty in assessing the effects of variation in these regions [@alexandre_complex_2018]. 

The stable version of ViVa is hosted at https://www.plantsynbiolab.bse.vt.edu/ViVa/. The development version of ViVa can be accessed as a Docker container https://hub.docker.com/r/wrightrc/r1001genomes/ or as an R-package at https://github.com/wrightrc/r1001genomes. 

(ref:app-figure-elements) **Key elements of the webtool** (a) The first section contains two collapsible panels, Gene Select and Annotation Files, which are used to input information about the genes to be investigated. (b) The SNP Stats tab provides gene-structure level counts and statistics on SNPs. (c) The Diversity Plot tab plots the nucleotide diversity of SNP sites along the length of the coding region of a selected gene. (d) The SNP Mapping tab plots accessions on a world map colored according to the selected set of SNPs. (e) The SNP Browser tab allows variants and accessions to be filtered by any combination of text and numeric fields. (f) The Alignments tab aligns DNA and amino acid sequences of homologous genes and colors sequence elements based on SNPs and annotations.
```{r app-figure-elements, fig.cap = "(ref:app-figure-elements)"}
knitr::include_graphics("app_figure/App_figure.png")
```

(ref:app-workflow) **ViVa workflow** (a) Workflow diagram of ViVa. Blue indicates user actions, yellow indicates processing steps performed by the application. (b) Detailed look at mapping tab parameters: User selects which genes to look at (1) then clicks the Submit button (2). The "Allele selection" pane is then filled in with all non-reference variants meeting the criteria. The user can adjust the range of nucleotide diversity and the type of SNP with a slider and radio buttons (3) to make the list of variants a manageable size. The list of variants is updated as changes are made to these controls. The user then selects variants to display on the map (4). Clicking the Update Map button (5) populates the map below with points located at the collection coordinates of each accession and colored by the selected alleles. (c) Detailed look at the browser tab options: The user first selects which genes to analyze and clicks submit (1). The “hide 0|0 genotype?” checkbox (2) removes rows from the table containing the reference allele. Four configurable filters of two types are provided. The first filter type is text matching (3); the user selects a column to filter from a drop-down menu then enters one or more text strings to match in that column. The second type of filter is numeric range matching (4); the user again selects a column, then specifies a range to match by typing or selecting minimum and maximum values. After configuring the filters, the user clicks the Apply Filters button (5) to update the table. The table is updated such that only rows that meet all filter conditions are displayed.
```{r app-workflow, fig.cap = "(ref:app-workflow)"}
knitr::include_graphics("app_figure/workflow_fig.png")
```

### Gene Select and Annotation Files  
At the top of the ViVa webtool are two collapsible panels used for entering the genes to query and custom annotations for those genes (Figure \@ref(fig:app-figure-elements)a and \@ref(fig:app-workflow)a). The Gene Select panel permits gene input by either typing in or uploading a .csv file of AGI/TAIR locus identifiers. The Annotation Files panel is optionally used to upload an annotation file containing coordinates of domains, mutations, or any other sequence knowledge that can be plotted on some of the tabs of the ViVa analysis tabs.

Below the data input section, the rest of the webtool is divided into several tabs containing interactive output. 

### SNP Stats: Summary of gene information, structure, and diversity 

The SNP Stats tab provides general information on the gene transcripts being queried, as well as calculated counts/statistics on the content of variants found in the sample population (Figure \@ref(fig:app-figure-elements)b). The first table of this section is basic information about the transcripts, including TAIR locus and symbol, the chromosomal start and end position, and the transcript length. This information was collected from the Araport11 Official Release (06/2016) annotation dataset [@cheng_araport11_2017].  

The next two tables provide counts of SNPs across the gene body for each transcript. The Total Polymorphism Counts table provides the total number of variant observations of non-reference alleles categorized by type and location (the Col-0 accession TAIR9 genome is the reference genome for this dataset). The Unique Allele Count table only counts the number of unique variant alleles within the population of accessions (*e.g.* if multiple accessions have the same variant, these are counted as a single allele at that position).

The Nucleotide Diversity Statistics table provides nucleotide diversity statistic (${\pi}$) values for the transcript and the coding sequence of each gene [@nei_mathematical_1979]. Given a set of nucleotide sequences from a population, ${\pi}$ is the average number of nucleotide differences per site. Nucleotide diversity is also calculated for the set of only synonymous (${\pi_S}$) and only nonsynonymous polymorphisms (${\pi_N}$). As nonsynonymous polymorphisms are more likely to give rise to functional change than synonymous polymorphisms, the ratio of the presence of nonsynonymous to synonymous polymorphisms provides a measure of the potential for functional diversity [@firnberg_genetic_2013; @whitehead_optimization_2012]. We present ${\pi_N/\pi_S}$, the ratio of nonsynonymous to synonymous diversity here as a correlate for functional diversity throughout ViVa [@nelson_snpgenie_2015; @hughes_adaptive_2000]. While imperfect, this metric may be suggestive of functional constraint when ${\pi_N/\pi_S << 1}$ and functional diversity when ${\pi_N/\pi_S >> 1}$ [@hughes_adaptive_1999].    

### Diversity plot: Visualize allelic diversity across the coding sequence
The Diversity Plot tab shows the nucleotide diversity of each variant in the coding region of a selected gene (Figure \@ref(fig:app-figure-elements)c and \@ref(fig:IAA6-div-plot)). Although the X-axis is marked by codon number from the N-terminus for interpretability, the diversity values are based on single nucleotide sites. The colors of markers on the plot identify the effect of the polymorphism. If annotation files are provided, the background of the plot is color-coded by the annotated regions. If points on the plot are selected by clicking and dragging a box over them, the data for the selected points appears in the grey box below the plot. Below this is a complete data table containing all points on the plot which can be downloaded as a .csv file. This tab allows users to identify regions of high diversity as well as isolate polymorphisms that may affect gene function and exist in multiple accessions, facilitating phenotypic analysis. 

### SNP Mapping: View distributions of SNPs across the globe
The SNP Mapping tab plots the accessions' collection locations on a world map and colors the points based on selected variant alleles (Figure \@ref(fig:app-figure-elements)d). After the user selects genes and filters based on the SNP type and level of nucleotide diversity, a group of checkboxes becomes available to select variant alleles to display on the map (Figure \@ref(fig:app-workflow)b and \@ref(fig:AFB1-map)). The variant alleles are labeled with the Transcript_ID and Amino_Acid_Change fields, in the form [Transcript_ID|Amino_Acid_Change]. After selecting the variant alleles and updating the map, the accessions are plotted on the map colored by each unique combination of the selected alleles. Below the map is a table containing the accession details for all mapped accessions. This tab may help users formulate hypotheses about the relatedness of accessions sharing a common allele and environments in which that allele may be favorable.

### SNP Browser: Filter and search for variants
The SNP Browser tab provides a way to search and filter the variant data by different fields (Figure \@ref(fig:app-figure-elements)e). After selecting the transcripts to include, a number of filters can be applied to the dataset to match text values (e.g. gene name or variant effect) or set minimum and maximum limits on the values of numeric fields (e.g. nucleotide diversity) (\@ref(fig:app-workflow)c). When these filters are applied, the table below is updated to only contain rows meeting the criteria for all filters. This tab can be useful for identifying all accessions with a particular allele, or any non-reference alleles in a particular region of a gene that may not have been easily accessible in another tab.

### Alignments: Visualize SNPs on alignments of homologous genes
The Alignments tab provides DNA and amino acid sequence alignments of selected genes, colored according to the variant allele with the strongest functional effect at each position (Figure \@ref(fig:app-figure-elements)f and \@ref(fig:A-IAA-alignment), for a full description of the color scale see Figure \@ref(fig:SNPeff-order)). The content of this tab is most useful if the selected genes are all family members or have significant sequence homology. If annotation files are uploaded, open boxes will be overlaid on the alignment, colored by annotation. Hovering the cursor over variants will provide additional details about the alleles present at that locus. This tab facilitates family-wise analysis of functional conservation, allowing users to identify potential functional regions and alleles which may be useful in deciphering this function.

### Gene Tree: Visualize functional diversity and sequence divergence of a gene family
The Gene Tree tab provides a neighbor joining tree (or uploaded tree created by the user) for the selected genes with the tips of the tree mapped with predicted functional diversity as represented by ${\pi_N/\pi_S}$ in the 1001 Genomes dataset (Figure \@ref(fig:IAAtree)). This tab allows users to generate hypotheses regarding functional diversity and redundancy within the context of the predicted evolution of the gene family.

### ViVa R package: Programmatic access to ViVa's functionality
All of the functionalities of the ViVa webtool are implemented through functions within the ViVa R package. In addition to being able to generate the same sets of figures and tables as in the webtool, users of the R package also gain direct access to the underlying data structures, providing greater control over parameters when processing and visualizing the data. The ViVa R package is intended for users familiar with R programming who want to extend the capabilities of the webtool. The ViVa R package can be found at https://github.com/wrightrc/r1001genomes and can be installed in your R environment via the `devtools` package: 
`remotes::install_github("wrightrc/r1001genomes")`. 

## Visualizing Variation within the auxin signaling pathway

To test the usability and accessibility of ViVa, we assembled a group of alpha testers comprising postdoctoral, graduate, and undergraduate researchers at a research university (University of Washington) and at a primarily-undergraduate institution (Whitman College). Our testers focused their investigation of natural variation on the nuclear auxin signaling pathway. We selected this signaling pathway for multiple reasons including a wealth of functional data and solved structures of several domains or entire proteins. Using this existing knowledge, we were able to qualitatively assess the predictive ability of the ViVa modules. Below, we describe the results for detailed analysis of the *Aux/IAA* family in more detail as a brief vignette of ViVa use. Analysis of each gene family in the nuclear auxin signaling network can be found in the [Supplemental Analysis](#si). 

The Aux/IAA family plays a critical role in transmitting auxin signals. Aux/IAA degradation is triggered by auxin accumulation [@zenser_auxin_2001] and is mediated by ubiquitination via a SKP-Cullin-F-box ubiquitin ligase complex containing an Auxin-signaling F-box (AFB) auxin receptor protein [@gray_auxin_2001]. Variation in the *AFB* family is presented in [Supplemental Analysis](#si) Figures \@ref(fig:AFBtree), \@ref(fig:full-AFBs-alignment), and \@ref(fig:AFB1-map)). The Aux/IAAs repress transcription of Auxin Response Factor (ARF) bound genes [@tiwari_aux_2004], via recruitment of TOPLESS (TPL) and TOPLESS-related (TPR) co-repressors [@szemenyei_topless_2008]. Variation in the *TPL/TPR* family is presented in [Supplemental Analysis](#si) Figures \@ref(fig:TPLtree) and \@ref(fig:full-TPLs-alignment)). Thus Aux/IAA degradation, relieves this repression allowing ARFs to activate transcription of auxin response genes [@tiwari_roles_2003]. Variation in the *ARF* family is presented in [Supplemental Analysis](#si) Figures \@ref(fig:ARFtree) and \@ref(fig:full-ARFs-alignment)).

To predict the functional impact of variation in gene coding sequences, ViVa uses the frequency of nonsynonymous and synonymous polymorphisms. In most cases, nonsynonymous polymorphisms in critical functional domains are likely to have deleterious effects on gene function [@hughes_adaptive_2000]. Therefore, domains which are critical to plant fitness will accumulate fewer nonsynonymous polymorphism than regions which are non-critical domains. Thus, we reasoned that scanning gene coding sequences for regions of relatively low nonsynonymous diversity should highlight functional domains. This general principle can be seen clearly in the analysis of the *Aux/IAA* family of transcriptional co-repressors/co-receptors. Most Aux/IAAs have three major domains. Domain I contains an EAR motif that facilitates interaction with TPL/TPR transcriptional co-repressors [@tiwari_aux_2004; @szemenyei_topless_2008]. Domain II, the degron, facilitates interactions with the TIR1/AFB receptors in the presence of auxin [@tan_mechanism_2007]. Domain III (which was originally considered domains III and IV) is a PB1 domain and facilitates interactions with the ARF transcription factors [@ulmasov_aux_1997; @guilfoyle_getting_2012; @nanao_structural_2014; @korasick_molecular_2014]. 

We focused our analysis on the most conserved A class Aux/IAAs, which possess all three of these domains. We began by mapping the natural genetic variation onto an alignment of the coding sequences via the ViVa Alignment tab. A similar visualization of the full gene family is presented in the [Supplemental Analysis](#si) (Figure \@ref(fig:full-IAA-alignment)). In the alignment of A class Aux/IAAs, the EAR motif and degron can be readily identified by the drop in nonsynonymous variation, as visualized by the lack of strong variant functional effects (Figure \@ref(fig:A-IAA-alignment)). The PB1 domain is not as readily identified, perhaps because the multiple key residues are spread out in linear sequence space. It is worth noting that these key residues that facilitate electrostatic PB1-PB1 interactions show little variation.

```{r Load IAA gene info and VCF}
gene_info <- geneInfoFromFile(
  fname = system.file("shiny-app", "gene_family_data",
                      "IAA_gene_ids.csv", package = "r1001genomes"))
#annotation <- readAnnotationFile(filename = system.file("shiny-app", "gene_family_data",
#                      "IAA_annotations.csv", package = "r1001genomes"), gene_info = gene_info)

VCF <- readRDS(system.file("shiny-app", "gene_family_data",
                      "IAAs_VCF.rds", package = "r1001genomes"))
```
```{r IAA alignment}
alignment <- alignCDS(gene_info$transcript_ID)
# make an alignment data frame
aln_df <- makeAlnDF(alignment[[2]])

# make a collapsed alignment for translating positions
collapsed_aln_df <- 
  aln_df %>% dplyr::select(aln_pos, seq_pos, seq_name) %>%
  tidyr::spread(key = seq_name, value = seq_pos)
```
```{r Mask_IAA_alignment, eval=FALSE}
aliscore(as.AAbin(alignment[[2]]), path = "/Volumes/SSD/Applications/ALISCORE_v2.0/Aliscore_v.2.0/")
id <- scan(file = "input.fas_List_random.txt", sep = " ", quiet = TRUE)
AAAlign_no_gaps <- as.matrix(alignment[[2]])[, id]
write.nexus.data(AAAlign_no_gaps, file = "manuscript/IAAs.nex", format = "protein")
```
```{bash IAA_phylogeny_construction, eval=FALSE}
~/Desktop/natural-variation-webtool/manuscript

/Applications/MrBayes/mb

execute IAAs.nex
# set IAA33 as the outgroup
outgroup AT5G57420.1

lset nucmodel=protein Rates=invgamma
# or include Nst=6  ngammacat=4 omegavar=M3 
prset aamodelpr=fixed(jones) statefreqpr=fixed(empirical) 
# initial run with aamodelpr=mixed yielded posterior prob of 1 for jones model
propset ExtTBR$prob=0

mcmcp ngen=250000 nchains=4 printfreq=1000 samplefreq=100 burnin=200

mcmc
sump
sumt

```
```{r Load IAA phylogeny and data}
tree <- treeio::read.beast(file = "trees/IAAs.nex.con.tre")


IAA_data <- read.csv("tables/IAAs_table.csv")
#tip_labels need to be in the first column
IAA_data %<>% dplyr::select(name, everything())
IAA_data$name %<>% as.character()
IAA_data$AGI %<>% as.character()

IAA_data <- IAA_data[match(gsub(x = tree@phylo$tip.label, pattern = "\\.1",replacement = ""), IAA_data$AGI), ] 
tree@phylo$tip.label <- IAA_data$name
```


```{r A-IAA-alignment, fig.cap="(ref:A-IAA-alignment)"}
A_IAAs <- subset(gene_info, tair_symbol %in% IAA_data[which(IAA_data$Class == "A"), "name"])
A_IAAs <- alignCDS(A_IAAs$transcript_ID)
consensus <- BALCONY::RealValET_conservativity(as.alignment(Biostrings::as.matrix(A_IAAs[[2]])))

# make an alignment data frame
A_IAAs <- makeAlnDF(A_IAAs[[2]])
A_IAAs <- left_join(A_IAAs, dplyr::select(gene_info, "tair_locus",
                                       "tair_symbol", "transcript_ID"),
                        by = c("transcript_ID" = "transcript_ID"))
A_IAAs$tair_symbol %<>% as.factor()
A_IAAs$tair_symbol %<>% fct_reorder(
    {as.integer(str_remove(A_IAAs$tair_symbol, 
                           pattern = "IAA"))}, 
    .desc = TRUE)
VCF <- ldply(.data = VCF, .fun = subset, gt_GT != "0|0")
coding_VCF <- getCodingDiv(VCF)
A_IAAs <- addSNPsToAlnDF(aln_df = A_IAAs,SNPs = coding_VCF, effect_order = SNPeff_order)
A_IAAs$gap <- A_IAAs$seq_pos == "-"

p <-ggplot(A_IAAs, aes(x = aln_pos, y = tair_symbol)) +
  geom_raster(data = A_IAAs, 
             # mapping = aes(fill = effects)) +
              mapping = aes(fill = strength,
                            alpha = gap)) +
  scale_alpha_manual(values = c("TRUE" = 0, "FALSE" = 1), 
                     labels = c("aligned", "gap"), 
                     guide = guide_legend(override.aes = 
                                            list(fill = c("grey85", "white")),
                       title = "alignment",
                       direction = "vertical")) + 
   scale_fill_viridis(breaks = range(SNPeff_order$strength),
                      limits = range(SNPeff_order$strength),
                      na.value = "grey85",
                      direction = -1,
                      labels = c("weak", "strong"),
                      option = "A", 
                      guide = guide_colorbar(ticks = FALSE, 
                                             title.vjust = .8, 
                                             title.position = "top", 
                                             title.hjust = 0.5)) +
  scale_y_discrete(expand = expand_scale(add = c(1,2.5))) +
  scale_x_continuous(expand = c(0,0)) +  # expand increases distance from axis
  labs(x = "codon position in alignment", y = "", 
       fill = "variant functional effect", title = "") +
  theme_logo(base_family = "Helvetica") +
  theme(panel.grid = element_blank(), 
        panel.grid.minor = element_blank(),
        strip.background = element_blank(),
        strip.text.x = element_blank(),
        #panel.background = element_rect(fill = "grey85", color = "white"),
        legend.position = c("bottom"),
        legend.key = element_rect(linetype = "solid"),
        legend.spacing = unit(4, "char"), 
        legend.box.spacing = unit(0.1,"line")) +
  annotation_raster(matrix(gray(rescale(consensus, to = c(0,1))), nrow = 1),  xmin = -Inf, xmax = Inf, ymin = -10, ymax = 0.4)

p + annotate("rect", xmin = 69, xmax = 75, ymin = 0.4, 
             ymax = length(levels((A_IAAs$tair_symbol))) + 0.6, 
             color= "black", fill = NA) + 
  annotate("text", x = 72, y = length(levels((A_IAAs$tair_symbol))) + 1.2, 
                          label = "EAR domain", family="Arial") + 
  annotate("rect", xmin = 193, xmax = 202, ymin = 0.4, 
           ymax = length(levels(A_IAAs$tair_symbol)) + 0.6, color= "black", 
           fill = NA) + 
  annotate("text", x = (202-193)/2+193, 
           y = length(levels(A_IAAs$tair_symbol)) + 1.2, label = "degron", 
           family="Arial") + 
  annotate("rect", xmin = 245, xmax = 247, ymin = 0.4, 
           ymax = length(levels(A_IAAs$tair_symbol))+0.6, color = "black", 
           fill = NA) +
  annotate("rect", xmin = 255, xmax = 257, ymin = 0.4, 
           ymax = length(levels(A_IAAs$tair_symbol))+0.6, color = "black", 
         fill = NA) +
  annotate("rect", xmin = 315, xmax = 317, ymin = 0.4, 
           ymax = length(levels(A_IAAs$tair_symbol)) + 0.6, color = "black", 
           fill = NA) +
  annotate("rect", xmin = 317, xmax = 319, ymin = 0.4, 
           ymax = length(levels(A_IAAs$tair_symbol))+0.6, color = "black", 
           fill = NA) +
  annotate("rect", xmin = 319, xmax = 321, ymin = 0.4, 
           ymax = length(levels(A_IAAs$tair_symbol)) + 0.6, color = "black", 
           fill = NA) + 
  annotate("rect", xmin = 325, xmax = 327, ymin = 0.4, 
           ymax = length(levels(A_IAAs$tair_symbol))+0.6, color = "black", 
           fill = NA) +
  annotate("text", x = (327-245)/2+245, y= length(levels(A_IAAs$tair_symbol)) + 1.2, label = "PB1 domain", family="Arial")

#ggsave("A-IAAs_alignment.png", width = 6, height = 4)
```

(ref:A-IAA-alignment) **Critical functional domains of the conserved *Aux/IAA* genes show low nonsynonymous variation compared to regions of unknown functional importance.** Protein sequences were aligned with DECIPHER [@wright_decipher_2015] and variants were mapped to this alignment and colored according to the predicted functional effect of the allele of strongest effect at that position, with light colors having weaker effects on function and darker colors stronger effects. Red indicates missense variants. Color scale is further explained in [Methods](#methods). Alignment consensus is shown in grayscale underneath the plot as measured by Evolutionary trace [@pluciennik_balcony_2018; @wilkins_evolutionary_2012], with high consensus positions in black and low consensus in white. Key functional domains are outlined in black and labeled above the plot. The EAR domain spans codon alignment positions 70 to 74, corresponding to IAA1 amino acids LRLGL, 14-18. The degron domain spans alignment positions 194-201, corresponding to IAA1 amino acids QIVGWPPV, 55-62. The charged residues of the PB1 domain correspond to alignment positions 246, 256, 316, 318, 320, and 326 corresponding to IAA1 amino acids K77, R88, D133, D135, D137, and D143.

Natural variation also provides a means to study how gene families are evolving. To visualize this, we used ViVa to map the diversity at nonsynonymous variant sites relative to synonymous variant sites (${\pi_N/\pi_S}$) onto the Aux/IAA phylogenetic tree  (Figure \@ref(fig:IAAtree)). This visualization enables straightforward comparison of rates of recent functional divergence as predicted by diversity within natural variation in the context of the sequence divergence throughout the history of a gene family. Clades and individual genes exhibiting low nonsynonymous diversity are likely functionally conserved. Conversely, genes with high nonsynonymous diversity are more likely to be under relaxed functional selection, indicating the possibility of functional drift, emergence of novel function, or pseudogenization.

Previous research has found evidence of both broad genetic redundancy among the *Aux/IAAs* and also specificity within closely related pairs or groups of Aux/IAA proteins [@overvoorde_functional_2005; @winkler_variation_2017]. For example, the *iaa8-1 iaa9-1* double mutant and the *iaa5-1 iaa6-1 iaa19-1* triple mutant have wild-type phenotypes [@overvoorde_functional_2005], yet the *IAA6/IAA19* sister pair has significant differences in expression patterns, protein abundances and functions suggesting they have undergone functional specialization since their divergence [@winkler_variation_2017]. A closer examination of the *IAA19* and *IAA6* pair within *Brassicaceae* found evidence for positive selection and subfunctionalization of *IAA6* relative to *IAA19* [@winkler_variation_2017]. Consistent with these results, ViVa revealed higher conservation (*i.e.* lower ratio of nonsynonymous to synonymous diversity) for *IAA19* (${\pi_N/\pi_S}$ = 0.55) compared to *IAA6* (${\pi_N/\pi_S}$ = 2.3) (Figure \@ref(fig:IAAtree)), and also detected high nonsynonymous diversity within the same regions of *IAA6* as seen by Winkler et al. (Figure \@ref(fig:IAA6-div-plot)). This pattern---one sister showing high nonsynonymous diversity while the other sister was more conserved---was observed frequently across the *Aux/IAA* as well as the *AFB* and *ARF* families (Figure \@ref(fig:AFBtree) and \@ref(fig:ARFtree)), suggesting this could be a recurring feature in the evolution of these families supporting the large diversity in auxin functions. 


(ref:IAAtree) **IAA protein sequence phylogenetic tree mapped with ${\pi_N/\pi_S}$ reveals patterns of sister pair diversity/conservation.** Protein sequences were aligned with DECIPHER [@wright_decipher_2015] and low information content regions were masked with Aliscore [@kuck_parametric_2010] prior to inferring a phylogeny with MrBayes [@ronquist_mrbayes_2003]. Tips of the tree are mapped with circles of color and diameter proportional to ${\pi_N/\pi_S}$. The ${\pi_N/\pi_S}$ statistic provides a prediction of functional diversity. Nodes are labeled with the posterior probability of monophyly, a measure of confidence in the branch assignment, with one representing high confidence and zero, low confidence. There are two distinct clades of Aux/IAAs represented by the majority of the A and B classes. C class Aux/IAAs are missing one or more of the canonical Aux/IAA domains. These classes are represented by the text color of the gene name.

```{r Load IAA phylogeny and data 2}
tree <- treeio::read.beast(file = "trees/IAAs.nex.con.tre")


IAA_data <- read.csv("tables/IAAs_table.csv")
#tip_labels need to be in the first column
IAA_data %<>% dplyr::select(name, everything())
IAA_data$name %<>% as.character()
IAA_data$AGI %<>% as.character()

IAA_data <- IAA_data[match(gsub(x = tree@phylo$tip.label, pattern = "\\.1",replacement = ""), IAA_data$AGI), ] 
tree@phylo$tip.label <- IAA_data$name
```
```{r IAAtree, fig.height=5.5, fig.width=5, fig.cap="(ref:IAAtree)"}
p <- ggtree(tree, layout = "rectangular", color = "grey") %<+% IAA_data + theme_tree() + scale_x_continuous(expand = c(0, 1.2)) + 
  geom_tippoint(mapping = aes(size = 4*PiN_PiS, fill = PiN_PiS), shape = 21, color = "white", show.legend = TRUE) + 
  scale_fill_viridis(values = rescale(c(min(IAA_data$PiN_PiS), 1,
                                        max(IAA_data$PiN_PiS)), 
                                      to = c(0,1)), guide = "colourbar", breaks = c(min(IAA_data$PiN_PiS), 1,
                                        max(IAA_data$PiN_PiS)), labels = c(min(IAA_data$PiN_PiS), 1,
                                        max(IAA_data$PiN_PiS))) + scale_size(guide = "none") + 
  theme(legend.position = c(.85,.7)) + 
  geom_nodelab(mapping = aes(label = round(as.numeric(prob),digits = 2)), nudge_x = -.3, nudge_y = .5, size = 3) + 
  geom_tiplab(mapping = aes(color = Class), hjust = -0.25) + 
  scale_color_viridis(end = .9, discrete = TRUE, option = "A") +  
  labs(fill = expression(pi [N] / pi [S]), color = "Aux/IAA class")
p

#ggsave("IAA-tree.png", width = 6, height = 5.5)
```

```{r IAA-tree-order}
IAA_order <- fct_inorder(na.exclude(p$data$label[base::order(p$data$y, decreasing = TRUE)]), ordered = TRUE)
```


The *Aux/IAA* phylogeny clusters into two distinct clades represented by the A and B classes [@remington_contrasting_2004]. The C class Aux/IAAs are missing one or more of the canonical Aux/IAA domains. We found notable exceptions to the pattern of diversification and conservation between sister pairs within the Class B *Aux/IAA* genes. The *IAA10*/*IAA11*, *IAA18*/*IAA26*, and *IAA20*/*IAA30* pairs showed similar levels of nonsynonymous diversity. For example, *IAA10* and *IAA11* both showed functional conservation (${\pi_N/\pi_S}$ of 0.80 and 0.67 respectively). The *Arabidopsis thaliana* ePlant browser indicates that *IAA10* and *IAA11* have almost identical expression patterns [@waese_eplant_2017]. Together this evidence suggests a strong dosage requirement for these genes or that they have taken on novel functions since their emergence, with both genes contributing similarly to plant fitness. In support of novel function, expression and mutant analysis during embryogenesis suggest that IAA10 is required for suspensor-hypophysis transition, while IAA11 is involved in later cell fate transitions [@rademacher_different_2012].

