# *TIR1/AFB* genes {#AFBs} 

Auxin acts by binding to receptors (Auxin-signaling F-Boxes, or AFBs) that in turn target co-repressors (Aux/IAAs) for degradation. The six auxin receptor genes in the model plant *Arabidopsis thaliana* , *TIR1* and *AFB1-5*, evolved through gene duplication and diversification early in the history of vascular plants [@parry_complex_2009]. The rate of co-repressor degradation is determined by the identity of both the receptor and co-repressor [@havens_synthetic_2012], and this rate sets the pace of lateral root development [@guseman_auxininduced_2015]. 

All members of this family have been shown to bind auxin and Aux/IAA proteins. However, AFB1 has drastically reduced ability to assemble into an SCF complex, due to the substitution E8K in its F-box domain, preventing it from inducing degradation of Aux/IAAs[@yu_untethering_2015]. This lack of complexation may allow observed high ubiquitous AFB1 accumulation [@parry_complex_2009]. Higher order mutants in the family containing *afb1* mutants suggest that *AFB1* has a moderate positive effect on auxin signaling. Additionally, AFB4 and AFB5 have been shown to preferentially and functionally bind the synthetic auxin picloram, while other family members preferentially bind indole-3-acetic acid [@prigge_arabidopsis_2016]. Interestingly, the strength and rate with which TIR1/AFBs are able to bind and mark Aux/IAAs for degradation are variable [@calderonvillalobos_combinatorial_2012; @havens_synthetic_2012]. AFB2 induced the degradation of certain Aux/IAA proteins at a faster rate than TIR1, suggesting some functional specificity has arisen since the initial duplication between the *TIR1/AFB1* and *AFB2/AFB3* clades.

Although it is unclear what unique roles each receptor plays in growth and development, a number of studies have pointed out differences in the ways the six different receptors in A. thaliana differ in biochemical function and expression domain[@dharmasiri_plant_2005; @parry_complex_2009; @prigge_arabidopsis_2016]. *TIR1/AFB* genes are expressed ubiquitously in A. thaliana tissues, TIR1, AFB2, and AFB3 have been shown to accumulate in the shoot and root meristems and leaf tissues, with slightly different expression patterns for TIR1 [@parry_complex_2009]. Additionally, the expression of AFB5 is strongly circadian-regulated [@covington_circadian_2007] and AFB3 is more highly expressed in the roots in the presence of nitrate, facilitating increased lateral root formation [@vidal_nitrate_2014], suggesting more broad environmental regulation of this gene family may exist. 

All members of this family have been shown to bind auxin and Aux/IAA proteins. However, AFB1 has drastically reduced ability to assemble into an SCF complex, due to the substitution E8K in its F-box domain, preventing it from inducing degradation of Aux/IAAs[@yu_untethering_2015]. This lack of complexation may be the cause of high, ubiquitous AFB1 accumulation [@parry_complex_2009]. Higher order mutants in the family containing *afb1* mutants suggest that *AFB1* has a moderate positive effect on auxin signaling. Additionally, AFB4 and AFB5 have been shown to preferentially bind the synthetic auxin picloram, while other family members preferentially bind indole-3-acetic acid [@prigge_arabidopsis_2016]. Interestingly, the strength and rate with which each TIR1/AFB is able to bind and mark an Aux/IAAs for degradation is variable [@calderonvillalobos_combinatorial_2012; @havens_synthetic_2012]. 


(ref:TIR1AFBdivstats) **_TIR1/AFB_ nucleotide diversity.** The nucleotide diversity as Nei and Li's $\pi$ (the average number of nucleotide differences per site between all possible pairs of sequences at synonymous ($\pi_S$) and nonsynonymous ($\pi_N$) sites, as well as all sites in the coding sequence and full transcript of the *TIR1/AFB* family.

```{r TIR1AFBdivstats, fig.cap="(ref:TIR1AFBdivstats)"}
gene_info <- geneInfoFromFile(
  fname = system.file("shiny-app", "gene_family_data",
                      "AFB_gene_ids.csv", package = "r1001genomes"))

VCF <- readRDS(system.file("shiny-app", "gene_family_data",
                      "AFBs_VCF.rds", package = "r1001genomes"))
div_stats <- ldply(VCF, diversityStats, geneInfo=gene_info, .id="transcript_ID")
div_stats$tair_symbol %<>% fct_reorder(.x =  div_stats$tair_symbol, 
                                       .fun = function(x) 
                                         {as.integer(str_remove(x, 
                                              pattern = "AFB|TIR"))})
div_stats$tair_symbol %<>% fct_relevel("TIR1")
div_stats %<>% arrange(tair_symbol)
div_stats %<>% select(tair_symbol, everything())

if (knitr::is_html_output()) {
  formatRound(datatable(div_stats,
                  colnames = c("transcript",
                               "symbol",
                               "&pi;<sub>N</sub>",
                               "&pi;<sub>S</sub>",
                               "&pi;<sub>N</sub>/&pi;<sub>S</sub>",
                               "&pi; coding",
                               "&pi; transcript"),
                  rownames = FALSE, escape = FALSE,
                  options = list(paging=FALSE, searching=FALSE)),
                columns = 2:7, digits = 6)
  } else 
{
  dt <- div_stats
  names(dt) <- c("transcript",
                                 "symbol",
                                 "&pi;~N~",
                                 "&pi;~S~",
                                 "&pi;~N~/&pi;~S~",
                                 "&pi; coding",
                                 "&pi; transcript")
  dt %<>% mutate_if(is.numeric, round, 6)
  names(dt) <- paste0("**", names(dt), "**")
  kable(dt, caption = "(\\#tab:TIR1AFBdivstats)") 
}
```

Examining the natural sequence variation across the *AFB* family \@ref(fig:TIR1AFBdivstats) revealed that *TIR1* and *AFB1* both had very low nonsynonymous diversity, hinting at their likely functional importance and bringing in to question the inconclusive role of *AFB1* in auxin signaling. *AFB3* and *AFB4* had higher nonsynonymous diversity, while their sister genes, *AFB2* and *AFB5* were more conserved. This matches our current understanding of *AFB3* as playing a minor role in the auxin signaling pathway and *AFB4* perhaps undergoing pseudogenization. 

Although most known functional regions are highly conserved in *AFB1*, there is a nonsynonymous polymorphism  in the oligomerization domain, only found in the Can-0 accession. Mutations in this domain of *TIR1* frequently have a semidominant effect on root phenotypes [@dezfulian_oligomerization_2016; @wright_insights_2017]. Characterization of this allele and accession may help determine the role of *AFB1* in this pathway.

```{r AFB variants for PyMOL}
alignment <- alignCDS(gene_info$transcript_ID)
# make an alignment data frame
aln_df <- makeAlnDF(alignment[[2]])

# make a collapsed alignment for translating positions
collapsed_aln_df <- 
  aln_df %>% dplyr::select(aln_pos, seq_pos, seq_name) %>%
  tidyr::spread(key = seq_name, value = seq_pos)
coding_vcf <- ldply(.data = VCF, .fun = subset, 
             !is.na(Transcript_ID) & gt_GT != "0|0")
aln_df <- addSNPsToAlnDF(aln_df, coding_vcf, seq_name = Transcript_ID, 
               seq_pos = Codon_Number)
collapsed_aln_df %<>% left_join(y = subset(aln_df, seq_name == gene_info[gene_info$tair_symbol == "TIR1", "transcript_ID"]))

#' Select variants for importing into PyMOL
#'
#' @param gene_info a gene info data frame
#' @param aln_df an alignment data frame with attached SNPs
#' @param structure_ref string, which tair_symbol or transcript_ID (depending on 
#' \code{name_by_symbols}) corresponds to the protein structure to which 
#' selections should be mapped
#' @param effect_type string of the effect type to select from the alignment 
#' data frame
#' @param name_by_symbols logical, if true (default) tair_symbols will be used 
#' to name the selections, otherwise transcript_IDs will be used
#'
#' @return
#' @export
#'
#' @examples
select_variants <- function(gene_info, aln_df, structure_ref, 
                            effect_type = "missense", name_by_symbols = TRUE){
  collapsed_aln_df <- aln_df %>% as_tibble() %>% 
    dplyr::select(aln_pos, seq_pos, seq_name) %>%
    tidyr::spread(key = seq_name, value = seq_pos)
  if(name_by_symbols)
    column <- "tair_symbol"
    else 
      column <- "transcript_ID"
  
  selections <- ldply(.data = gene_info[,column], .fun = function(gene){
    collapsed_aln_df %<>% dplyr::left_join(y = subset(aln_df, seq_name ==  
                      gene_info[gene_info[, column] == gene, "transcript_ID"]))
    variants <- collapsed_aln_df[
      grepl(effect_type, collapsed_aln_df$effects), 
      gene_info[gene_info[, column] == structure_ref, "transcript_ID"]]
    c(gene, paste0(na.omit(as.numeric(variants[[1]])), collapse = "+"))
  })
  return(selections)
}

selections <- select_variants(gene_info, aln_df, structure_ref = "TIR1")
selections$V2 <- paste0("chain B and resi ", selections$V2)
#write.table(selections, file = "AFB_variant_selections", row.names = FALSE, 
#          col.names = FALSE, sep = ",", quote = FALSE)
```

```{python readSelection, eval=FALSE}
#
# -- readSelection.py
#
from pymol import cmd, stored

def readSelection (fileName):
	'''
	DESCRIPTION

	reads a csv file containing a column of selection names and a column of
	selection strings and creates selection objects for each
	'''
	#read in the selection file
	with open(fileName) as f:
		lines = f.readlines()
		for line in lines:
			sel = line.split(",")
			cmd.select(sel[0], sel[1])

	f.close()

cmd.extend('readSelection',readSelection);
```

```{PyMol, AFB_structure, eval=FALSE}
#from within pymol
cmd.load("Dropbox/180705_NIH_renewal/Aim1-figures/2P1Q.pdb")
hide everything

run ~/Desktop/natural-variation-webtool/manuscript/readSelection.py

readSelection ~/Desktop/natural-variation-webtool/manuscript/AFB_variant_selections

# Define selections
# select TIR1variants, (chain B and resi 154+239+546)
select fbox, (chain B and resi 1-50)
select LRR, (chain B and resi 50-576)
select TIR1, (chain B and resi 1-576)
# select AFB2variants, (chain B and resi 38+174+181+184+209+259+401+496)
select tir1_1, (chain B and resi 147)
select TIR1dm, (chain B and resi 170+473)
select insPcontacts, (chain B and resi 74+78+113+114+344+401+403+436+460+484+485+509)
select dimerNULL, (chain B and resi 142+143+151+117+147)
select auxin, resn IAC
select AUXcontacts, (chain B and resi 78+79+82+378+380+403+404+405+438+439+462+463+464+489)
select insP, resn IHP
select IAA7, chain C
select IAA7contacts, (chain B and resi 81+82+84+139+347+349+350+351+352+380+405+406+407+409+464+465+489)
select IAA7_N, chain C and resi 1
select IAA7_C, chain C and resi 13
select ASK1, chain A
select Y, chain C and resi 11
select Ycontacts, chain B and resi 165+166+195

#Define color palette
#set_color name, [ red-float, green-float, blue-float ]
set_color blue1, [0.00000000, 0.20392157, 0.6980392]
set_color blue2, [0.09803922, 0.36078431, 1.0000000]
set_color blue3, [0.00000000, 0.29019608, 1.0000000]
set_color teal,  [0.09803922, 1.00000000, 0.6980392]
set_color LRR,   [1.00000000, 0.71764706, 0.0000000]
set_color Fbox,  [0.69803922, 0.49411765, 0.0000000]
set_color TIR1l, [0.67058824, 0.07843137, 0.8000000]
set_color TIR1f, [0.63529412, 0.34117647, 0.6980392]
set_color AFB2l, [0.37647059, 0.70196078, 1.0000000]
set_color AFB2f, [0.07843137, 0.64313725, 0.8000000]

show cartoon, 2P1Q

color lrr, LRR

color TIR1l, TIR1
show sticks, TIR1
color TIR1f, AFB1
show sticks, AFB1
color AFB2l, AFB2
show sticks, AFB2
color AFB2f, AFB3
show sticks, AFB3

set ray_trace_mode, 3

# used get_view() to return the current view (position) of the molecule 
# copy the code to be able to set the identical viewpoint later
# to return the current view from Pymol
# get_view

set_view (\
     0.044671144,    0.083668359,   -0.995409012,\
     0.774909019,    0.625905454,    0.087390237,\
     0.630409002,   -0.775298953,   -0.036878210,\
    -0.001916260,    0.005584979, -202.498916626,\
    -5.538011074,   -8.507736206,   39.046234131,\
  -219.237945557,  624.123413086,  -20.000000000 )
  
#to render a high definition image use ray
ray
```

(ref:AFB_structure_vars) A TIR1 structure (2P1Q) in gold, with residues 

```{r AFB_structure_vars, fig.cap='(ref:AFB_structure_vars)'}
#knitr::include_graphics()
```


```{r AFB1 diversity plot}
coding_variants <- getCodingDiv(VCF[[gene_info[gene_info$tair_symbol == "AFB1", "transcript_ID"]]])
p <- plotCodingDiv(uniqueCodingVars = coding_variants)
annotation <- readAnnotationFile(filename = system.file("shiny-app", "gene_family_data",
                      "AFB_annotations.csv", package = "r1001genomes"), gene_info = gene_info)
annotation <- addAlnPosToAnno(anno_df = annotation, aln_df)

# translate positions from TIR1 to AFB1
AFB1_positions <- annotation$positions
AFB1_positions %<>% left_join(y = select(collapsed_aln_df, aln_pos, gene_info[gene_info$tair_symbol == "AFB1", "transcript_ID"]), by = "aln_pos") 
AFB1_positions$position <- AFB1_positions$AT4G03190.1

p <- append_layers(p,list(
        geom_rect(data = AFB1_positions,
                   mapping = aes(xmin = as.integer(position)-1,
                                 xmax = as.integer(position)+2,
                                 fill = annotation),
                  ymin = -Inf, ymax = Inf, inherit.aes = FALSE, alpha = 0.8)),
      position = "bottom")
p
```


The AFB4 and AFB5 receptors have an N-terminal extension prior to the F-box domains. This extension had very high nonsynonymous diversity \@ref(fig:FBoxAlignment), suggesting that this extension does not play an important functional role in these proteins. Additionally, two frameshift variants and one stop-gained variant were observed in *AFB4* supporting its pseudogenization. 


(ref:FBoxAlignment) **Alignment of A. thaliana TIR1/AFB F-box domains showing variants.** Cyan marks the F-box binding domain. Yellow, green, and magenta, mark missense and synonymous, synonymous only, and missense only variant locations respectively. Yellow squared E’s show Cul1-interacting residues, differing in AFB1.

```{r FBoxAlignment, fig.cap = "(ref:FBoxAlignment)"}
aln_df <- makeAlnDF(alignment[[2]])
aln_df <- left_join(aln_df, dplyr::select(gene_info, "tair_locus",
                                       "tair_symbol", "transcript_ID"),
                        by = c("transcript_ID" = "transcript_ID"))
aln_df$tair_symbol <- fct_rev(fct_relevel(aln_df$tair_symbol, "TIR1"))
aln_df <- chunkAlnDF(aln_df = aln_df, chunk_width = 55)
VCF <- ldply(.data = VCF, .fun = subset, gt_GT != "0|0")
VCF <- getCodingDiv(VCF)
aln_df <- addSNPsToAlnDF(aln_df, VCF)
F_box <- subset(aln_df, aln_pos <= 96)
F_box <- chunkAlnDF(aln_df = F_box, chunk_width = 80)
p <-ggplot(F_box, aes(x = aln_pos, y = tair_symbol,
                             group = seq_pos, text = variants))
p <- p +
  geom_tile(data = na.omit(F_box), mapping = aes(fill = effects),
            width = 1, height = 1, alpha = 0.8) +
  geom_text(aes(label=letter), alpha= 1, family = "Courier") +
  scale_fill_brewer(type = "qual", palette = 1, direction = -1) +
  scale_x_continuous(breaks=seq(1,max(F_box$aln_pos), by = 10)) +
  scale_y_discrete() +
  # expand increases distance from axis
  labs(x = "", y = "", fill = "variant") +
  theme_logo(base_family = "Helvetica") +
  theme(panel.grid = element_blank(), panel.grid.minor = element_blank()) +
  facet_wrap(facets = ~chunk, ncol = 1, scales = "free") +
  theme(strip.background = element_blank(),
        strip.text.x = element_blank(),
        legend.box = "vertical")
p
```
