# Methods {#methods}

## Data Sources

### Variant Data
Variant data was queried from the 1001 genomes project (http://1001genomes.org) via URL requests to their API service (http://tools.1001genomes.org/api/index.html). These queries returned subsets of the whole-genome variant call format (VCF) file as SnpEFF VCF files. The whole-genome VCF file can be found on the project’s website at http://1001genomes.org/data/GMI-MPI/releases/v3.1/. 

### Germplasm Accession Information
A dataset of each of the 1135 accessions including CS stock numbers and geographic locations where the samples were collected was retrieved from the 1001 Genomes website at http://1001genomes.org/accessions.html, via the download link at the bottom of the page. This data file has been embedded in the R package as `accessions`.

### Gene and Transcript Accession Information
Information on the genes and transcripts including chromosomal coordinates, start and end location, and transcript length were downloaded from Araport11 as a General Feature Format (GFF) file [@cheng_araport11_2017]. The Araport11 full genome general feature format file, which can also be found on the TAIR website (https://www.arabidopsis.org/download/index-auto.jsp?dir=%2Fdownload_files%2FGenes%2FAraport11_genome_release), has been embedded in the R-package as GRanges object, `gr`. The TAIR10 database, found at http://arabidopsis.org, was accessed via the `biomart` function, using the biomaRt R-package. Gene identifiers used in this study are in Table \@ref(tab:gene-ids-table).
```{r gene-ids-table}
AFBs <- read.csv(system.file("shiny-app", "gene_family_data",
                      "AFB_gene_ids.csv", package = "r1001genomes"))
IAAs <- read.csv(system.file("shiny-app", "gene_family_data",
                      "IAA_gene_ids.csv", package = "r1001genomes"))
TPLs <- read.csv(system.file("shiny-app", "gene_family_data",
                      "TPL_gene_ids.csv", package = "r1001genomes"))
ARFs <- read.csv(system.file("shiny-app", "gene_family_data",
                      "ARF_gene_ids.csv", package = "r1001genomes"))
gene_ids <- bind_rows(AFBs, IAAs, TPLs, ARFs)
kable(gene_ids, col.names = c("Arabidopsis AGI locus identifier", "Gene symbol", "Clade/class"),caption = "Full list of genes used in this study, by identifier, symbol and classification.")
```

## Ranking of variant functional effects
Alignments were colored according to the strongest effect variant allele occurring at any frequency at that position as reported in the SnpEFF "effect" field, per the scale in Figure \@ref(fig:SNPeff-order). 

(ref:SNPeff-order) **Rank order of the strength of functional effects** Variant effect classes were ordered by subjective prediction of average strength of effect on gene function. Strength was then assigned to each effect on an integer scale.
```{r SNPeff-order, fig.cap="(ref:SNPeff-order)"}
SNPeff_order$effect <- factor(as.character(SNPeff_order$effect), levels = as.character(SNPeff_order$effect), ordered = TRUE)


ggplot(data = SNPeff_order, aes(x = effect, y = 1)) + 
  geom_raster(mapping = aes(fill = strength)) +
  scale_fill_viridis(breaks = range(SNPeff_order$strength),
                      limits = range(SNPeff_order$strength),
                      na.value = "grey85",
                      direction = -1,
                      labels = c("weak", "strong"),
                      option = "A", 
                      guide = "none") + 
  coord_flip() + 
  theme_minimal() + theme(axis.text.x = element_blank()) + 
  geom_text(mapping = aes(y = 1, label = strength, color = strength)) +
  scale_color_gradient(low = "black", high = "white", guide = "none") + 
  labs(y = "strength of functional effect", x = "variant effect")

```

 
## Nucleotide Diversity Calculation
Nei and Li defined the nucleotide diversity statistic in their original paper as:
"the average number of nucleotide differences per site between two randomly chosen DNA sequences” [@nei_mathematical_1979], and provided the equation: 

\begin{equation}
\displaystyle \pi = \sum_{ij}x_{i}x_{j}\pi_{ij}.
(\#eq:nei)
\end{equation}

Where $x_{i}$ is the frequency of the ith sequence in the population and $\pi_{ij}$ is the number of sites that are different between the $i$th and $j$th sequence divided by sequence length.

A more general form, that treats each sequence in the population as unique can be written as:

\begin{equation}
\displaystyle \pi = \frac{1}{L*n^2} \sum_{i=1}^n \sum_{j=1}^n \sum_{k=1}^L{\pi_{ijk}}
\qquad
\displaystyle \pi_{ijk} = \left\{\begin{array}{l}
1 \quad \textrm{if} \quad N_{ik} \neq N_{jk} \\
0 \quad \textrm{if} \quad N_{ik} = N_{jk}
\end{array}\right.
(\#eq:nei-gen)
\end{equation}

where $N_{ik}$ is the nucleotide (A, T, C or G) at position $k$ on the *i*th sequence of the population.
$L$ is the length of the sequence. Indels are excluded from the diversity calculation leading to a single $L$ for the population. $n$ is the total number of sequences in the population.

From this form we can re-arrange summations to the form below:

\begin{equation}
\displaystyle \pi = \frac{1}{L} \sum_{k=1}^L{\pi_{k}}
\qquad
\pi_{k} = \frac{1}{n^2} \sum_{i=1}^n \sum_{j=1}^n {\pi_{ijk}}
(\#eq:nei-rearrange)
\end{equation}

where $\pi_k$ can be thought of as the site-wise nucleotide diversity at position $k$, and is equal to 
the nucleotide diversity of a sequence of length 1 at location $k$. We can calculate $\pi_k$ for each
site, then average those over the sequence length to calculate $\pi$, the nucleotide diversity 
of the sequence.

The function `Nucleotide_diversity` in the r1001genomes package calculates $\pi_k$ for each position in the gene or region that contains a variant. Note, $\pi_k$ is equal to 0 at all locations without variants.
This is also what is displayed in the Diversity Plot tab of the webtool.

## Detailed $\pi_k$ calculation simplification.

The formula for $\pi_k$ above requires comparing every sequence to every other sequence at location k,
however, we know there are only a few variant forms at each individual location.

So, we can revert back to using Nei and Li's original formula \@ref(eq:nei), modifying it slightly,
replacing $x_i$ with $\frac{n_i}{n}$, $n_i$ being the number of sequences in the population with nucleotide $N_i$ at location $k$:

\begin{equation}
\displaystyle \pi_k = \sum_{ij} \frac{n_i}{n}\frac{n_j}{n} \pi_{ij} = \frac{1}{n^2}\sum_{ij} n_i n_j \pi_{ij}
\qquad
\pi_{ij(k)} = \left\{\begin{array}{l}
1 \quad \textrm{if} \quad i \neq j \\
0 \quad \textrm{if} \quad i = j
\end{array}\right.
(\#eq:nei-simp1)
\end{equation}

Note that in equation \@ref(eq:nei) subscripts $i$ and $j$ are summed over all sequences in the population, however in equation \@ref(eq:nei-simp1) $i$ and $j$ are only summed over unique variants at a particular location k.

We will define $n_{!i} = n - n_i$ as the number of sequences different from $i$ at position $k$. We can also see that the summed term will be zero if $i=j$, and $n_i n_j$ if $i\neq j$. Therefore:

\begin{equation}
\displaystyle \pi_k = \frac{1}{n^2} \sum_{i} n_i n_{!i}
(\#eq:nei-simp2)
\end{equation}

Next we substitute our definition of $n_{!i}$:

\begin{equation}
\displaystyle \pi_k = \frac{1}{n^2} \sum_{i} n_i (n - n_{i})
(\#eq:nei-simp3)
\end{equation}

Distributing and splitting summation yields:

\begin{equation}
\displaystyle \pi_k = \frac{1}{n^2} (n \sum_{i} n_i - \sum_{i} n_{i}^2)
(\#eq:nei-simp4)
\end{equation}

Finally, summing $\displaystyle \sum_{i}n_i$ is equal to $n$:

\begin{equation}
\displaystyle \pi_k = \frac{1}{n^2} (n^2 - \sum_{i} n_{i}^2)
(\#eq:nei-simp5)
\end{equation}

This simplified form for $\pi_k$ is used by the app, because the counts of unique variants at a single nucleotide location can easily be summarized in R.

## Software 

The r1001genomes package has many software dependencies on other R packages, a few of the key bioinformatics packages used are listed below.

**biomaRt:** used for accessing the TAIR10 database on arabidopsis.org

**vcfR:** used to read in the VCF files in a flat “tidy” format for easy manipulation

**BSgenome:** used as the source for the complete DNA string of the reference genome (Col-0).

**DECIPHER:** used to align nucleotide and amino acid sequences of homologous genes 

**GenomicFeatures:** used for handling sequence annotations.

**Biostrings:** provides the underlying framework for the sequence manipulations used for generating and aligning sequences with BSgenome, Decipher, and GenomicFeatures

Other packages that were critical to building ViVa and/or writing this document include: [@R-ape; @R-base; @R-bindrcpp; @R-BiocGenerics; @R-Biostrings; @R-bookdown; @R-colorspace; @R-DECIPHER; @R-dplyr; @R-DT; @R-forcats; @R-gginnards; @R-ggplot2; @R-ggpmisc; @R-ggseqlogo; @R-ggthemes; @R-ggtree; @R-ips; @R-IRanges; @R-knitr; @R-magrittr; @R-plyr; @R-purrr; @R-r1001genomes; @R-RColorBrewer; @R-readr; @R-reshape2; @R-reticulate; @R-rmarkdown; @R-RSQLite; @R-S4Vectors; @R-scales; @R-stringr; @R-tibble; @R-tidyr; @R-tidyverse; @R-treeio; @R-viridis; @R-viridisLite; @R-XML; @R-XVector]

## Testing
A group consisting of undergraduate and graduate students, post-doctoral and post-baccalaureate researchers (the authors of this publication) were assembled to test the functionality of ViVa. Testers were asked to use the ViVa web interface to analyze the genetic variation at the clade or whole gene family level of the auxin nuclear signaling pathway. Testers were provided with a brief vignette on how to use ViVa to formulate new hypotheses or support existing hypotheses about gene function and evolution, which has been expanded to the "An Overview of ViVa" section. The testers had background knowledge of these genes as members of laboratories studying auxin nuclear signaling. Testers met weekly with the developers to discuss their user experience and their findings. Tester experiences, issues, and suggestions were incorporated into the ViVa software. The results collected by these analyses are summarized in the "Visualizing Variation within the auxin signaling pathway" section and [Supplemental Analysis](#si).




