---
title: Accelerating structure–function mapping using the ViVa webtool to mine natural variation 
author:  |
  | R. Clay Wright^[Corresponding author: wrightrc@vt.edu] $^1$, 
  | Morgan O. Hamm $^2$,
  | Jennifer Nemhauser $^2$
  | $^1$Biological Systems Engineering, Virginia Tech, Blacksburg, VA 24061, 
  | $^2$Deparment of Biology, University of Washington, Seattle, WA 98195

date: "`r Sys.Date()`"

abstract: Thousands of sequenced genomes are now publically available and represent a large swath of plant evolution; yet, much of this data remains inaccessible to researchers without significant bioinformatics experience. Here, we present a webtool called ViVa (Visualizing Variation) which aims to empower any researcher to take advantage of the amazing genetic resource collected in the Arabidopsis 1001 Genomes Project. ViVa facilates data mining on the gene, gene family or gene network level. To test the utility and accessibility of ViVa, we assembled a team with a range of expertise with biology and bioinformatics to analyze the natural variation in the well-studied nuclear auxin signaling pathway. Our analysis has provided further confirmation of existing knowledge and has also helped generate new hypotheses regarding this well studied pathway. These results highlight how natural variation can be used to generate and test hypotheses about less studied gene families and networks, especially when paired with biochemical and genetic characterization. ViVa is also readily extensible to newly emerging databases of genetic variation, such as the 3,000 Rice Genomes Project. 

keywords: natural variation, Arabidopsis thaliana, genome diversity, structure, function, genotype, phenotype, bioconductor, bookdown, r markdown

bibliography:
- natural-variation.bib
- TIR1_AFB.bib
- Aux_IAA.bib
- ARF.bib
- TPL.bib
biblio-style: apalike
link-citations: yes
fig_caption: yes

---

```{r setup, message=FALSE, warning=FALSE, include=FALSE}
knitr::opts_chunk$set(warning = FALSE, message = FALSE, echo = FALSE, collapse = TRUE, comment = "#>", tidy = TRUE, cache = TRUE, cache.lazy = FALSE, out.width = "90%", fig.height=4, fig.width=7, out.extra='style="margin: auto; display: block; padding-top: 15px;"')

#devtools::install_github("wrightrc/r1001genomes", ref = "auxin-natural-variation")
library(bookdown)
library(ape)
library(r1001genomes)
library(DT)
library(tidyverse)
library(magrittr)
library(RColorBrewer)
library(ggpmisc)
library(gginnards)
library(ggthemes)
library(ggseqlogo)
library(ggtree)
library(reticulate)
library(scales)
library(viridis)
library(ips)
library(knitr)
library(dplyr)
library(stringr)

#use_python(python = "/usr/local/bin/python")
```

## List of working titles  
* Predicting structure--function relationships using natural genetic variation 
* Accelerating structure--function mapping using natural genetic variation 
* A comprehensive analysis of natural sequence variation within the <i>Arabidopsis thaliana</i> nuclear auxin signaling pathway 
* Visualizing natural variation with ViVa, a new webtool
* Accelerating structure–function mapping using a new webtool to mine natural variation


```{r include=FALSE}
# automatically create a bib database for R packages
knitr::write_bib(c(
  .packages(), 'bookdown', 'knitr', 'rmarkdown'
), 'packages.bib')
```
