---
title: "A comprehensive analysis of natural sequence variation within the <i>Arabidopsis thaliana</i> nuclear auxin signaling pathway"
author: 
  affiliation: University of Washington
  name: "R. Clay Wright"

date: "`r Sys.Date()`"

output: 
    html_document:
      toc: true
      fig_caption: yes

keywords: r markdown
fontsize: 11pt
fontfamily: Helvetica

bibliography: ["natural_variation.bib", "TIR1_AFB.bib"]
csl: apa.csl
link-citations: true
---

```{r, echo = FALSE, message=FALSE, warning=FALSE}
knitr::opts_chunk$set(warning = FALSE, collapse = TRUE, comment = "#>", tidy = TRUE, out.width = "90%", fig.height=4, fig.width=7, out.extra='style="margin: auto; display: block; padding-top: 15px;"')
library(r1001genomes)
library(DT)
library(dplyr)

```
<!--This is a CSS (Cascading Style Sheet).
 It controls how panels and images are formatted.></!-->
<style>
  img {
    display: block;
    margin: auto;
    width: 90%;
    height: auto;
  }
</style>

# List of working titles  
* A comprehensive analysis of natural sequence variation within the <i>Arabidopsis thaliana</i> nuclear auxin signaling pathway  
* *Add your suggestions* 

# Abstract  

* Motivating, high level statement  
* Statement defining the cutting edge of your research area  
* A statement of the form: "However, it remains unclear ..." that defines what problem you are  
addressing.  
* A statement of the form:"Here, we ..."  that concisely states your contribution.  
* A series of 3?5 statement about your results.  
* A concluding statement.  


    
# Related work  
What and how have we, as a field, learned from natural variation?  
[@hollister_genomic_2014; @bielawski_detecting_2013; @kryazhimskiy_population_2008; @schmid_multilocus_2005; @nieduszynski_sequence_2011; @melamed_combining_2015; @weigel_population_2015; @schmitz_patterns_2013; @weigel_natural_2005; @hardtke_case_2010; @koornneef_naturally_2004; @mitchell-olds_genetic_2006; @joly-lopez_developing_2016; @wright_insights_2017a; @park_distributions_2017; @_135_2016a]  
 
# Methods

# Results  
## TIR1/AFB genes  
Romi
[@gray_identification_1999; @kepinski_auxininduced_2004; @tan_mechanism_2007; @dezfulian_oligomerization_2016; @wu_evolution_2017b]

## Aux/IAA genes  


### Group A Aux/IAAs
Britney

### Group B Aux/IAAs
Mollye/Alex 

### Group C Aux/IAAs
Mega/Alex

## TPL/TPR genes  
  The Auxin signaling pathway utilizes the TOPLESS (TPL) family of Gro/TLE/TUP1 type co-repressor proteins to maintain auxin responsive genes in a transcriptionally-repressed state in the absence of auxin (cite). In Arabidopsis the TPL family includes TOPLESS and four TPL-related proteins (TPR1-4), which are comprised of three structural domains: an N-terminal TPL domain and two beta-propellar domains. TPL is recruited to the AUX/IAA proteins through interaction with a conserved protein motif known as an Ethylene-responsive element binding factor-associated amphiphilic repression (EAR) domain (cite). Canonical EAR domains have the amino acid sequence LxLxL, as is found in domain I of the AUX/IAAs (figure xA, cite). TPL co-repressors bind EAR domains through direct interaction with the TPL N-terminal C-terminal to LisH (CTLH) domain (citations of pre-structure founding papers/reviews). Recent structural analyses of the TPL N-terminal domain have highlighted the precise interaction interface between TPL and AUX/IAA EAR domains, as well as the TPL-TPL dimerization and tetramerization motifs (Avarelilo et al 2017, Ke et al, 2015), and even the residues required for higher-order multimers of TPL tetramers (Ma et al, 2017). The two tandem beta propellar domains of TPL/TPRs have as yet not been crystallized, however numerous high resolution crystal structures of beta propellar domains allow for structural prediction of the TPL c-terminal beta propellar domains and the locations of likely interaction faces (Figure, X). 
  
  The TOPLESS co-repressor family generally exhibits a high level of sequence conservation at the amino acid sequence level across the Arabidopsis accessions with all ${\pi_N/\pi_S}$ values below 1 (Table X). The closely related TPL and TPR1 have the highest ${\pi_N/\pi_S}$ values (TPL-0.425, TPR1-0.739), suggesting that these these two related genes tolerate a higher degree of sequence diversity compared to TPR2-4. The TPL/TPR family demonstrates a high level of sequence conservation across the N-terminal TPL domain, with the lowest of observed ${\pi_N/\pi_S}$ values (Figure X). Indeed, all non-synonymous mutations observed in this region are either in linker domains between alpha helices, or are highly conservative mutations within helices (i.e. Valine to Isoleucine), which would be predicted to exhibit little effect on TPL folding and activity. 
  
  The increased N-terminal conservation underscores the importance of the N-terminal TPL domain. For example, the initial tpl-1 mutation (N176H) in the ninth helix is a dominant gain-of-function allele (Long et al, 2002, PMID12050130), which is capable of binding wild-type TPL protein and inducing protein agreggation (Ma, 2017, pmid28630893). It is therefore understandable that very little diversity is observed in the ninth alpha helix of TPL/TPRs as mutations in this domain could act in a dominant negative fashion, rendering any de novo mutation in this region embryo lethal. However, the entire N-terminal domain exhibits very low rates of nucleotide diversity, suggesting that each domain in this region is critical at the population level. Indeed, the crystal structure of TPL has revealed that the TPL domain is responsible for the multiple roles of TPL/TPR proteins: The LisH domain and CRA domain coordinate TPL dimerization and tetramerization respectively (26601214, 28698367), the CTLH domain is critical for interaction with DNA-binding transcription factors (26601214), and helix 9 within the CRA domain may be critical for the formation of higher-order TPL complexes (multimers of TPL tetramers, 28630893). 
  
  We then compared the frequency of SNPs that cause nonsynonymous mutations across the TPL domains. Compared to the TPL domain, both the proline rich and WD-40 domains exhibit a higher rate of missense mutations accross both of the WD-40 repeat regions, here labeled WD40-1 and WD40-2 respectively. We utilized the Phyre2.0 modeling software to generate a rough model of the TPL/TPR WD-40 repeats stricly for the purpose of mapping non-synonymous mutations onto a visible protein chassis (Figure X). Here we can see both faces of each predicted beta-propellar domains, which are connected by a short linker region. Through utilization of protein modelling, it becomes clear that we observe conservation of amino acid sequence at the center of the beta-propellar domains. These observations could be used as the basis for forward mutagensis strategies to test whether this region of TPL functions similarly to its homolog TLE, which binds its target genes EH1 and the C-terminal WRPW/Y motif (Hairy/Hes/Runx) in the center of the beta-propeller domain (pmid 16762837).  
  
  It is interesting to note that despite the presence of 5 TPL/TPR genes in the genome, which are considered to be redundant (pmid 16763149), very few SNPs are observed that are predicted to result in loss of function for any one TPL/TPR gene. In fact, only one deletion was observed (665delG) that creates a frameshift mutation in TPR4 in a single accession (Aiell-4 in Italy). This mutation occurs in codon 222, just downstream of the TPL N-terminal domain, and would generate an early stop codon 11 codons downstream of the deletion site. It is unclear whether this N-terminal fragment would be expressed at any appreciable level, as it is possible that the generation of an early stop codon would trigger nonsense-mediated decay. However, if expressed, it would be fascinating to examine whether this naturally occuring N-terminal truncation retains activity as a co-repressor. Given the observation that the TPL N-terminal domain is sufficient to bind histones (28630893), and repress transcription via synthetic assays in yeast (24979769), it is possible that this represents a semi-functional TPL variant. There has been no evidence for priotization of individual TPL/TPR family members in the auxin response pathway, but experiments modulating the gene copy number of TPR1 revealed the importance of maintaing the proper dosage of TPL/TPR in order to accurately trigger the salicyclic acid innate immune response (20647385). These results, taken together with the absence of TPL/TPR loss-of-function alleles in the Arabidopsis accessions suggest that perhaps the entire family is being conserved at the nucleotide level in the population to meet a gene dosage requirement, with little opportunity for neo- or sub-functionalization.


## ARF genes  

### Class A ARFs
Amy

### Class B ARFs
Andrew

### Class C ARFs
Hardik






