---
title: "A comprehensive analysis of natural sequence variation within the <i>Arabidopsis thaliana</i> nuclear auxin signaling pathway"
author: 
  affiliation: University of Washington
  name: "R. Clay Wright"

date: "`r Sys.Date()`"

output: 
    html_document:
      toc: true
      fig_caption: yes

keywords: r markdown
fontsize: 11pt
fontfamily: Helvetica

bibliography: ["natural_variation.bib", "TIR1_AFB.bib", "ARF_C.bib", "AuxIAA_citations.bib"]

csl: apa.csl
link-citations: true
---

```{r, echo = FALSE, message=FALSE, warning=FALSE}
knitr::opts_chunk$set(warning = FALSE, collapse = TRUE, comment = "#>", tidy = TRUE, out.width = "90%", fig.height=4, fig.width=7, out.extra='style="margin: auto; display: block; padding-top: 15px;"')
library(r1001genomes)
library(DT)
library(dplyr)

```
<!--This is a CSS (Cascading Style Sheet).
 It controls how panels and images are formatted.></!-->
<style>
  img {
    display: block;
    margin: auto;
    width: 90%;
    height: auto;
  }
</style>

# List of working titles  
* A comprehensive analysis of natural sequence variation within the <i>Arabidopsis thaliana</i> nuclear auxin signaling pathway  
* *Add your suggestions* 

# Abstract  

* Motivating, high level statement  
* Statement defining the cutting edge of your research area  
* A statement of the form: "However, it remains unclear ..." that defines what problem you are  
addressing.  
* A statement of the form:"Here, we ..."  that concisely states your contribution.  
* A series of 3?5 statement about your results.  
* A concluding statement.  


    
# Related work  
What and how have we, as a field, learned from natural variation?  
[@hollister_genomic_2014; @bielawski_detecting_2013; @kryazhimskiy_population_2008; @schmid_multilocus_2005; @nieduszynski_sequence_2011; @melamed_combining_2015; @weigel_population_2015; @schmitz_patterns_2013; @weigel_natural_2005; @hardtke_case_2010; @koornneef_naturally_2004; @mitchell-olds_genetic_2006; @joly-lopez_developing_2016; @wright_insights_2017a; @park_distributions_2017; @_135_2016a]  
 
# Methods

# Results  
## TIR1/AFB genes  
Romi
[@gray_identification_1999; @kepinski_auxininduced_2004; @tan_mechanism_2007; @dezfulian_oligomerization_2016; @wu_evolution_2017b]
1) Plant biologist of all types should use this web tool!

>Makes it really easy to look at the evolution of genes in arabidopsis.
>Makes it really easy to look at how families of genes are evolving.
>Easy way to find accessions that might be interesting to you.

2) This web tool can be used to identify domains or residues of potential functional importance in any gene or gene family. (E.g. stretches of conservation in regions of unknown function).

>I was able to find that domains of iaa and phytate interaction are highly conserved, which is what we would expect.
>I was able to show that extra flanking domains of AFB4 and 5 are very mutated and therefore probably not important.

3) It can help identify sites of relevant functional divergence between sister pairs and more broadly within gene families (E.g. one sister has much higher diversity than the other. What are the shared regions of variation? What regions are highly variable in the more diverse sister?)

>AFB4 the most diverse sister. All regions highly variable. Iaa binding mutated. likely not functional at all. All have higher variation before fbox binding domain and before dimerization domain.

4) It can also identify accessions containing multiple nonsynonymous genes in a family or network that can be used for studying network evolution. (E.g. there is a nonsynonymous polymorphism in a key functional residue of a gene that you expect will alter its function, however there are many other polymorphisms in other genes in the pathway that may or may not compensate for this change. We can assess this by backcrossing this accession with Col-0 or other distantly related accessions and resequencing phenotypic segregants). 

>Did not find this yet. None seem to be in a critical site, so I will be looking later on.
## Aux/IAA genes  


###Introduction
The Aux/IAA proteins (IAAs) mediate both auxin perception and transcriptional co-repression of auxin responsive genes. Auxin perception is carried out along with TIR1/AFB proteins, which together act as a coreceptor to perceive intracellular auxin. Interaction of IAAs with auxin and TIR1/AFBs is conferred by the highly conserved 13 amino acid degron (DII) motif [@tan_mechanism_2007]. Stabilizing mutations within the degron can lead to dominant phenotypes [@berleth_auxin_2004; @yang_iaa1_2004]. Degradation of IAAs relieves their repression upon ARF transcription factors. IAA degradation rate is influenced both by the degron sequence and by rate motifs found outside the degron, such as the KR motif [@moss_rate_2015; @dreher_arabidopsis_2006]. IAA-mediated repression of ARFs depends upon an EAR domain that recruits co-repressors of the TOPLESS (TPL) family, and the PB1 domain that enables interaction with ARF PB1 domains [@nanao_structural_2014; @kagale_ear_2011]. The Arabidopsis Aux/IAA gene family has 29 members, most of which possess a canonical degron. Notable exceptions are the Auxin insensitive IAA20 and IAA30 that lack a degron sequence entirely, while the closely related IAA31 retains a semi-functional degron sequence with a very slow degradation rate [@dreher_arabidopsis_2006; @havens_synthetic_2012]. Using the Natural Variation Webtool, we found that for the canonical Aux/IAAs, no missense mutations were present in the KR motif or PB1 interaction faces, and very few missense mutations were observed in the EAR domain and degron (Supp. Fig. Alignment of canonical IAAs). However, as discussed below, there were several interesting missense mutations found in EAR domains and degrons. The majority of nonsynonymous mutations observed were in the variable regions between the conserved domains (Supp. Fig. Alignment of canonical IAAs).

###Sister Pair Diversification
Previous research has pointed to genetic redundancy within closely related pairs or families of IAA proteins [@overvoorde_functional_2005]. For example, the iaa8-1, iaa9-1 double mutant and the iaa5-1, iaa6-1, iaa19-1 triple mutant have wild-type phenotypes [@overvoorde_functional_2005]. Despite this, most Arabidopsis IAA sister pairs have one member exhibiting higher diversity while the other sister is more conserved (Figure IAA phylogeny heat map). For example, despite having only minor differences in sequence (61.4% sequence identity), the IAA6/19 sister pair has significant differences in expression patterns, protein abundances and functions, suggesting they have undergone functional specialization since their divergence [@winkler_variation_2017]. IAA19 was expressed at higher levels than IAA6 and showed less sequence divergence; IAA6 showed regions of increased divergence upstream of the degron and in the N-terminal half of the PB1 domain [@winkler_variation_2017]. Consistent with these results, the Natural Variation Webtool (Figure IAA phylogeny heat map) revealed a lower natural variation for IAA19 (ᴨN/ᴨS = 0.55) compared to IAA6 (ᴨN/ᴨS = 2.3), and also detected high natural variation diversity within the same regions of IAA6 as seen in Winkler et al. (Supp Fig IAA6 diversity plot). Within IAA Group B, there are four sister pairs--IAA10 and IAA11, IAA12/BDL and IAA13, IAA18 and IAA26/PAP1, and IAA20 and IAA30 (Fig X) [@remington_contrasting_2004]. Interestingly, among these sister pairs only IAA12 and 13 have noticeable differences in diversity. IAA10 and IAA11 show little difference in ℼN/ℼS values, with ℼN/ℼS of 0.80 and 0.67 respectively, implicating the possibility that IAA10 and 11 play redundant roles. In rice however, it has been shown that IAA11 gain of function mutants block lateral root initiation [@zhu_gain--function_2012]. The Arabidopsis EFP browser shows that IAA10 and 11 have almost identical expression patterns, with the exception of IAA10 being more highly expressed in the developing seed, and IAA11 being more highly expressed in root tissue during lateral root development. As discussed below, the Natural Variation Webtool provides a potentially expedited route to conducting experiments meant to elucidate the extent of redundancy between family members such as IAA10 and 11.
	
###Identification of new Loss-of-Function/Gain-of-Function mutations
To test the extent of redundancy/specialization between sister pairs, the Natural Variation Webtool allows for identifications of accessions that can be potentially used to investigate the phenotypic effects in loss of function variants. For example, it has been shown that IAA10 has a missense mutation in the EAR domain, which converts the second leucine in the LxLxL motif to a serine (L49S), a mutation likely to reduce the ability of the IAA to recruit TOPLESS co-repressor [@lee_biological_2016]. This mutation is found in three accessions in the southern UK. A second potentially stronger mutation, represented in one accession from northeastern Spain is a missense resulting in an early stop codon in the PB1 domain (need Supp. Fig?). Such a mutation is likely to prevent IAA10 from interacting with an ARF, rendering it impotent. These alleles would be strong candidates for loss-of-function mutants, allowing researchers to discover the function of IAA10. In particular, it would make possible a straightforward path for creating an iaa10/11 double mutant using CRISPR/Cas9 directed mutagenesis, allowing for examination of the extent of redundancy between these family members. 

Several examples of possible gain-of-function degron missense mutations were found in IAA2, IAA3, IAA6, IAA7, IAA8, IAA13, IAA15, IAA18, IAA19, IAA27, IAA28, IAA29. Most degron mutations were found in one or two accessions represented in the database (Supplemental Table or Map?). Interestingly, almost all of the IAAs found to have accessions with degron missense mutations have a closely related sister, which may indicate a redundant function within the sister-pairs. IAA6 natural variation has two potentially dominant missense mutations found within the degron (Supp Fig. alignment), one previously believed to lead to the shy1-1 phenotype [@kim_two_1996]. However, this variant is widely represented in 31 accessions across europe (Supplemental Map Fig). This finding supports the proposition that iaa6[C78R] may not be the mutation causing the shy1-1 phenotype but rather represents natural variation in the degron sequence ([note from Mike Prigge in TAIR](https://www.arabidopsis.org/servlets/TairObject?type=notepad&id=11247)). These accessions could be used to confirm whether the shy1-1 phenotype exists within these plants. Additionally, an accession was identified with a V77M mutation in the IAA6 degron that may be useful to explore potential iaa6 gain-of-function phenotypes.

###Identifying novel functional sequences in IAAs
An exciting possible use for the Natural Variation Webtool is to search for conserved, low-diversity sites that may represent possible sites of post-translational modification or protein-protein interaction. For example, recent detailed biochemical analysis identified putative lysine ubiquitination sites in members of the IAA6/19 sister pairs [@winkler_variation_2017]; Supp Fig IAA6/19 alignment). These sites show no natural sequence variation in Arabidopsis, while several other lysines are represented in the natural sequence diversity (Supp Figs alignment and map). In IAA6, there are 17 lysines of which 6 are putative ubiquitination sites, one is in the KR motif, while two others show natural variation. In IAA19, there are 15 total lysines, with 8 as putative ubiquitination sites, 2 in the KR motif and degron, and 1 that exhibits natural variation. Natural variation sequence diversity may be used to narrow down a subset of lysines to assess for functional ubiquitination.

The members of the IAA8/9/27 group have a unique N-terminal extension with a highly conserved region common to orthologs in several other species [@dreher_arabidopsis_2006]. IAA8/9 have  redundant roles in leaf serration [@koenig_auxin_2009], while IAA8 important in lateral roots [@arase_iaa8_2012]. IAA8 has been reported to uniquely localize in the cytosol [@arase_iaa8_2012] where it may interact with LSD1 protein [@coll_programmed_2011]. The N-terminal extension has a conserved region shared between IAA8 and IAA9 with low natural sequence diversity (Supp Fig IAA8/9 alignment) that may represent a site of novel functionality, possibly mediating interaction with LSD1. There are a few accessions with missense alleles in this region in both IAA8 and IAA9 that may be valuable in probing the functionality of this region (Supplemental Fig IAA8/9 map).

Members of the IAA17/7/14/16 group have a second EAR motif (LxLxL) with no observed missense mutations in IAA7/16/17 (Supp Fig IAA7/14/16/17 alignment). This observation supports the experimental evidence for IAA7 that the second EAR motif is functional: it can interact with TPR1 corepressor and has a minor repressive role in auxin-related developmental phenotypes [lee_biological_2016]. For IAA14, we did observe two missense mutations that may be of interest to probe functionality of the IAA14 EAR motif: one in the central “L” (L41F) that is represented in only one accession, and the other in the second “x” (K40N) that is found in 12 accessions around Copenhagen (Supp Fig. IAA14 map). 
## TPL/TPR genes  
Alex

## ARF genes  

Auxin response is mediated by the auxin responsive transcription factors, or ARFs. There are 23 ARFs in Arabidopsis that are divided into three phylogenetic classes. Class A ARFs, ARF5, ARF6, ARF7, ARF8, and ARF19, activate transcription. These ARFs have a Q-rich region in the middle of the protein that may mediate activation (Guilefoyle, 2007). It has recently been shown that the middle region of ARF5 interacts with the SWI/SNF chromatin remodeling ATPases BRAMA and SPLAYED, possibly to reduce nucleosome occupancy and allow for the recruitment of transcription machinery (Wu, 2016). Additionally ARF7 can interact with Mediator subunits, directly tethering transcriptional proteins to its binding regions (Ito, 2016). Class B and C ARFs are historically categorized as repressor ARFs, though the mechanism through which they confer repression has not been identified. Their middle regions tend to be proline- and serine-rich. 

ARFs are comprised of three major domains, recent structures of which have informed structure-function analysis of the ARFs (Boer, 2014; Nanao, 2014; Korasick, 2014). These domains are conserved throughout land plants. ARFs share an N-terminal B3 DNA binding domain. Flanking this DNA-binding domain is a dimerization domain, which folds up into a single “taco-shaped” domain to allow for dimerization between ARFs. There is an auxillary domain that immediately follows and interacts with the dimerization domain. The middle region is the most variable between ARFs, as mentioned above, but is characterized by repetitive units of glutamine (class A), serine, or proline residues (classes B and C). On the C-terminal end of the protein a protein-protein interaction domain mediates interactions among ARFs, between ARFs and other transcription factors, and between ARFs and the Aux/IAA repressors. This interaction domain was recently characterized as a Phox and Bem1 (PB1) domain, which is comprised of a positive and negative face with conserved basic and acidic residues, respectively. Theoretically the dipolar nature of the PB1 domain can mediate multimerization by the pairwise interaction of these faces on different proteins, and indeed the ARF7 PB1 domain was crystallized as a multimer. It is unclear whether this multimerization occurs or plays a developmental significant role in vivo. Interfering with ARF dimerization in either the DNA-binding proximal dimerization domain or the PB1 domain decreases the ability of class A ARFs to activate transcription in a heterologous yeast system (Pierre-Jerome, 2016). 

While domain architecture is broadly conserved among the ARFs, there are exceptional cases. Three ARFs do not contain a PB1 domain at all, ARF3, ARF13, and ARF17, and several more have lost the conserved acidic or basic residues in the PB1 domain, suggesting they may be reduced to a single interaction domain. Several ARFs additionally have an expanded conserved region within the DNA-binding domain, of unknown function. The majority of domain variation among ARFs occurs in the large B-class subfamily. The liverwort Marcantia polymorpha has a single representative ARF of each class (Flores-Sandoval, 2015). The much expanded classes of the flowering plants arise from both whole genome and tandem duplication events (Remington, 2004), and may have allowed for the expansion of the quantity and complexity of loci regulated by the ARFs and a consequent expansion in their regulation of developmental processes. Here we examine natural variation in the three ARF classes and how this variation may be informative of conservation and function among the ARFs. 

### Class A ARFs
Amy

Class A ARFs are the most well-studied ARF subfamily--the five family members all act as transcriptional activators and have well-characterized, distinct developmental targets. Class A ARFs all contain the canonical B3 DNA-binding domain and PB1 interaction domain, and all have glutamine-rich middle regions (Guilefoyle, 2007). Analysis of class A ARF nucleotide diversity showed that the majority of these ARFs had low nucleotide diveristy scores, all ${\pi_N/\pi_S}$ values substantially lower than 1 with the exception of ARF19, with a relatively high ${\pi_N/\pi_S}$ value of 1.8. Comparing diversity within sister pairs, the two class A ARF sister pairs had divergent diversity scores, with ARF19 and ARF8 having scores at least triple those of ARF7 and ARF6 respectively. This may suggest that ARF6 and ARF7 have more conserved function whereas ARF19 and ARF8 are more diverse. Overall the diversity of class A ARFs was generally low, especially compared to the class B and C ARFs, suggesting that class A ARFs are central to auxin signal transduction and plant development.

For all class A ARFs, the middle region of the protein was the predominant high diversity region. The DNA-binding domain had very few, low-diversity missense mutations, as did the PB1 domain. Considering the necessity of their conserved functions, the low level of variation in these key DNA and protein-protein interaction domains is expected. 

Within the middle region, ARF7 had several expansions of polyglutamine sequences compared to the other class A ARFs. Polyglutamine regions are known to readily expand and contract throughout evolutionary time due to replication error, and variation in polyQ length can be acted on by natural selection and have phenotypic consequences (Press, 2014). 

A few missense variations in ARF5 highlighted potential functional variation among ecotypes. Two high-diversity changes at the positive face of the PB1 domain, very close to the conserved lysine residue, to phenylalanine were interstingly always found together in all ecotypes in which they occured: Tu-B2-3,Ru-2, Haes-1, Rd-0, 627ME-13Y1, Le-0,Bu-0, Bsch-0. These mutations to a rather bulky side chain may change the folding of the PB1 domain and interfere with interactions on the positive face. A high diversity polymorphism that results in change in the 17th amino acid of ARF5 from a leucine to a methionine is also intriguing. ARF5 has an N-terminal extension of unknown function relative to the other class A ARFs *how many amino acids*. It is possible that this methionine acts as an alternative start site in the accessions in which it is found yielding a truncated ARF5 variant without this N-terminal extension. 


### Class B ARFs
Andrew

ARF23, which is truncated in its DNA-binding region, has a high ${\pi_N/\pi_S}$ value of 4.1, and has many high-diversity valued nonsense variants, as may be expected of a gene that is pseudogenizing. 

ARF3, ARF13, and ARF17 all lack a PB1 C-terminal domain, which suggests they may be unable to interact with Aux/IAA repressors and may act as an auxin-insensitive mechanism of adding competitive inhibition to ARF-mediated transcription. These ARFs span the range of ${\pi_N/\pi_S}$ ratios relative to the other ARFs, ranging from 0.85 to 4.1. ARF13, with the highest ${\pi_N/\pi_S}$ of 4.1 has many nonsense variants of high diversity, perhaps suggesting it is pseudogenizing. 

ARF14, ARF15, ARF16, ARF20, and ARF21 all do not have one or multiple of the conserved acidic residues in the PB1 domain, and themselves form a subclade of the B ARFs. Interestingly, no non-synonymous variation is seen in these amino acid residues of these ARFs, suggesting that the amino acids that are in the regions may still be important for PB1 function. 






### Class C ARFs
Hardik (All figure cite below are in folder "/Dropbox/natural-variation-webtool/manuscript_ARF_C/FIGURES")

Class C ARFs shows low nucleotide diveristy scores, all pi_N/pi_S values substantially lower than 1.
Comparing diversity within coding region of class C ARF, ARF16  has least diversity whereas its family members (ARF10, ARF17) shows  scores at least four times higher [@remington_contrasting_2004].
Structurally, all three members of Class C ARFs contain both canonical B3 DNA-binding domain and whereas only ARF10 and ARF16 contains PB1 interaction domain. DNA binding domain exhibit overall low diversity and presence of largely synonymous changes for all the ARFs. Only ARF10, shows mis-sense variants (p.Ser147Thr/c.439T>A  and p.Ala157Ser/c.469G>T) DNA binding domains with higher diversity score for conservative variation (codon 147) compared to non conservation variation (codon 157) (Fig. ARF10_DBD Variants distribution). Of the PB1 domain containing class C ARFs, ARF16 exhibits several naturally occurring non-synonymous variants in contrast to ARF10 (Fig.  Arf16_PB_variants distribution) and is sporadically distributed in location.

Few members of class B ARFs (ARF 2,3,4) and C ARFs (ARF10,16, 17) distinguish from other ARFs on the basis of its ability to be post transcriptionally regulated by trans-acting small interfering RNA (tasiRNA) and microRNA (miRNA) [@rhoades_prediction_2002; @allen_micrornadirected_2005].  (@AMY Arf 6 and 8 targeted by miRNA 167 - check for conservation).  Interestingly, miRNA binding motif of the class C ARFs show high degree of conservation across the natural variation of 1001 genomes queried here (Fig. ARF10_16_17_Alignment). This is consistent with perturbation studies that has shown drastic developmental phenotype for class C ARFs [@wang_control_2005; @mallory_micrornadirected_2005; @liu_repression_2007]. For the class B ARFs (2,3,4) tasiRNA binding motif is highly conserved and show no variation suggesting strong regulatory constrains at the locus (Fig ARF2_3_4_Alignment). DNA binding domain of ARF4 exhibits non-synonymous variation in 4 accession and would be interesting to carry out phenotypic characterize them for developmental growth defects (Fig ARF4_DBD Variants distribution).






