---
title: "A comprehensive analysis of natural sequence variation within the <i>Arabidopsis thaliana</i> nuclear auxin signaling pathway"
author: 
  affiliation: University of Washington
  name: "R. Clay Wright"

date: "`r Sys.Date()`"

output: 
    html_document:
      toc: true
      fig_caption: yes

keywords: r markdown
fontsize: 11pt
fontfamily: Helvetica

bibliography: ["natural_variation.bib", "TIR1_AFB.bib", "AuxIAA_citations.bib"]
csl: apa.csl
link-citations: true
---

```{r, echo = FALSE, message=FALSE, warning=FALSE}
knitr::opts_chunk$set(warning = FALSE, collapse = TRUE, comment = "#>", tidy = TRUE, out.width = "90%", fig.height=4, fig.width=7, out.extra='style="margin: auto; display: block; padding-top: 15px;"')
library(r1001genomes)
library(DT)
library(dplyr)

```
<!--This is a CSS (Cascading Style Sheet).
 It controls how panels and images are formatted.></!-->
<style>
  img {
    display: block;
    margin: auto;
    width: 90%;
    height: auto;
  }
</style>

# List of working titles  
* A comprehensive analysis of natural sequence variation within the <i>Arabidopsis thaliana</i> nuclear auxin signaling pathway  
* *Add your suggestions* 

# Abstract  

* Motivating, high level statement  
* Statement defining the cutting edge of your research area  
* A statement of the form: "However, it remains unclear ..." that defines what problem you are  
addressing.  
* A statement of the form:"Here, we ..."  that concisely states your contribution.  
* A series of 3?5 statement about your results.  
* A concluding statement.  


    
# Related work  
What and how have we, as a field, learned from natural variation?  
[@hollister_genomic_2014; @bielawski_detecting_2013; @kryazhimskiy_population_2008; @schmid_multilocus_2005; @nieduszynski_sequence_2011; @melamed_combining_2015; @weigel_population_2015; @schmitz_patterns_2013; @weigel_natural_2005; @hardtke_case_2010; @koornneef_naturally_2004; @mitchell-olds_genetic_2006; @joly-lopez_developing_2016; @wright_insights_2017a; @park_distributions_2017; @_135_2016a]  
 
# Methods

# Results  
## TIR1/AFB genes  
Romi
[@gray_identification_1999; @kepinski_auxininduced_2004; @tan_mechanism_2007; @dezfulian_oligomerization_2016; @wu_evolution_2017b]

## Aux/IAA genes

###Introduction
The Aux/IAA proteins (IAAs) mediate both auxin perception and transcriptional co-repression of auxin responsive genes. Auxin perception is carried out along with TIR1/AFB proteins, which together act as a coreceptor to perceive intracellular auxin. Interaction of IAAs with auxin and TIR1/AFBs is conferred by the highly conserved 13 amino acid degron (DII) motif [@tan_mechanism_2007]. Stabilizing mutations within the degron can lead to dominant phenotypes [@berleth_auxin_2004; @yang_iaa1_2004]. Degradation of IAAs relieves their repression upon ARF transcription factors. IAA degradation rate is influenced both by the degron sequence and by rate motifs found outside the degron, such as the KR motif [@moss_rate_2015; @dreher_arabidopsis_2006]. IAA-mediated repression of ARFs depends upon an EAR domain that recruits co-repressors of the TOPLESS (TPL) family, and the PB1 domain that enables interaction with ARF PB1 domains [@nanao_structural_2014; @kagale_ear_2011]. The Arabidopsis Aux/IAA gene family has 29 members, most of which possess a canonical degron. Notable exceptions are the Auxin insensitive IAA20 and IAA30 that lack a degron sequence entirely, while the closely related IAA31 retains a semi-functional degron sequence with a very slow degradation rate [@dreher_arabidopsis_2006; @havens_synthetic_2012]. Using the Natural Variation Webtool, we found that for the canonical Aux/IAAs, no missense mutations were present in the KR motif or PB1 interaction faces, and very few missense mutations were observed in the EAR domain and degron (Supp. Fig. Alignment of canonical IAAs). However, as discussed below, there were several interesting missense mutations found in EAR domains and degrons. The majority of nonsynonymous mutations observed were in the variable regions between the conserved domains (Supp. Fig. Alignment of canonical IAAs).

###Sister Pair Diversification
Previous research has pointed to genetic redundancy within closely related pairs or families of IAA proteins [@overvoorde_functional_2005]. For example, the iaa8-1, iaa9-1 double mutant and the iaa5-1, iaa6-1, iaa19-1 triple mutant have wild-type phenotypes [@overvoorde_functional_2005]. Despite this, most Arabidopsis IAA sister pairs have one member exhibiting higher diversity while the other sister is more conserved (Figure IAA phylogeny heat map). For example, despite having only minor differences in sequence (61.4% sequence identity), the IAA6/19 sister pair has significant differences in expression patterns, protein abundances and functions, suggesting they have undergone functional specialization since their divergence [@winkler_variation_2017]. IAA19 was expressed at higher levels than IAA6 and showed less sequence divergence; IAA6 showed regions of increased divergence upstream of the degron and in the N-terminal half of the PB1 domain [@winkler_variation_2017]. Consistent with these results, the Natural Variation Webtool (Figure IAA phylogeny heat map) revealed a lower natural variation for IAA19 (ᴨN/ᴨS = 0.55) compared to IAA6 (ᴨN/ᴨS = 2.3), and also detected high natural variation diversity within the same regions of IAA6 as seen in Winkler et al. (Supp Fig IAA6 diversity plot). Within IAA Group B, there are four sister pairs--IAA10 and IAA11, IAA12/BDL and IAA13, IAA18 and IAA26/PAP1, and IAA20 and IAA30 (Fig X) [@remington_contrasting_2004]. Interestingly, among these sister pairs only IAA12 and 13 have noticeable differences in diversity. IAA10 and IAA11 show little difference in ℼN/ℼS values, with ℼN/ℼS of 0.80 and 0.67 respectively, implicating the possibility that IAA10 and 11 play redundant roles. In rice however, it has been shown that IAA11 gain of function mutants block lateral root initiation [@zhu_gain--function_2012]. The Arabidopsis EFP browser shows that IAA10 and 11 have almost identical expression patterns, with the exception of IAA10 being more highly expressed in the developing seed, and IAA11 being more highly expressed in root tissue during lateral root development. As discussed below, the Natural Variation Webtool provides a potentially expedited route to conducting experiments meant to elucidate the extent of redundancy between family members such as IAA10 and 11.
	
###Identification of new Loss-of-Function/Gain-of-Function mutations
To test the extent of redundancy/specialization between sister pairs, the Natural Variation Webtool allows for identifications of accessions that can be potentially used to investigate the phenotypic effects in loss of function variants. For example, it has been shown that IAA10 has a missense mutation in the EAR domain, which converts the second leucine in the LxLxL motif to a serine (L49S), a mutation likely to reduce the ability of the IAA to recruit TOPLESS co-repressor [@lee_biological_2016]. This mutation is found in three accessions in the southern UK. A second potentially stronger mutation, represented in one accession from northeastern Spain is a missense resulting in an early stop codon in the PB1 domain (need Supp. Fig?). Such a mutation is likely to prevent IAA10 from interacting with an ARF, rendering it impotent. These alleles would be strong candidates for loss-of-function mutants, allowing researchers to discover the function of IAA10. In particular, it would make possible a straightforward path for creating an iaa10/11 double mutant using CRISPR/Cas9 directed mutagenesis, allowing for examination of the extent of redundancy between these family members. 

Several examples of possible gain-of-function degron missense mutations were found in IAA2, IAA3, IAA6, IAA7, IAA8, IAA13, IAA15, IAA18, IAA19, IAA27, IAA28, IAA29. Most degron mutations were found in one or two accessions represented in the database (Supplemental Table or Map?). Interestingly, almost all of the IAAs found to have accessions with degron missense mutations have a closely related sister, which may indicate a redundant function within the sister-pairs. IAA6 natural variation has two potentially dominant missense mutations found within the degron (Supp Fig. alignment), one previously believed to lead to the shy1-1 phenotype [@kim_two_1996]. However, this variant is widely represented in 31 accessions across europe (Supplemental Map Fig). This finding supports the proposition that iaa6[C78R] may not be the mutation causing the shy1-1 phenotype but rather represents natural variation in the degron sequence ([note from Mike Prigge in TAIR](https://www.arabidopsis.org/servlets/TairObject?type=notepad&id=11247)). These accessions could be used to confirm whether the shy1-1 phenotype exists within these plants. Additionally, an accession was identified with a V77M mutation in the IAA6 degron that may be useful to explore potential iaa6 gain-of-function phenotypes.

###Identifying novel functional sequences in IAAs
An exciting possible use for the Natural Variation Webtool is to search for conserved, low-diversity sites that may represent possible sites of post-translational modification or protein-protein interaction. For example, recent detailed biochemical analysis identified putative lysine ubiquitination sites in members of the IAA6/19 sister pairs [@winkler_variation_2017]; Supp Fig IAA6/19 alignment). These sites show no natural sequence variation in Arabidopsis, while several other lysines are represented in the natural sequence diversity (Supp Figs alignment and map). In IAA6, there are 17 lysines of which 6 are putative ubiquitination sites, one is in the KR motif, while two others show natural variation. In IAA19, there are 15 total lysines, with 8 as putative ubiquitination sites, 2 in the KR motif and degron, and 1 that exhibits natural variation. Natural variation sequence diversity may be used to narrow down a subset of lysines to assess for functional ubiquitination.

The members of the IAA8/9/27 group have a unique N-terminal extension with a highly conserved region common to orthologs in several other species [@dreher_arabidopsis_2006]. IAA8/9 have  redundant roles in leaf serration [@koenig_auxin_2009], while IAA8 important in lateral roots [@arase_iaa8_2012]. IAA8 has been reported to uniquely localize in the cytosol [@arase_iaa8_2012] where it may interact with LSD1 protein [@coll_programmed_2011]. The N-terminal extension has a conserved region shared between IAA8 and IAA9 with low natural sequence diversity (Supp Fig IAA8/9 alignment) that may represent a site of novel functionality, possibly mediating interaction with LSD1. There are a few accessions with missense alleles in this region in both IAA8 and IAA9 that may be valuable in probing the functionality of this region (Supplemental Fig IAA8/9 map).

Members of the IAA17/7/14/16 group have a second EAR motif (LxLxL) with no observed missense mutations in IAA7/16/17 (Supp Fig IAA7/14/16/17 alignment). This observation supports the experimental evidence for IAA7 that the second EAR motif is functional: it can interact with TPR1 corepressor and has a minor repressive role in auxin-related developmental phenotypes [lee_biological_2016]. For IAA14, we did observe two missense mutations that may be of interest to probe functionality of the IAA14 EAR motif: one in the central “L” (L41F) that is represented in only one accession, and the other in the second “x” (K40N) that is found in 12 accessions around Copenhagen (Supp Fig. IAA14 map). 
## TPL/TPR genes  
Alex

## ARF genes  

### Class A ARFs
Amy

### Class B ARFs
Andrew

### Class C ARFs
Hardik






