<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Predicting structure–function relationships using natural genetic variation</title>
  <meta name="description" content="Predicting structure–function relationships using natural genetic variation">
  <meta name="generator" content="bookdown <!--bookdown:version--> and GitBook 2.6.7">

  <meta property="og:title" content="Predicting structure–function relationships using natural genetic variation" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Predicting structure–function relationships using natural genetic variation" />
  
  
  



<meta name="date" content="2018-10-18">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<!--bookdown:link_prev-->
<!--bookdown:link_next-->
<style type="text/css">
p.abstract{
  text-align: center;
  font-weight: bold;
}
div.abstract{
  margin: auto;
  width: 90%;
}
</style>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/htmlwidgets-1.3/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.4/datatables.js"></script>
<link href="libs/dt-core-1.10.16/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.16/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.16/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.0.0/js/crosstalk.min.js"></script>



</head>

<body>



<!--bookdown:title:start-->
<div id="header">
<h1 class="title">Predicting structure–function relationships using natural genetic variation</h1>
<h4 class="author"><em><div style="white-space: pre-line;">R. Clay Wright<a href="#fn1" class="footnoteRef" id="fnref1"><sup>1</sup></a> <span class="math inline">\(^1\)</span>,
Morgan O. Hamm <span class="math inline">\(^2\)</span>,
Jennifer Nemhauser <span class="math inline">\(^2\)</span>
<span class="math inline">\(^1\)</span>Biological Systems Engineering, Virginia Tech, Blacksburg, VA 24061,
<span class="math inline">\(^2\)</span>Deparment of Biology, University of Washington, Seattle, WA 98195</div></em></h4>
<h4 class="date"><em>2018-10-18</em></h4>
<div class="abstract">
<p class="abstract">Abstract</p>
<p>The way genes and genomes have evolved can provide information about which genes are critical to an organisms fitness, which genes faciliate adaptation, and how robust genes and gene networks are to perturbation. Thousands of genomes have now been sequenced representing a large swath of plant species and genetic variation within many species which has only begun to be explored in the last few years. However, much of this data remains inaccessible to researchers without significant bioinformatics experience. Here, we have developed a webtool ViVa (Visualization of Variation) to facilitate analysis, by any researcher, of The 1001 <em>Arabidopsis</em> Genomes Project dataset on a gene, gene family, or gene network level. We We have demonstrated the utility and accessibility of this tool by analyzing the natural variation in the well-studied nuclear auxin signaling pathway as a group of researchers from an array of experience levels. Our analysis has provided further confirmation of existing knowledge of this well studied pathway and has also helped generate new hypotheses. These results suggest there is much to be learned about less studied gene families and networks from similar analysis, especially when paired with biochemical and genetic characterization. This application provides access for every researcher allowing the community to extract maximum value from this large and expensive-to-generate dataset. ViVa is also extensible to any other database of genetic variation such as The 3,000 Rice Genomes Project.</p>
</div>
</div>
<!--bookdown:title:end-->

<!--bookdown:toc:start-->
  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">
<!--bookdown:toc2:start-->
<ul>
<li><a href="#list-of-working-titles"><span class="toc-section-number">0.1</span> List of working titles</a></li>
<li><a href="#intro"><span class="toc-section-number">1</span> Introduction</a></li>
<li><a href="#methods"><span class="toc-section-number">2</span> Methods</a><ul>
<li><a href="#data-sources"><span class="toc-section-number">2.1</span> Data Sources</a><ul>
<li><a href="#variant-data"><span class="toc-section-number">2.1.1</span> Variant Data</a></li>
<li><a href="#accession-information"><span class="toc-section-number">2.1.2</span> Accession Information</a></li>
<li><a href="#gene-and-transcript-information"><span class="toc-section-number">2.1.3</span> Gene and transcript information</a></li>
</ul></li>
<li><a href="#nucleotide-diversity-calculation"><span class="toc-section-number">2.2</span> Nucleotide Diversity Calculation</a></li>
<li><a href="#detailed-pi_k-calculaiton-simplification."><span class="toc-section-number">2.3</span> Detailed <span class="math inline">\(\pi_k\)</span> calculaiton simplification.</a></li>
<li><a href="#software"><span class="toc-section-number">2.4</span> Software</a></li>
</ul></li>
<li><a href="#AFBs"><span class="toc-section-number">3</span> <em>TIR1/AFB</em> genes</a></li>
<li><a href="#IAAs"><span class="toc-section-number">4</span> <em>Aux/IAA</em> genes</a><ul>
<li><a href="#sister-pair-diversification"><span class="toc-section-number">4.0.1</span> Sister Pair Diversification</a></li>
<li><a href="#new-loss-of-functiongain-of-function-alleles"><span class="toc-section-number">4.0.2</span> New Loss-of-Function/Gain-of-Function alleles</a></li>
<li><a href="#identifying-novel-functional-sequences-in-iaas"><span class="toc-section-number">4.0.3</span> Identifying novel functional sequences in IAAs</a></li>
</ul></li>
<li><a href="#TPLs"><span class="toc-section-number">5</span> <em>TPL/TPR</em> genes</a></li>
<li><a href="#ARFs"><span class="toc-section-number">6</span> <em>ARF</em> genes</a><ul>
<li><a href="#class-a-arfs"><span class="toc-section-number">6.0.1</span> Class A <em>ARFs</em></a></li>
<li><a href="#class-b-arfs"><span class="toc-section-number">6.0.2</span> Class B <em>ARFs</em></a></li>
<li><a href="#class-c-arfs"><span class="toc-section-number">6.0.3</span> Class C <em>ARFs</em></a></li>
</ul></li>
<li><a href="#discussion"><span class="toc-section-number">7</span> Discussion</a></li>
<li><a href="#refs"><span class="toc-section-number">8</span> References</a></li>
</ul>
<!--bookdown:toc2:end-->
      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Predicting structure–function relationships using natural genetic variation</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<!--bookdown:toc:end-->
<!--bookdown:body:start-->
<div id="list-of-working-titles" class="section level2">
<h2><span class="header-section-number">0.1</span> List of working titles</h2>
<ul>
<li>Predicting structure–function relationships using natural genetic variation</li>
<li>Accelerating structure–function mapping using natural genetic variation</li>
<li>A comprehensive analysis of natural sequence variation within the <i>Arabidopsis thaliana</i> nuclear auxin signaling pathway</li>
<li><strong>Add your suggestions</strong></li>
</ul>
<!--chapter:end:index.Rmd-->
</div>
<div id="intro" class="section level1">
<h1><span class="header-section-number">1</span> Introduction</h1>
<p>The first genome sequence of <em>Arabidopsis thaliana</em> facilitated rapid advancement of plant biology through molecular genetics. Since this initial genome, massive scaling of sequencing technology has allowed a pioneering group to survey the genomic variation in natural <em>Arabidopsis thaliana</em> populations. This valuable population genetics resource has led to several associations of genetic loci with phenotypic traits and provided insights into how selective pressure has influenced the evolution of plant genomes. Outside of genome-wide studies however, this valuable resequencing dataset has seen little use.</p>
<p>Natural genomic variation also provides a set of feasible functional variation at the gene, gene family and gene network level. Characterizing this variation can provide insights into the function and evolution of genes, families and networks <span class="citation">(Delker et al. <a href="#ref-delker_natural_2010">2010</a>; R. C. Wright et al. <a href="#ref-wright_insights_2017">2017</a>)</span> . Here, we present ViVa a web application and R-package for Visualizing Variation within genes, gene families and gene networks. We apply this to analysis of the 1001 genomes dataset [@. This application allows plant molecular biologists without bioinformatics experience to use genomic variation to formulate hypotheses as to the sequence/function/phenotype relationships within their gene, family, or network of their interest.</p>
<p>This application allows easy access to<br />
1. lists of missense polymorphisms to facilitate biochemical assays of variant effects <span class="citation">(Starita et al. <a href="#ref-starita_variant_2017">2017</a>)</span>,<br />
2. family-wise alignments of variants to facilitate de novo functional domain identification <span class="citation">(Melamed et al. <a href="#ref-melamed_combining_2015">2015</a>)</span>,<br />
3. lists of accessions containing missense (or any type of) polymorphisms to facilitate segregation analysis and measurment of the distribution of variant effects on phenotype <span class="citation">(Park et al. <a href="#ref-park_distributions_2017">2017</a>)</span>,<br />
4. as well as the study of gene, gene family and network evolution <span class="citation">(Delker et al. <a href="#ref-delker_natural_2010">2010</a>; Kliebenstein <a href="#ref-kliebenstein_role_2008">2008</a>)</span>.</p>
<p>Here, we demonstrate the utility of this tool through comprehensive analysis of genomic variation in the nuclear auxin signaling pathway. We identify potential functional variation within the coding sequence of the genes within this network and formulate hypotheses regarding the functional implications of this variation and the evolution of these gene families. Our analysis provides further confirmation of much of the existing knowledge of these genes demonstrating the validity of this approach. We also identify natural alleles of several genes and accessions bearing these alleles to facilitate future characterization structure/function, genotype/phenotype, and epistatic relationships.</p>
<!--chapter:end:01-intro.Rmd-->
</div>
<div id="methods" class="section level1">
<h1><span class="header-section-number">2</span> Methods</h1>
<div id="data-sources" class="section level2">
<h2><span class="header-section-number">2.1</span> Data Sources</h2>
<div id="variant-data" class="section level3">
<h3><span class="header-section-number">2.1.1</span> Variant Data</h3>
<p>Variant data was queried from the 1001 genomes project website (<a href="http://1001genomes.org" class="uri">http://1001genomes.org</a>) via URL requests to an (undocumented) API service that provides subset files of the whole-genome VCF. The whole-genome VCF file can be found on the project’s website at <a href="http://1001genomes.org/data/GMI-MPI/releases/v3.1/" class="uri">http://1001genomes.org/data/GMI-MPI/releases/v3.1/</a></p>
</div>
<div id="accession-information" class="section level3">
<h3><span class="header-section-number">2.1.2</span> Accession Information</h3>
<p>Details on each of the 1135 accessions including CS stock numbers, and geographic location where the samples were collected were taken from the 1001 genomes website at <a href="http://1001genomes.org/accessions.html" class="uri">http://1001genomes.org/accessions.html</a>, via the download link at the bottom of the page. This data file has been embedded in the R package.</p>
</div>
<div id="gene-and-transcript-information" class="section level3">
<h3><span class="header-section-number">2.1.3</span> Gene and transcript information</h3>
<p>Information on the genes and transcripts including chromosomal coordinates, start and end location, and transcript length were taken from either the TAIR10 or Araport11 databases. The TAIR10 database, found at <a href="http://arabidopsis.org" class="uri">http://arabidopsis.org</a>, was accessed via the biomart protocol, using the R package biomaRt. The araport11 database, which can also be found on the arabidopsis.org website (<a href="https://www.arabidopsis.org/download/index-auto.jsp?dir=%2Fdownload_files%2FGenes%2FAraport11_genome_release">https://www.arabidopsis.org/download/index-auto.jsp?dir=%2Fdownload_files%2FGenes%2FAraport11_genome_release</a>), has been embedded in the R package.</p>
</div>
</div>
<div id="nucleotide-diversity-calculation" class="section level2">
<h2><span class="header-section-number">2.2</span> Nucleotide Diversity Calculation</h2>
<p>Nei defined the nucleotide diversity statistic in his original paper as: “…the average number of nucleotide differences per site between two randomly chosen DNA sequences.” (Nei, 1979) And provided the equation:</p>
<span class="math display">\[\begin{equation}
\displaystyle \pi = \sum_{ij}x_{i}x_{j}\pi_{ij}
(\#eq:nei)
\end{equation}\]</span>
<p>Where Where <span class="math inline">\(x_{i}\)</span> is the frequency of the ith sequence in the population and <span class="math inline">\(\pi_{ij}\)</span> is the number of sites that are different between the ith and jth sequence divided by sequence length.</p>
<p>a more general form, that treats each sequence in the population as unique can be written as:</p>
<span class="math display">\[\begin{equation}
\displaystyle \pi = \frac{1}{L*n^2} \sum_{i=1}^n \sum_{j=1}^n \sum_{k=1}^L{\pi_{ijk}}
\qquad
\displaystyle \pi_{ijk} = \left\{\begin{array}{l}
1 \quad \textrm{if} \quad N_{ik} \neq N_{jk} \\
0 \quad \textrm{if} \quad N_{ik} = N_{jk}
\end{array}\right.
(\#eq:nei-gen)
\end{equation}\]</span>
<p>where <span class="math inline">\(N_{ik}\)</span> is the nucleotide (A, T, C or G) at position k on the ith sequence of the population. <span class="math inline">\(L\)</span> is the length of the sequence (the same for all sequences in the population, assumes only SNPs exist, no indels). n is the total number of sequences in the population.</p>
<p>from this form we can re-arrange summations to the form below:</p>
<span class="math display">\[\begin{equation}
\displaystyle \pi = \frac{1}{L} \sum_{k=1}^L{\pi_{k}}
\qquad
\pi_{k} = \frac{1}{n^2} \sum_{i=1}^n \sum_{j=1}^n {\pi_{ijk}}
(\#eq:nei-rearrange)
\end{equation}\]</span>
<p>here <span class="math inline">\(\pi_k\)</span> can be thought of as the site-wise nucleotide diversity at position k, and is equal to the nucleotide diversity of a sequence of length 1 at location k. we can calculate <span class="math inline">\(\pi_k\)</span> for each site, then average those over the sequence length to calculate <span class="math inline">\(\pi\)</span>, the nucleotide diversity of the sequence.</p>
<p>The function <code>Nucleotide_diversity</code> in the r1001genomes package calculates <span class="math inline">\(\pi_k\)</span> for each position in the gene or region that contains a variant. note, <span class="math inline">\(\pi_k\)</span> is equal to 0 at all locations without variants. This is also what is displayed in the Diversity Plot tab of the web app.</p>
</div>
<div id="detailed-pi_k-calculaiton-simplification." class="section level2">
<h2><span class="header-section-number">2.3</span> Detailed <span class="math inline">\(\pi_k\)</span> calculaiton simplification.</h2>
<p>The formula for <span class="math inline">\(\pi_k\)</span> above requires comparing every sequence to every other sequence at location k, however, we know there are only a few variant forms at each individual location.</p>
<p>so we can revert back to using Nei’s original formula @ref(eq:nei), modifying it slightly, replacing <span class="math inline">\(x_i\)</span> with <span class="math inline">\(\frac{n_i}{n}\)</span>, <span class="math inline">\(n_i\)</span> being the number of sequences in the population of the form <span class="math inline">\(N_i\)</span> at location k.</p>
<span class="math display">\[\begin{equation}
\displaystyle \pi_k = \sum_{ij} \frac{n_i}{n}\frac{n_j}{n} \pi_{ij} = 
\frac{1}{n^2}\sum_{ij} n_i n_j \pi_{ij}\\
\pi_{ij(k)} = \left\{\begin{array}{l}
1 \quad \textrm{if} \quad i \neq j \\
0 \quad \textrm{if} \quad i = j
\end{array}\right.
(\#eq:nei-simp1)
\end{equation}\]</span>
<p>Note that in equation @ref(eq:nei) subscripts i and j are summed over all sequences in the population, however in equation @ref(eq:nei-simp1) i and j are only summed over unique variants at a particular location k.</p>
<p>we will define <span class="math inline">\(n_{!i} = n - n_i\)</span> as the number of sequences different from i at k.</p>
<p>we can see the summed term will be zero if <span class="math inline">\(i=j\)</span>, and <span class="math inline">\(n_i n_j\)</span> if <span class="math inline">\(i\neq j\)</span> so:</p>
<span class="math display">\[\begin{equation}
\displaystyle \pi_k = \frac{1}{n^2} \sum_{i} n_i n_{!i}
(\#eq:nei-simp2)
\end{equation}\]</span>
<p>next we substitute our definition of <span class="math inline">\(n_{!i}\)</span></p>
<span class="math display">\[\begin{equation}
\displaystyle \pi_k = \frac{1}{n^2} \sum_{i} n_i (n - n_{i})
(\#eq:nei-simp3)
\end{equation}\]</span>
<p>then distribute and split summation:</p>
<span class="math display">\[\begin{equation}
\displaystyle \pi_k = \frac{1}{n^2} (n \sum_{i} n_i - \sum_{i} n_{i}^2)
(\#eq:nei-simp4)
\end{equation}\]</span>
<p>finally, summing <span class="math inline">\(\displaystyle \sum_{i}n_i\)</span> is equal to n:</p>
<span class="math display">\[\begin{equation}
\displaystyle \pi_k = \frac{1}{n^2} (n^2 - \sum_{i} n_{i}^2)
(\#eq:nei-simp5)
\end{equation}\]</span>
<p>this simplified form for <span class="math inline">\(\pi_k\)</span> is used by the app, because the counts of unique variants at a single nucleotide location can easily be summarized in R.</p>
</div>
<div id="software" class="section level2">
<h2><span class="header-section-number">2.4</span> Software</h2>
<p>The r1001genomes package has many software dependencies on other R packages, a few of the key bioinformatics packages used are listed below.</p>
<p><strong>biomaRt:</strong> The biomaRt package was used for accessing the TAIR10 database on arabidopsis.org</p>
<p><strong>vcfR:</strong> vcfR was used to read in the VCF files in a flat “tidy” format for easy manipulaiton</p>
<p><strong>BSgenome:</strong> The BSgenome package BSgenome.Athaliana.TAIR.TAIR9 was used as the source for the complete DNA string of the reference genome (Col-0).</p>
<p><strong>DECIPHER:</strong> DECIPHER was used to align nucleotide and amino acid sequences of homologous genes</p>
<p><strong>GenomicFeatures:</strong> The GenomicFeatures package was used for handling sequence annotations.</p>
<p><strong>Biostrings:</strong> Biostrings provides the underlying framework for the sequence manipulations used for generating and aligning sequences with BSgenome, Decipher, and GenomicFeatures</p>
<!--chapter:end:02-methods.Rmd-->
</div>
</div>
<div id="AFBs" class="section level1">
<h1><span class="header-section-number">3</span> <em>TIR1/AFB</em> genes</h1>
<p>Auxin acts by binding to receptors (Auxin-signaling F-Boxes, or AFBs) that in turn target co-repressors (Aux/IAAs) for degradation. The six auxin receptor genes in the model plant <em>Arabidopsis thaliana</em> , <em>TIR1</em> and <em>AFB1-5</em>, evolved through gene duplication and diversification early in the history of vascular plants <span class="citation">(G. Parry et al. <a href="#ref-parry_complex_2009">2009</a>)</span>. The rate of co-repressor degradation is determined by the identity of both the receptor and co-repressor <span class="citation">(Havens et al. <a href="#ref-havens_synthetic_2012">2012</a>)</span>, and this rate sets the pace of lateral root development <span class="citation">(Guseman et al. <a href="#ref-guseman_auxininduced_2015">2015</a>)</span>.</p>
<p>All members of this family have been shown to bind auxin and Aux/IAA proteins. However, AFB1 has drastically reduced ability to assemble into an SCF complex, due to the substitution E8K in its F-box domain, preventing it from inducing degradation of Aux/IAAs<span class="citation">(H. Yu et al. <a href="#ref-yu_untethering_2015">2015</a>)</span>. This lack of complexation may allow observed high ubiquitous AFB1 accumulation <span class="citation">(G. Parry et al. <a href="#ref-parry_complex_2009">2009</a>)</span>. Higher order mutants in the family containing <em>afb1</em> mutants suggest that <em>AFB1</em> has a moderate positive effect on auxin signaling. Additionally, AFB4 and AFB5 have been shown to preferentially and functionally bind the synthetic auxin picloram, while other family members preferentially bind indole-3-acetic acid <span class="citation">(M. J. Prigge et al. <a href="#ref-prigge_arabidopsis_2016">2016</a>)</span>. Interestingly, the strength and rate with which TIR1/AFBs are able to bind and mark Aux/IAAs for degradation are variable <span class="citation">(Calderón Villalobos et al. <a href="#ref-calderonvillalobos_combinatorial_2012">2012</a>; Havens et al. <a href="#ref-havens_synthetic_2012">2012</a>)</span>. AFB2 induced the degradation of certain Aux/IAA proteins at a faster rate than TIR1, suggesting some functional specificity has arisen since the initial duplication between the <em>TIR1/AFB1</em> and <em>AFB2/AFB3</em> clades.</p>
<p>Although it is unclear what unique roles each receptor plays in growth and development, a number of studies have pointed out differences in the ways the six different receptors in A. thaliana differ in biochemical function and expression domain<span class="citation">(N. Dharmasiri et al. <a href="#ref-dharmasiri_plant_2005">2005</a>; G. Parry et al. <a href="#ref-parry_complex_2009">2009</a>; M. J. Prigge et al. <a href="#ref-prigge_arabidopsis_2016">2016</a>)</span>. <em>TIR1/AFB</em> genes are expressed ubiquitously in A. thaliana tissues, TIR1, AFB2, and AFB3 have been shown to accumulate in the shoot and root meristems and leaf tissues, with slightly different expression patterns for TIR1 <span class="citation">(G. Parry et al. <a href="#ref-parry_complex_2009">2009</a>)</span>. Additionally, the expression of AFB5 is strongly circadian-regulated <span class="citation">(Covington and Harmer <a href="#ref-covington_circadian_2007">2007</a>)</span> and AFB3 is more highly expressed in the roots in the presence of nitrate, facilitating increased lateral root formation <span class="citation">(Vidal, Álvarez, and Gutiérrez <a href="#ref-vidal_nitrate_2014">2014</a>)</span>, suggesting more broad environmental regulation of this gene family may exist.</p>
<p>All members of this family have been shown to bind auxin and Aux/IAA proteins. However, AFB1 has drastically reduced ability to assemble into an SCF complex, due to the substitution E8K in its F-box domain, preventing it from inducing degradation of Aux/IAAs<span class="citation">(H. Yu et al. <a href="#ref-yu_untethering_2015">2015</a>)</span>. This lack of complexation may be the cause of high, ubiquitous AFB1 accumulation <span class="citation">(G. Parry et al. <a href="#ref-parry_complex_2009">2009</a>)</span>. Higher order mutants in the family containing <em>afb1</em> mutants suggest that <em>AFB1</em> has a moderate positive effect on auxin signaling. Additionally, AFB4 and AFB5 have been shown to preferentially bind the synthetic auxin picloram, while other family members preferentially bind indole-3-acetic acid <span class="citation">(M. J. Prigge et al. <a href="#ref-prigge_arabidopsis_2016">2016</a>)</span>. Interestingly, the strength and rate with which each TIR1/AFB is able to bind and mark an Aux/IAAs for degradation is variable <span class="citation">(Calderón Villalobos et al. <a href="#ref-calderonvillalobos_combinatorial_2012">2012</a>; Havens et al. <a href="#ref-havens_synthetic_2012">2012</a>)</span>.</p>
<p>(ref:TIR1AFBdivstats) <strong><em>TIR1/AFB</em> nucleotide diversity.</strong> The nucleotide diversity as Nei and Li’s <span class="math inline">\(\pi\)</span> (the average number of nucleotide differences per site between all possible pairs of sequences at synonymous (<span class="math inline">\(\pi_S\)</span>) and nonsynonymous (<span class="math inline">\(\pi_N\)</span>) sites, as well as all sites in the coding sequence and full transcript of the <em>TIR1/AFB</em> family.</p>
<div class="figure">
<div id="htmlwidget-66272f940465a949a758" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-66272f940465a949a758">{"x":{"filter":"none","data":[["TIR1","AFB1","AFB2","AFB3","AFB4","AFB5","COI1"],["AT3G62980.1","AT4G03190.1","AT3G26810.1","AT1G12820.1","AT4G24390.1","AT5G49980.1","AT2G39940.1"],[8.24934764195103e-05,0.000155905305232546,0.000224563382205154,0.000352823887379111,0.00085301134552636,0.000198930991690219,0.000254655007842795],[0.000267186918390591,0.000627138440963786,0.00127514192277866,0.000470398761868722,0.00220820263610869,0.00166605923370737,0.00176830083644993],[0.308748186162752,0.248597909247839,0.176108539915156,0.750052755193213,0.386292150719258,0.119402112281177,0.144011133509411],[0.000349680394810101,0.000783043746196332,0.00149970530498381,0.000823222649247832,0.00306025939634989,0.00186499022539759,0.0020188803271466],[0.00131150246937744,0.00132004688921502,0.00322448189903305,0.00186821680486955,0.00449590260756773,0.00339259199705168,0.00235101788883503]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>transcript<\/th>\n      <th>symbol<\/th>\n      <th>&pi;<sub>N<\/sub><\/th>\n      <th>&pi;<sub>S<\/sub><\/th>\n      <th>&pi;<sub>N<\/sub>/&pi;<sub>S<\/sub><\/th>\n      <th>&pi; coding<\/th>\n      <th>&pi; transcript<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"paging":false,"searching":false,"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6]}],"order":[],"autoWidth":false,"orderClasses":false,"rowCallback":"function(row, data) {\nDTWidget.formatRound(this, row, data, 1, 6, 3, ',', '.');\nDTWidget.formatRound(this, row, data, 2, 6, 3, ',', '.');\nDTWidget.formatRound(this, row, data, 3, 6, 3, ',', '.');\nDTWidget.formatRound(this, row, data, 4, 6, 3, ',', '.');\nDTWidget.formatRound(this, row, data, 5, 6, 3, ',', '.');\nDTWidget.formatRound(this, row, data, 6, 6, 3, ',', '.');\n}"}},"evals":["options.rowCallback"],"jsHooks":[]}</script>
<p class="caption">
(#fig:TIR1AFBdivstats)(ref:TIR1AFBdivstats)
</p>
</div>
<p>Examining the natural sequence variation across the <em>AFB</em> family @ref(fig:TIR1AFBdivstats) revealed that <em>TIR1</em> and <em>AFB1</em> both had very low nonsynonymous diversity, hinting at their likely functional importance and bringing in to question the inconclusive role of <em>AFB1</em> in auxin signaling. <em>AFB3</em> and <em>AFB4</em> had higher nonsynonymous diversity, while their sister genes, <em>AFB2</em> and <em>AFB5</em> were more conserved. This matches our current understanding of <em>AFB3</em> as playing a minor role in the auxin signaling pathway and <em>AFB4</em> perhaps undergoing pseudogenization.</p>
<p>(ref:AFBtree) <strong>AFB protein sequence tree mapped with <span class="math inline">\({\pi_N/\pi_S}\)</span>.</strong> Protein sequences were aligned <span class="citation">(E. S. Wright <a href="#ref-wright_decipher_2015">2015</a>)</span> and low information content regions were masked <span class="citation">(Kück et al. <a href="#ref-kuck_parametric_2010">2010</a>)</span> prior to infering a phylogeny <span class="citation">(Ronquist and Huelsenbeck <a href="#ref-ronquist_mrbayes_2003">2003</a>)</span>. Tips of the tree are mapped with circles of diameter proportional to <span class="math inline">\({\pi_N/\pi_S}\)</span> and also are colored according to <span class="math inline">\({\pi_N/\pi_S}\)</span>. Nodes are labeled with the poster probability of monophyly.</p>
<div class="figure">
<img src="manuscript_files/figure-html/AFBtree-1.png" alt="(ref:AFBtree)" width="90%" style="margin: auto; display: block; padding-top: 15px;" />
<p class="caption">
(#fig:AFBtree)(ref:AFBtree)
</p>
</div>
<p><img src="manuscript_files/figure-html/full-AFBs-alignment-1.png" width="90%" style="margin: auto; display: block; padding-top: 15px;" /></p>
<p>(ref:AFBstructure) A TIR1 structure (2P1Q) in gold in complex with ASK1 in grey and auxin, inositol phosphate and an Aux/IAA degron in teal. TIR1 residues or analagous residues containing nonsynonymous polymorphisms in TIR1 and AFB1 colored in purple and in AFB2 and AFB3 colored in blue.</p>
<div class="figure">
<img src="structures/natural_variation_TIR1AFB1-3.png" alt="(ref:AFBstructure)" width="90%" style="margin: auto; display: block; padding-top: 15px;" />
<p class="caption">
(#fig:AFBstructure)(ref:AFBstructure)
</p>
</div>
<p>Although most known functional regions are highly conserved in <em>AFB1</em>, there is a nonsynonymous polymorphism in the oligomerization domain, only found in the Can-0 accession. Mutations in this domain of <em>TIR1</em> frequently have a semidominant effect on root phenotypes <span class="citation">(Dezfulian et al. <a href="#ref-dezfulian_oligomerization_2016">2016</a>; R. C. Wright et al. <a href="#ref-wright_insights_2017">2017</a>)</span>. Characterization of this allele and accession may help determine the role of <em>AFB1</em> in this pathway.</p>
<p><strong>not sure this is very informative, and a map doesn’t seem very exciting, perhaps a domain scale alignment plot would be better here, or above, and would allow us to comment more on where the functional diversity lies within the family.</strong></p>
<p><img src="manuscript_files/figure-html/AFB1%20diversity%20plot-1.png" width="90%" style="margin: auto; display: block; padding-top: 15px;" /></p>
<p>The AFB4 and AFB5 receptors have an N-terminal extension prior to the F-box domains. This extension had very high nonsynonymous diversity @ref(fig:FBoxAlignment), suggesting that this extension does not play an important functional role in these proteins. Additionally, two frameshift variants and one stop-gained variant were observed in <em>AFB4</em> supporting its pseudogenization.</p>
<p>(ref:FBoxAlignment) <strong>Alignment of A. thaliana TIR1/AFB F-box domains showing variants.</strong> Cyan marks the F-box binding domain. Yellow, green, and magenta, mark missense and synonymous, synonymous only, and missense only variant locations respectively. Yellow squared E’s show Cul1-interacting residues, differing in AFB1.</p>
<div class="figure">
<img src="manuscript_files/figure-html/FBoxAlignment-1.png" alt="(ref:FBoxAlignment)" width="90%" style="margin: auto; display: block; padding-top: 15px;" />
<p class="caption">
(#fig:FBoxAlignment)(ref:FBoxAlignment)
</p>
</div>
<!--chapter:end:03-AFBs.Rmd-->
</div>
<div id="IAAs" class="section level1">
<h1><span class="header-section-number">4</span> <em>Aux/IAA</em> genes</h1>
<p>The Aux/IAA proteins (IAAs) mediate both auxin perception and transcriptional co-repression of auxin responsive genes. Auxin perception is carried out along with TIR1/AFB proteins, which together with IAAs act as co-receptors to bind and perceive nuclear auxin. Interaction of IAAs with auxin and TIR1/AFBs is conferred by the highly conserved 13 amino acid degron motif <span class="citation">(Tan et al. <a href="#ref-tan_mechanism_2007">2007</a>)</span>. Formation of this auxin co-receptor complex triggers ubiquitination and subsequent degradation of IAAs. This degradation in turn relieves their repression upon ARF transcription factors. Stabilizing mutations within the degron can lead to dominant phenotypes <span class="citation">(Berleth, Krogan, and Scarpella <a href="#ref-berleth_auxin_2004">2004</a>; Yang et al. <a href="#ref-yang_iaa1_2004">2004</a>)</span>. The <em>Arabidopsis thaliana</em> Aux/IAA family has 29 members, most of which possess a canonical auxin degron. Notable exceptions are the auxin-insensitive IAA20 and IAA30 that lack a degron sequence entirely, while the closely related IAA31 retains a semi-functional degron sequence with a very slow degradation rate <span class="citation">(Dreher et al. <a href="#ref-dreher_arabidopsis_2006">2006</a>; Havens et al. <a href="#ref-havens_synthetic_2012">2012</a>)</span>. Additionally, IAA degradation rate is influenced by rate motifs found outside the degron, such as the KR motif <span class="citation">(Dreher et al. <a href="#ref-dreher_arabidopsis_2006">2006</a>; Moss et al. <a href="#ref-moss_rate_2015">2015</a>)</span>. IAA-mediated repression of ARFs depends upon both an EAR domain, N-terminal of the degron, that recruits co-repressors of the TOPLESS (TPL) family, and a Phox and Bem1 (PB1) domain that enables interaction via hetero-oligomerization with ARF PB1 domains <span class="citation">(Nanao et al. <a href="#ref-nanao_structural_2014">2014</a>; Kagale and Rozwadowski <a href="#ref-kagale_ear_2011">2011</a>)</span>.</p>
<p>(ref:IAAdivstats) <strong><em>Aux/IAA</em> family nucleotide diversity.</strong> The nucleotide diversity as Nei and Li’s <span class="math inline">\(\pi\)</span> (the average number of nucleotide differences per site between all possible pairs of sequences at synonymous (<span class="math inline">\(\pi_S\)</span>) and nonsynonymous (<span class="math inline">\(\pi_N\)</span>) sites, as well as all sites in the coding sequence and full transcript of the <em>Aux/IAA</em> family.</p>
<div class="figure">
<div id="htmlwidget-fb446b5dc171e92084db" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-fb446b5dc171e92084db">{"x":{"filter":"none","data":[["IAA1","IAA2","IAA3","IAA4","IAA5","IAA6","IAA7","IAA8","IAA9","IAA10","IAA11","IAA12","IAA13","IAA14","IAA15","IAA16","IAA17","IAA18","IAA19","IAA20","IAA26","IAA27","IAA28","IAA29","IAA30","IAA31","IAA32","IAA33","IAA34"],["AT4G14560.1","AT3G23030.1","AT1G04240.1","AT5G43700.1","AT1G15580.1","AT1G52830.1","AT3G23050.1","AT2G22670.1","AT5G65670.1","AT1G04100.1","AT4G28640.1","AT1G04550.1","AT2G33310.1","AT4G14550.1","AT1G80390.1","AT3G04730.1","AT1G04250.1","AT1G51950.1","AT3G15540.1","AT2G46990.1","AT3G16500.1","AT4G29080.1","AT5G25890.1","AT4G32280.1","AT3G62100.1","AT3G17600.1","AT2G01200.1","AT5G57420.1","AT1G15050.1"],[0.000978002684921435,0.00141687273463947,0.00109975838325222,0.00325610820988199,0.000507301240682839,0.00262810080039828,0.00139699927625588,0.00111691929160705,0.000805774898786333,0.000491899287447567,0.000541733158745168,0.00122016421700908,0.000187329234369619,0.000663592882986632,0.000412419274997007,6.22434967832331e-05,5.02352619606025e-05,0.00146734078333859,0.00236555135334682,0.000398340407861304,0.000443286583422058,0.000463569276031215,6.54252502166953e-05,0.000352721395033481,0.000103809594389683,0.000707326932955556,0.000336182107878931,0.000571423385780743,0.00250699427223016],[0.000176654832006112,0.00443852102254111,0.000447992485046701,0.00319115635322068,0.00201447625559741,0.00117435514093195,0.00259998569239452,0.00146354318938311,0.000883866192123875,0.000738110147070409,0.000675679514034984,0.00057669494919639,0.000667455751318308,0.000203402190028366,0.00151159800422187,0.00176316370827969,0.000542209818198006,0.00128530823740691,0.00426746473352174,0.00131182695143566,0.00048863499597182,0.000103779159300869,0.000699687550009876,0.000889858975690369,0.00178947361553536,6.40661816367797e-05,0.000407149982350127,0.000349813916110428,0.00106541372526838],[5.53623511915938,0.319221814528726,2.45485899866729,1.0203537055136,0.251827858121064,2.23790973343265,0.537310370723341,0.763161141884604,0.911648059363043,0.666430734491237,0.801760520324313,2.11578793729571,0.280661652850575,3.26246675561403,0.272836609895704,0.0353021653581809,0.0926491189841521,1.14162559659536,0.554322414140875,0.303653166620309,0.907193686650358,4.46688216742311,0.0935063804061853,0.396378982141337,0.0580112461499614,11.0405664093689,0.825695990304212,1.63350672876135,2.35307112417634],[0.00115465751692755,0.00585539375718058,0.00154775086829892,0.00644726456310267,0.00252177749628025,0.00377311786526529,0.00399698496865041,0.00258046248099016,0.00168964109091021,0.00122773086504421,0.00121741267278015,0.00179685916620547,0.000854784985687927,0.000866995073014997,0.00192070376335355,0.00182540720506293,0.000592445080158608,0.00274598082705056,0.00663301608686856,0.00171016735929696,0.000931921579393879,0.000541311087253382,0.000765112800226571,0.00124258037072385,0.00189328320992505,0.000771393114592335,0.000743332090229059,0.000921237301891171,0.00356594415540432],[0.00465726691089045,0.00992181446645353,0.00297557600149668,0.016298471531899,0.00385065766630035,0.00602953421695526,0.00580842086339012,0.00664599120908252,0.00592091731578971,0.00155219088018314,0.00217875326462765,0.00342466804397996,0.00326960827089915,0.00626001341636234,0.00551048662677981,0.00873021756748692,0.00290446316606571,0.00830063236585972,0.0141513990350614,0.00686193866494216,0.00384902037174487,0.0018219020961383,0.00516767803775319,0.00250753703596684,0.00580149547003706,0.00262277036470645,0.00440001657785426,0.00260451647426036,0.00410987401515705]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>transcript<\/th>\n      <th>symbol<\/th>\n      <th>&pi;<sub>N<\/sub><\/th>\n      <th>&pi;<sub>S<\/sub><\/th>\n      <th>&pi;<sub>N<\/sub>/&pi;<sub>S<\/sub><\/th>\n      <th>&pi; coding<\/th>\n      <th>&pi; transcript<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"paging":false,"searching":false,"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6]}],"order":[],"autoWidth":false,"orderClasses":false,"rowCallback":"function(row, data) {\nDTWidget.formatRound(this, row, data, 1, 6, 3, ',', '.');\nDTWidget.formatRound(this, row, data, 2, 6, 3, ',', '.');\nDTWidget.formatRound(this, row, data, 3, 6, 3, ',', '.');\nDTWidget.formatRound(this, row, data, 4, 6, 3, ',', '.');\nDTWidget.formatRound(this, row, data, 5, 6, 3, ',', '.');\nDTWidget.formatRound(this, row, data, 6, 6, 3, ',', '.');\n}"}},"evals":["options.rowCallback"],"jsHooks":[]}</script>
<p class="caption">
(#fig:IAAdivstats)(ref:IAAdivstats)
</p>
</div>
<div id="sister-pair-diversification" class="section level3">
<h3><span class="header-section-number">4.0.1</span> Sister Pair Diversification</h3>
<p>Previous research has found evidence of both broad genetic redundancy and specificity within closely related pairs or groups of IAA proteins <span class="citation">(Overvoorde et al. <a href="#ref-overvoorde_functional_2005">2005</a>)</span>. For example, the <em>iaa8-1 iaa9-1</em> double mutant and the <em>iaa5-1 iaa6-1 iaa19-1</em> triple mutant have wild-type phenotypes <span class="citation">(Overvoorde et al. <a href="#ref-overvoorde_functional_2005">2005</a>)</span>, yet the <em>IAA6/19</em> sister pair has significant differences in expression patterns, protein abundances and functions suggesting they have undergone functional specialization since their divergence <span class="citation">(Winkler et al. <a href="#ref-winkler_variation_2017">2017</a>)</span>.</p>
<p>In our analysis of the <em>IAA</em> gene family, we frequently observed that one member of most <em>IAA</em> sister pairs had high nonsynonymous diversity while the other sister was more conserved @ref(fig:IAAtree). In an interspecific comparison between <em>A. thaliana</em> and <em>A. lyrata</em>, <em>IAA19</em> is more conserved, while <em>IAA6</em> has regions of increased sequence divergence upstream of the degron and in the N-terminal half of the PB1 domain <span class="citation">(Winkler et al. <a href="#ref-winkler_variation_2017">2017</a>)</span>. Consistent with these results, our analysis of intraspecific natural variation revealed higher conservation for <em>IAA19</em> (<span class="math inline">\({\pi_N/\pi_S}\)</span> = 0.55) compared to <em>IAA6</em> (<span class="math inline">\({\pi_N/\pi_S}\)</span> = 2.3) @ref(fig:IAAtree), and also detected high diversity within the same regions of <em>IAA6</em> as seen in Winkler et al. (Supp Fig IAA6 diversity plot).</p>
<p><strong>I think IAA20, 30, and 15 should also be in class C. So there are really two large clades A and B and then there are truncations/loss of function/dominant variants scattered throughout, Class C.</strong></p>
<p>(ref:IAAtree) <strong>IAA protein sequence tree mapped with <span class="math inline">\({\pi_N/\pi_S}\)</span>.</strong> Protein sequences were aligned <span class="citation">(E. S. Wright <a href="#ref-wright_decipher_2015">2015</a>)</span> and low information content regions were masked <span class="citation">(Kück et al. <a href="#ref-kuck_parametric_2010">2010</a>)</span> prior to infering a phylogeny <span class="citation">(Ronquist and Huelsenbeck <a href="#ref-ronquist_mrbayes_2003">2003</a>)</span>. Tips of the tree are mapped with circles of diameter proportional to <span class="math inline">\({\pi_N/\pi_S}\)</span> and also are colored according to <span class="math inline">\({\pi_N/\pi_S}\)</span>. Nodes are labeled with the poster probability of monophyly.</p>
<div class="figure">
<img src="manuscript_files/figure-html/IAAtree-1.png" alt="(ref:IAAtree)" width="90%" style="margin: auto; display: block; padding-top: 15px;" />
<p class="caption">
(#fig:IAAtree)(ref:IAAtree)
</p>
</div>
<p>Within the Group B <em>IAA</em> genes, there are four sister pairs—<em>IAA10</em>/<em>IAA11</em>, <em>IAA12</em>/<em>IAA13</em>, <em>IAA18</em>/<em>IAA26</em>, and <em>IAA20</em>/<em>IAA30</em> @ref(fig:IAAtree) <span class="citation">(Remington et al. <a href="#ref-remington_contrasting_2004">2004</a>)</span>. Interestingly, among these sister pairs only <em>IAA12</em> and <em>IAA13</em> have a noticeable difference in nonsynonymous diversity. The <em>Arabidopsis thaliana</em> EFP browser shows that <em>IAA10</em> and <em>IAA11</em> have almost identical expression patterns, with the exception of <em>IAA10</em> being more highly expressed in the developing seed, and <em>IAA11</em> being more highly expressed in root tissue during lateral root development <strong>need to cite original data source here</strong>. In our analysis, <em>IAA10</em> and <em>IAA11</em> both showed functional conservation (<span class="math inline">\({\pi_N/\pi_S}\)</span> of 0.80 and 0.67 respectively) in support of <em>IAA10</em> and <em>IAA11</em> playing redundant roles. Accessions and associated variants identified here provide a potentially expedited route to elucidating the extent of redundancy between family members such as <em>IAA10</em> and <em>IAA11</em>.</p>
</div>
<div id="new-loss-of-functiongain-of-function-alleles" class="section level3">
<h3><span class="header-section-number">4.0.2</span> New Loss-of-Function/Gain-of-Function alleles</h3>
<p>Accessions with functional variants can be potentially used to investigate the extent of redundancy/specialization within sister pairs. For example, three accessions have a missense variant affecting the EAR domain of IAA10 (HR-5, UKSE06-118 and UKSW06-240, all from southern UK). This variant (L49S) is in the second leucine of the LxLxL motif and likely reduces the ability of IAA10 to recruit the TOPLESS co-repressor <span class="citation">(M.-S. Lee, An, and Cho <a href="#ref-lee_biological_2016">2016</a>)</span>. The nonsense variant (IAA10-Y219*) resulting in truncated PB1 domain may be more indicative of <em>IAA10</em> null phenotype, as it likely prevents IAA10 from interacting with an ARF. This variant is only found in one accession from northeastern Spain (IP-Svi-0). These accessions and potential loss-of-function alleles may facilitate exploration of the function of <em>IAA10</em>.</p>
<p>Within the available natural sequence variation, we found several missense polymorphisms affecting the EAR domain or degron of several canonical clade A IAAs. However, no missense polymorphisms were present in the KR motif or residues critical for PB1-PB1 domain interactions (@ref(fig:A-IAA-alignment)). The majority of missense mutations observed were in the variable regions between the conserved domains.</p>
(ref:A-IAA-alignment)
<div class="figure">
<img src="manuscript_files/figure-html/A-IAA-alignment-1.png" alt="(ref:A-IAA-alignment)" width="90%" style="margin: auto; display: block; padding-top: 15px;" />
<p class="caption">
(#fig:A-IAA-alignment)(ref:A-IAA-alignment)
</p>
</div>
<p>Several examples of possible gain-of-function missense variants were found in the degron of <em>IAA2</em>, <em>IAA3</em>, <em>IAA6</em>, <em>IAA7</em>, <em>IAA8</em>, <em>IAA13</em>, <em>IAA15</em>, <em>IAA18</em>, <em>IAA19</em>, <em>IAA27</em>, <em>IAA28</em> and <em>IAA29</em>. Most degron variants were very minor alleles in the population, only represented in one or two accessions. Interestingly, degron missense variants occured more frequently in <em>IAA</em> genes that have a closely related sister, which may indicate redundant function within the sister-pairs. <strong>I think this is an interesting finding, but I’m not sure about this hypothesis of redundant function. If pairs are acting redundantly and we assume the degron variants are dominant, we would expect the auxin signal they transmit (their function) to be lower, in sum, as a pair. So I would actually hypothesize that there has been some divergence in function as a pair.</strong></p>
<p><em>IAA6</em> has two potentially dominant missense mutations found within the degron (Supp Fig. alignment), one previously believed to lead to the <em>shy1-1</em> phenotype <span class="citation">(Kim et al. <a href="#ref-kim_two_1996">1996</a>)</span>. Interestingly, this variant is widely represented in 31 accessions across Europe (Supplemental Map Fig). This finding supports the proposition that iaa6-C78R may not be the mutation causing the <em>shy1-1D</em> phenotype but rather represents natural variation in the degron sequence (<a href="https://www.arabidopsis.org/servlets/TairObject?type=notepad&amp;id=11247">note from Mike Prigge in TAIR</a>). These accessions could be used to confirm whether <em>iaa6-C78R</em> is responsible for the <em>shy1-1D</em> phenotype. Additionally, an accession was identified with the IAA6-V77M variant, also in the degron, that may be useful to further explore <em>iaa6</em> gain-of-function phenotypes.</p>
</div>
<div id="identifying-novel-functional-sequences-in-iaas" class="section level3">
<h3><span class="header-section-number">4.0.3</span> Identifying novel functional sequences in IAAs</h3>
<p>An exciting possible use for the Natural Variation Webtool is to search for conserved, low-diversity sites that may represent possible sites of functional import, such as post-translational modification or protein-protein interaction. For example, recent detailed biochemical analysis identified putative lysine ubiquitination sites in members of the <em>IAA6/19</em> sister pairs <span class="citation">(Winkler et al. <a href="#ref-winkler_variation_2017">2017</a>)</span>; Supp Fig IAA6/19 alignment). These sites are conserved across natural sequence variation, while several other lysine residues vary (Supp Figs alignment and map). In IAA6, there are 17 lysines of which ~6 are putative ubiquitination sites~ **I think this should be 5, K3, K32, K33, K91, and K97. Missense variants occured in K34Q, K140N *. One is proximal to the KR motif, while two others show natural variation. In IAA19, there are 15 total lysines, 8 of which are putative ubiquitination sites (K3, K25, K68, L87, K93, K100, K111, and K141). Two of these ubiquitination sites are proximal to the KR motif and degron. <strong>This sentence is not clear. Only 1 of the ubiquitination sites varies? or one lysine, I think you mean this only 1 lysine that varies, but it is not ubiquitinated.</strong> Natural sequence diversity may be used to narrow down a subset of lysines to assess for functional ubiquitination.</p>
<p>The members of the <em>IAA8/9/27</em> clade have a unique N-terminal extension with a highly conserved region common to orthologs in several other species <span class="citation">(Dreher et al. <a href="#ref-dreher_arabidopsis_2006">2006</a>)</span>. <em>IAA8</em> and <em>IAA9</em> play redundant roles in leaf serration <span class="citation">(Koenig et al. <a href="#ref-koenig_auxin_2009">2009</a>)</span>, while <em>IAA8</em> also plays an important role in lateral root development <span class="citation">(Arase et al. <a href="#ref-arase_iaa8_2012">2012</a>)</span>. IAA8 has been reported to uniquely localize in the cytosol <span class="citation">(Arase et al. <a href="#ref-arase_iaa8_2012">2012</a>)</span> where it may interact with LSD1 protein <span class="citation">(N. S. Coll, Epple, and Dangl <a href="#ref-coll_programmed_2011">2011</a>)</span>. The N-terminal extension has low natural sequence diversity in <em>IAA8</em> and <em>IAA9</em> (Supp Fig IAA8/9 alignment) that may represent a site of novel functionality, possibly mediating interaction with LSD1. The few accessions with missense alleles in this region in both <em>IAA8</em> and <em>IAA9</em> may be valuable in probing the functionality of this region (Supplemental Fig IAA8/9 map).</p>
<p>Members of the <em>IAA17/7/14/16</em> clade have a second EAR motif (LxLxL) with no observed missense variants in <em>IAA7</em>, <em>IAA16</em>, or <em>IAA17</em> (Supp Fig IAA7/14/16/17 alignment). This observation supports the experimental evidence for IAA7 that the second EAR motif is functional, <em>i.e.</em> it can interact with TPR1 co-repressor and has a minor repressive role in auxin-related developmental phenotypes [lee_biological_2016]. For <em>IAA14</em>, there were two missense variants that may be of interest to probe functionality of this second EAR motif. One variant, L41F, in the central lysine is represented in only one accession, and the other in the second “x” (K40N) is found in twelve accessions around Copenhagen (Supp Fig. IAA14 map).</p>
<!--chapter:end:04-IAAs.Rmd-->
</div>
</div>
<div id="TPLs" class="section level1">
<h1><span class="header-section-number">5</span> <em>TPL/TPR</em> genes</h1>
<p>The Auxin signaling pathway utilizes the TOPLESS (TPL) and TOPLESS-related (TPR) family of Gro/TLE/TUP1 type co-repressor proteins to maintain auxin responsive genes in a transcriptionally-repressed state in the absence of auxin <span class="citation">(Szemenyei, Hannon, and Long <a href="#ref-szemenyei_topless_2008">2008</a>)</span>. In <em>Arabidopsis thaliana</em> the five member <em>TPL/TPR</em> family includes <em>TPL</em> and <em>TPR1-4</em>. The resulting proteins are comprised of three structural domains: an N-terminal TPL domain and two WD-40 domains <span class="citation">(Long et al. <a href="#ref-long_topless_2006">2006</a>)</span>. TPL/TPR proteins are recruited to the AUX/IAA proteins through interaction with the conserved Ethylene-responsive element binding factor-associated amphiphilic repression (EAR) domain <span class="citation">(Szemenyei, Hannon, and Long <a href="#ref-szemenyei_topless_2008">2008</a>)</span>. Canonical EAR domains have the amino acid sequence LxLxL, as found in most AUX/IAAs (figure xA) <span class="citation">(Overvoorde et al. <a href="#ref-overvoorde_functional_2005">2005</a>)</span>. TPL/TPR co-repressors bind EAR domains via their C-terminal to LisH (CTLH) domains found near their N-termini (citations of pre-structure founding papers/reviews). Recent structural analyses of the TPL N-terminal domain have highlighted the precise interaction interface between TPL and AUX/IAA EAR domains, as well as the TPL-TPL dimerization and tetramerization motifs <span class="citation">(Martin-Arevalillo et al. <a href="#ref-martin-arevalillo_structure_2017">2017</a>; Ke et al. <a href="#ref-ke_structural_2015">2015</a>)</span>. The residues required for higher-order multimers of TPL tetramers have also been identified <span class="citation">(Ma et al. <a href="#ref-ma_d53_2017">2017</a>)</span>. Additional interactions with transcriptional regulation and chromatin modifying machinery are likely mediated by two tandem beta propeller domains of TPL/TPRs. These domain have not yet been crystallized, however numerous high resolution crystal structures of beta propeller domains allow for structural prediction of the TPL C-terminal beta propeller domains and the locations of likely interaction faces (Figure, X).</p>
<p>The TOPLESS co-repressor family generally exhibits a high level of sequence conservation at the amino acid sequence level across resequenced <em>Arabidopsis thaliana</em> accessions, with all <span class="math inline">\({\pi_N/\pi_S}\)</span> values below 1 (Table X). The closely related <em>TPL</em> and <em>TPR1</em> have the highest <span class="math inline">\({\pi_N/\pi_S}\)</span> values (TPL-0.425, TPR1-0.739), suggesting that these these two related genes tolerate a higher degree of sequence and potentially functional diversity compared to <em>TPR2/3/4</em>. The N-terminal TPL domain of the TPL/TPR family is particularly conserved (Figure X). All nonsynonymous polymorphisms observed in this region are either in the coils between helices or are highly conservative mutations within helices (i.e. Valine to Isoleucine), which would be predicted to exhibit little effect on folding and function.</p>
<div id="htmlwidget-632172b4e74ed1327cc4" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-632172b4e74ed1327cc4">{"x":{"filter":"none","data":[["TPL","TPR1","TPR2","TPR3","TPR4"],["AT1G15750.1","AT1G80490.1","AT3G16830.1","AT5G27030.1","AT3G15880.1"],[0.00017771503556095,0.000324192270617907,7.76271245176626e-05,0.000339276823118661,0.000104421135985865],[0.000417868107969103,0.00043824915307264,0.000448125431560226,0.00173653877104452,0.000910590599866674],[0.425289779650019,0.739744203371391,0.17322633140322,0.195375322898542,0.114674076364454],[0.000595583143530053,0.000760161531848352,0.000525752556077888,0.00207581559416318,0.00101449468386616],[0.0010112838324352,0.00137107729335683,0.00100982938699376,0.0035164685407768,0.00184247594972934]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>transcript<\/th>\n      <th>symbol<\/th>\n      <th>&pi;<sub>N<\/sub><\/th>\n      <th>&pi;<sub>S<\/sub><\/th>\n      <th>&pi;<sub>N<\/sub>/&pi;<sub>S<\/sub><\/th>\n      <th>&pi; coding<\/th>\n      <th>&pi; transcript<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"paging":false,"searching":false,"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6]}],"order":[],"autoWidth":false,"orderClasses":false,"rowCallback":"function(row, data) {\nDTWidget.formatRound(this, row, data, 1, 6, 3, ',', '.');\nDTWidget.formatRound(this, row, data, 2, 6, 3, ',', '.');\nDTWidget.formatRound(this, row, data, 3, 6, 3, ',', '.');\nDTWidget.formatRound(this, row, data, 4, 6, 3, ',', '.');\nDTWidget.formatRound(this, row, data, 5, 6, 3, ',', '.');\nDTWidget.formatRound(this, row, data, 6, 6, 3, ',', '.');\n}"}},"evals":["options.rowCallback"],"jsHooks":[]}</script>
<p>The high degree of conservation in the entire N-terminal domain underscores its importance in TPL/TPR function. For example, the initial <em>tpl-1</em> mutation (N176H) in the ninth helix is a dominant gain-of-function allele <span class="citation">(Long et al. <a href="#ref-long_topless_2006">2006</a>)</span>, which is capable of binding wild-type TPL protein and inducing protein aggregation <span class="citation">(Ma et al. <a href="#ref-ma_d53_2017">2017</a>)</span>. It is therefore understandable that this helix had very low diversity as nonsynonymous variants in this domain could act in a dominant negative fashion. Addtionally, the crystal structure of TPL has revealed that the The LisH domain and CRA domain coordinate TPL dimerization and tetramerization respectively (26601214, 28698367), the CTLH domain is critical for interaction with DNA-binding transcription factors (26601214), and helix nine within the CRA domain may be critical for the formation of higher-order TPL complexes (multimers of TPL tetramers, 28630893).</p>
<p>Compared to the TPL domain, both the proline rich and WD-40 domains exhibited a higher frequency of missense variants. We utilized Phyre2.0 modeling software to predict the structure of the TPL/TPR WD-40 repeats in order to map non-synonymous mutations to protein structure (Figure X). Here we can see both faces of each predicted beta-propeller structure of the WD-40 domains, which are connected by a short peptide linker. This model reveals amino acid conservation in the core and central residues of the WD-40 domains. <del>These observations could be used as the basis for forward mutagenesis strategies to test whether this region of TPL functions similarly to its homolog TLE, which binds its target genes EH1 and the C-terminal WRPW/Y motif (Hairy/Hes/Runx) in the center of the WD-40 domain (pmid 16762837).</del> **Can we say something more general here? There is not enough background to support this. What is TLE from? What is EH1? Is</p>
<p><img src="manuscript_files/figure-html/Visualize%20TPL%20tree-1.png" width="90%" style="margin: auto; display: block; padding-top: 15px;" /></p>
<p>(ref:full-TPLs-alignment) <strong>Alignment of A. thaliana TPL/TPR genes showing variants.</strong> This alignment should probably be paired down to a key region rather than the eniter AA sequence.</p>
<div class="figure">
<img src="manuscript_files/figure-html/full-TPLs-alignment-1.png" alt="(ref:full-TPLs-alignment)" width="90%" style="margin: auto; display: block; padding-top: 15px;" />
<p class="caption">
(#fig:full-TPLs-alignment)(ref:full-TPLs-alignment)
</p>
</div>
<p>Despite the presence of five <em>TPL/TPR</em> genes in the genome, which are considered to be redundant (pmid 16763149), very few observed SNPs are predicted to result in loss of function for any <em>TPL/TPR</em> gene. Only one frameshift variant was observed, <em>TPR4-665delG</em>, in a single accession (Aiell-4 from Italy). This deletion occurs in codon 222, just after the TPL N-terminal domain, and generates an early stop eleven codons downstream from the deletion. It is unclear whether this N-terminal fragment would be expressed at any appreciable level, as it is possible that the generation of an early stop codon would trigger nonsense-mediated decay. However, if expressed, it would be fascinating to examine whether this naturally occurring N-terminal truncation retains activity as a co-repressor. Given the observation that the TPL N-terminal domain is sufficient to bind histones (28630893), and repress transcription via synthetic assays in yeast (24979769), it is possible that this represents a semi-functional TPL variant. There has been no evidence for prioritization of individual <em>TPL/TPR</em> family members in the auxin response pathway, but experiments modulating the gene copy number of <em>TPR1</em> revealed the importance of maintaining the proper dosage of <em>TPL/TPR</em> in order to accurately trigger the salicylic acid innate immune response (20647385). These results, taken together with the absence of <em>TPL/TPR</em> loss-of-function alleles in the <em>Arabidopsis thaliana</em> accessions suggest that perhaps the entire family is being conserved at the nucleotide level in the population to meet a gene dosage requirement, with little opportunity for neo- or sub-functionalization.</p>
<!--chapter:end:05-TPLs.Rmd-->
</div>
<div id="ARFs" class="section level1">
<h1><span class="header-section-number">6</span> <em>ARF</em> genes</h1>
<p>Auxin response is ultimately mediated by the auxin responsive transcription factors (ARFs). There are 23 ARFs in <em>Arabidopsis thaliana</em> that are divided into three phylogenetic classes. Class A ARFs (ARF5, ARF6, ARF7, ARF8 and ARF19) activate transcription. These ARFs have a glutamine-rich region in the middle of the protein that may mediate activation <span class="citation">(T. J. Guilfoyle and Hagen <a href="#ref-guilfoyle_auxin_2007">2007</a>)</span>. It has recently been shown that the middle region of ARF5 interacts with the SWI/SNF chromatin remodeling ATPases BRAMA and SPLAYED, possibly to reduce nucleosome occupancy and allow for the recruitment of transcription machinery <span class="citation">(Wu et al. <a href="#ref-wu_auxinregulated_2015">2015</a>)</span>. Additionally, ARF7 interacts with Mediator subunits, directly tethering transcriptional activation machinery to its binding sites in the chromosome <span class="citation">(Ito et al. <a href="#ref-ito_auxindependent_2016">2016</a>)</span>. Class B and C ARFs are historically categorized as repressor ARFs, though the mechanism through which they confer repression has not been identified. Their middle regions tend to be proline- and serine-rich (citation, Ulmasov?).</p>
<p>Canonical ARFs are comprised of three major domains. Recent crystallization of these domains have informed structure-function analysis of the ARFs <span class="citation">(Boer et al. <a href="#ref-boer_structural_2014">2014</a>; Korasick et al. <a href="#ref-korasick_molecular_2014">2014</a>; Nanao et al. <a href="#ref-nanao_structural_2014">2014</a>)</span>. These domains are conserved throughout land plants <span class="citation">(Mutte et al. <a href="#ref-mutte_origin_2018">2018</a>)</span>. ARFs share an N-terminal B3 DNA binding domain. Flanking this DNA-binding domain is a dimerization domain, which folds up into a single “taco-shaped” domain to allow for dimerization between ARFs. There is an auxiliary domain that immediately follows and interacts with the dimerization domain. The middle region is the most variable between ARFs, as mentioned above, but is characterized by repetitive units of glutamine (class A), serine, or proline residues (classes B and C).</p>
<p>The C-terminal domain of the ARF is a protein-protein interaction domain mediating interactions among ARFs, between ARFs and other transcription factors, and between ARFs and the Aux/IAA repressors. This interaction domain was recently characterized as a Phox and Bem1 (PB1) domain, which is comprised of a positive and negative face with conserved basic and acidic residues, respectively <span class="citation">(Korasick et al. <a href="#ref-korasick_molecular_2014">2014</a>; Nanao et al. <a href="#ref-nanao_structural_2014">2014</a>)</span>. The dipolar nature of the PB1 domain may mediate multimerization by the pairwise interaction of these faces on different proteins as the ARF7 PB1 domain was crystallized as a multimer <span class="citation">(Korasick et al. <a href="#ref-korasick_molecular_2014">2014</a>)</span>. However, it is unclear whether ARF multimerization occurs or plays a developmental significant role in vivo. Interfering with ARF dimerization in either the DNA-binding proximal dimerization domain or the PB1 domain decreases the ability of class A ARFs to activate transcription in a heterologous yeast system <span class="citation">(Pierre-Jerome et al. <a href="#ref-pierre-jerome_functional_2016">2016</a>)</span>.</p>
<p>While domain architecture is broadly conserved among the ARFs, there are exceptional cases. Three ARFs do not contain a PB1 domain at all, ARF3, ARF13, and ARF17, and several more have lost the conserved acidic or basic residues in the PB1 domain, suggesting they may be reduced to a single interaction domain. Several ARFs additionally have an expanded conserved region within the DNA-binding domain, of unknown function. The majority of domain variation among ARFs occurs in the large B-class subfamily. The liverwort <em>Marchantia polymorpha</em> has a single representative ARF of each class <span class="citation">(Flores-Sandoval, Eklund, and Bowman <a href="#ref-flores-sandoval_simple_2015">2015</a>)</span>. The expansion of these classes in flowering plants is the result of both whole genome and tandem duplication events <span class="citation">(Remington et al. <a href="#ref-remington_contrasting_2004">2004</a>)</span>. The growth of the ARF family may have allowed for the expansion of the quantity and complexity of loci regulated by the ARFs and subsequent expansion in their regulation of developmental processes.</p>
<p>Class A <em>ARFs</em> are the most well-studied ARF subfamily—the five family members all act as transcriptional activators and have well-characterized, distinct developmental targets. Overall the diversity of class A <em>ARFs</em> was generally low, especially compared to the class B and C <em>ARFs</em>, suggesting that class A <em>ARFs</em> are central to auxin signal transduction and plant development. Analysis of class A <em>ARF</em> nonsynonymous diversity suggests that the majority of these <em>ARFs</em> are highly functionally conserved, with <span class="math inline">\({\pi_N/\pi_S}\)</span> values much lower than 1 with the exception of <em>ARF19</em>, with <span class="math inline">\({\pi_N/\pi_S}\)</span> value of 1.8. Comparing diversity within sister pairs, there is a similar trade-off as seen in most <em>IAA</em> sister pairs, with one sister being highly conserved and the other more divergent. <em>ARF19</em> and <em>ARF8</em> are the more divergent class A <em>ARFs</em>, with <span class="math inline">\({\pi_N/\pi_S}\)</span> values at least three time those of their sisters, <em>ARF7</em> and <em>ARF6</em> respectively. This may suggest that ARF6 and ARF7 serve more essential purposes in plant development.</p>
<div id="htmlwidget-ed14d4a4e1beb4c0221e" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-ed14d4a4e1beb4c0221e">{"x":{"filter":"none","data":[["ARF1","ARF2","ARF3","ARF4","ARF5","ARF6","ARF7","ARF8","ARF9","ARF10","ARF11","ARF12","ARF13","ARF14","ARF15","ARF16","ARF17","ARF18","ARF19","ARF20","ARF21","ARF22","ARF23"],["AT1G59750.1","AT5G62000.1","AT2G33860.1","AT5G60450.1","AT1G19850.1","AT1G30330.1","AT5G20730.1","AT5G37020.1","AT4G23980.1","AT2G28350.1","AT2G46530.1","AT1G34310.1","AT1G34170.1","AT1G35540.1","AT1G35520.1","AT4G30080.1","AT1G77850.1","AT3G61830.1","AT1G19220.1","AT1G35240.1","AT1G34410.1","AT1G34390.1","AT1G43950.1"],[0.00108287213427914,0.000334747431576637,0.000659788539481217,0.000708796428127634,0.000184561866215317,0.000150194671138632,0.00061470541606095,0.000316551109649951,0.000728371192291026,0.000846628172063065,0.000672900692489234,0.00338990176847809,0.00907891167292848,0.00441746833252994,0.00232272174765166,0.000314572671802899,0.00145787659261908,0.000283457847822655,0.000755195492635983,0.00313814793706543,0.00264585050481181,0.0025577570926552,0.00846794681967124],[0.00183158873313992,0.000551047164619097,0.000477804275035395,0.000201911285919784,0.000368262396258417,0.000754820360892703,0.00149009075525496,0.000474705958526308,0.00222702050218282,0.00218453829004825,0.000582382907872521,0.0014726804154286,0.00218959739729998,0.0015942407844556,0.000641314874960719,0.000421542378534317,0.00172388039881431,0.000211725482512999,0.000412462628344974,0.00216237274763304,0.0013376089841852,0.00219139604331105,0.00207336437336855],[0.591220132929489,0.607475100263016,1.38087617452217,3.51043491649708,0.501169459848425,0.198980683246277,0.412528843557431,0.666836183461152,0.327060838271184,0.387554741393141,1.15542658170958,2.30185838893736,4.14638402663604,2.77089155891744,3.62181174698923,0.746242104759795,0.845694744033179,1.33879892235103,1.8309428315148,1.45125207506453,1.97804480688616,1.16718157836527,4.0841575790721],[0.00291446086741906,0.000883746252775778,0.0011297611396823,0.000910707714047418,0.000492536058436823,0.000903129151004816,0.00210177907781209,0.000791257068176259,0.00295539169447385,0.00302945089917473,0.00122823159307685,0.00473284564205623,0.010061646664003,0.00548703691801734,0.00292433377344723,0.000734324635477015,0.0031817569914334,0.000434402519127592,0.00116765812098096,0.00521982009787175,0.00374949029820893,0.00446533329016077,0.00849421615057723],[0.0057536145733293,0.00119341344515052,0.00112845477646123,0.00128473575862142,0.000776194611463173,0.00110781089188202,0.00328758991822984,0.00169939503103635,0.00370270853634483,0.00381114886403862,0.00145743068341118,0.00539613343479218,0.0107052836100089,0.00666196550014211,0.00357444620188709,0.000707406920720416,0.00320263737040692,0.000926161267393217,0.00249422699386394,0.00550100309191848,0.00403073802727379,0.00528442957131736,0.0103227347466484]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>transcript<\/th>\n      <th>symbol<\/th>\n      <th>&pi;<sub>N<\/sub><\/th>\n      <th>&pi;<sub>S<\/sub><\/th>\n      <th>&pi;<sub>N<\/sub>/&pi;<sub>S<\/sub><\/th>\n      <th>&pi; coding<\/th>\n      <th>&pi; transcript<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"paging":false,"searching":false,"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6]}],"order":[],"autoWidth":false,"orderClasses":false,"rowCallback":"function(row, data) {\nDTWidget.formatRound(this, row, data, 1, 6, 3, ',', '.');\nDTWidget.formatRound(this, row, data, 2, 6, 3, ',', '.');\nDTWidget.formatRound(this, row, data, 3, 6, 3, ',', '.');\nDTWidget.formatRound(this, row, data, 4, 6, 3, ',', '.');\nDTWidget.formatRound(this, row, data, 5, 6, 3, ',', '.');\nDTWidget.formatRound(this, row, data, 6, 6, 3, ',', '.');\n}"}},"evals":["options.rowCallback"],"jsHooks":[]}</script>
<p><img src="manuscript_files/figure-html/Visualize%20ARF%20tree-1.png" width="90%" style="margin: auto; display: block; padding-top: 15px;" /></p>
<p><img src="manuscript_files/figure-html/full-ARFs-alignment-1.png" width="90%" style="margin: auto; display: block; padding-top: 15px;" /></p>
<div id="class-a-arfs" class="section level3">
<h3><span class="header-section-number">6.0.1</span> Class A <em>ARFs</em></h3>
<p>For all class A ARFs, the middle region of the protein was the predominant high diversity region. The DNA-binding domain had very few, low-diversity missense mutations, as did the PB1 domain. Considering the necessity of their conserved functions, the low level of variation in these key DNA and protein-protein interaction domains is expected.</p>
<p><img src="manuscript_files/figure-html/A-ARF-Alignment-1.png" width="90%" style="margin: auto; display: block; padding-top: 15px;" /></p>
<p>Class A ARFs all contain the canonical B3 DNA-binding domain, PB1 interaction domain and glutamine-rich middle region <span class="citation">(T. J. Guilfoyle and Hagen <a href="#ref-guilfoyle_auxin_2007">2007</a>)</span>. In the analyzed natural variation, ARF7 had several expansions of polyglutamine sequences in the middle region. Polyglutamine regions are known to readily expand and contract throughout evolutionary time due to replication error, and variation in polyglutamine length can be acted on by natural selection and have phenotypic consequences <span class="citation">(Press, Carlson, and Queitsch <a href="#ref-press_overdue_2014">2014</a>)</span>.</p>
<p>A few missense variants in <em>ARF5</em> highlighted potential functional variation in the class A ARFs. A high frequency allele <em>(What are the variants?)</em>, resulting in substitutions of small amino acids on the positive face of the PB1 domain for bulky phenylalanines, may disrupt interactions with other ARF or Aux/IAA PB1 domains. Interestingly, these variants were always found together in accessions Tu-B2-3, Ru-2, Haes-1, Rd-0, 627ME-13Y1, Le-0, Bu-0 and Bsch-0 <strong>Where are these?</strong>. Another highly represented polymorphism in ARF5 results in change in the 17th amino acid from a leucine to a methionine. This position is in the N-terminal extension of ARF5 which is of unknown function. It is possible that this methionine acts as an alternative start site in the accessions.</p>
</div>
<div id="class-b-arfs" class="section level3">
<h3><span class="header-section-number">6.0.2</span> Class B <em>ARFs</em></h3>
<p>ARF23 has a truncated DNA-binding domain and had a high <span class="math inline">\({\pi_N/\pi_S}\)</span> value of 4.1 and several high-frequency nonsense variants, as may be expected of a gene that is pseudogenizing.</p>
<p>ARF3, ARF13 and ARF17 all lack a C-terminal PB1 domain, which suggests they may be unable to interact with Aux/IAA repressors, possibly acting as auxin-insensitive competitive inhibitors of ARF-mediated transcription (Fig ARF3_13_17 alignment). These ARFs span the range of <span class="math inline">\({\pi_N/\pi_S}\)</span> ratios relative to the other ARFs, ranging from 0.85 to 4.1. ARF17 is more conserved than the activator ARF, ARF19. ARF13 has many nonsense variants of high diversity and may be pseudogenizing.</p>
<div class="figure">
<img src="manuscript_files/figure-html/B-ARF-alignment-1.png" alt="(ref:B-ARFalignment)" width="90%" style="margin: auto; display: block; padding-top: 15px;" />
<p class="caption">
(#fig:B-ARF-alignment)(ref:B-ARFalignment)
</p>
</div>
<p>ARF14, ARF15, <strong>ARF16 I don’t think this is correct ARF 16 is class C, perhaps this should be 12?</strong>, ARF20, and ARF21 all do not have one or multiple of the conserved acidic residues in the PB1 domain, and all are within a subclade of the B ARFs. Interestingly, no non-synonymous variation is seen in these important residues for Aux/IAA and ARF interaction, suggesting that the amino acids that are in the regions may still be important for the function of these proteins. <strong>Might cut this, it is interesting that perhaps homo/heterodimerization is not as important for this clade of ARFs but I think this conclusion mostly comes from looking at the alignments and not from the natural variation. Perhaps if it looks like these ARFs have higher than expected, or more deleterious variation across their PB1 domains</strong></p>
</div>
<div id="class-c-arfs" class="section level3">
<h3><span class="header-section-number">6.0.3</span> Class C <em>ARFs</em></h3>
<p>Class C ARFs show low nucleotide diversity scores, all <span class="math inline">\({\pi_N/\pi_S}\)</span> values substantially lower than 1. ARF16 was the most conserved, whereas its clade members (ARF10, ARF17) had scores at least four times higher <span class="citation">(Remington et al. <a href="#ref-remington_contrasting_2004">2004</a>)</span>.</p>
<p>Structurally, all three members of Class C ARFs contain a canonical B3 DNA-binding domain, but only ARF10 and ARF16 contain a PB1 domain. The DNA binding domains exhibit overall low diversity. Only the ARF10 DNA binding domain contains missense variants (S147T and A157S) (Fig. ARF10_DBD Variants distribution). <strong>Any insights into whether these may affect function? They are quite conservative.</strong> Of the PB1 domain containing class C ARFs, ARF16 exhibits several missense variants which are sporadically distributed, in contrast to the conserved PB1 domain of ARF10 (Fig. Arf16_PB_variants distribution). <strong>Interesting that the PB1 domain of ARF10 is conserved and the DBD of ARF16 is conserved. It this suggestive of subfunctionalization in this family, where one retains DNA and/or dimerization specificity and the other retains specificity for IAAs?</strong></p>
<p><img src="manuscript_files/figure-html/ARF10_DBD%20Variants%20distribution-1.png" width="90%" style="margin: auto; display: block; padding-top: 15px;" /></p>
<p>Several members of class B (ARF2, 3 and 4) and C ARFs (ARF10, 16 and 17) are distinguished from other ARFs on the basis of their post transcriptional regulation by trans-acting small interfering RNAs (tasiRNAs) and microRNAs (miRNAs) <span class="citation">(Rhoades et al. <a href="#ref-rhoades_prediction_2002">2002</a>; Allen et al. <a href="#ref-allen_micrornadirected_2005">2005</a>)</span>. <strong>(AMY check for conservation of Arf 6 and 8 targeted by miRNA 167).</strong> The miRNA target of class C ARFs showed a high degree of conservation (Fig. ARF10_16_17_Alignment). This is consistent with perturbation <strong>what exactly does perturbation mean</strong> studies that have shown drastic developmental phenotype for class C ARFs <span class="citation">(J.-W. Wang et al. <a href="#ref-wang_control_2005">2005</a>; Mallory, Bartel, and Bartel <a href="#ref-mallory_micrornadirected_2005">2005</a>; P.-P. Liu et al. <a href="#ref-liu_repression_2007">2007</a>)</span>. For the class B ARFs (2, 3 and 4) the tasiRNA binding motif is also highly conserved suggesting strong regulatory constraints on these genes (Fig ARF2_3_4_Alignment). The DNA binding domain of ARF4 exhibits non-synonymous variation in 4 accession and would be interesting to carry out phenotypic characterization of these accessions for developmental growth defects (Fig ARF4_DBD Variants map).</p>
<!--chapter:end:06-ARFs.Rmd-->
</div>
</div>
<div id="discussion" class="section level1">
<h1><span class="header-section-number">7</span> Discussion</h1>
<p>Our analysis of the natural diversity in the gene families that make up the auxin nuclear signaling pathway has confirmed existing knowledge, supported existing hypotheses and generated new hypotheses. These results have been generated by a team of undergraduate and graduate students, postdoctoral researchers, and principal investigators as an alpha-test of the ViVa web application developed here. Our analysis has confirmed the importance of auxin-induced degradation of the Aux/IAA proteins as evidenced by the strong conservation of the <em>AFB</em> genes and the canonical degron region of the <em>Aux/IAA</em> genes. Despite this relative conservation, this analysis has also identified many accessions containing new alleles of interest in the <em>AFB</em> and <em>Aux/IAA</em> families. -interesting points from the AFBs -interesting points IAAs -TPLs -ARFs -DBD and PB1</p>
<p>Importance of -AFB -IAA degron -TPL -ARF DBD and PB1</p>
<!--chapter:end:07-discussion.Rmd-->
</div>
<div id="refs" class="section level1">
<h1><span class="header-section-number">8</span> References</h1>
<!--chapter:end:08-references.Rmd-->
<div id="refs" class="references">
<div id="ref-allen_micrornadirected_2005">
<p>Allen, Edwards, Zhixin Xie, Adam M. Gustafson, and James C. Carrington. 2005. “microRNA-Directed Phasing During Trans-Acting siRNA Biogenesis in Plants.” <em>Cell</em> 121 (2): 207–21. doi:<a href="https://doi.org/10.1016/j.cell.2005.04.004">10.1016/j.cell.2005.04.004</a>.</p>
</div>
<div id="ref-arase_iaa8_2012">
<p>Arase, Fumi, Hiroko Nishitani, Mayumi Egusa, Nami Nishimoto, Sumiko Sakurai, Naho Sakamoto, and Hironori Kaminaka. 2012. “IAA8 Involved in Lateral Root Formation Interacts with the Tir1 Auxin Receptor and ARF Transcription Factors in Arabidopsis.” <em>PloS One</em> 7 (8): e43414. doi:<a href="https://doi.org/10.1371/journal.pone.0043414">10.1371/journal.pone.0043414</a>.</p>
</div>
<div id="ref-berleth_auxin_2004">
<p>Berleth, Thomas, Naden T. Krogan, and Enrico Scarpella. 2004. “Auxin Signals–Turning Genes on and Turning Cells Around.” <em>Current Opinion in Plant Biology</em> 7 (5): 553–63. doi:<a href="https://doi.org/10.1016/j.pbi.2004.07.016">10.1016/j.pbi.2004.07.016</a>.</p>
</div>
<div id="ref-boer_structural_2014">
<p>Boer, D. Roeland, Alejandra Freire-Rios, Willy A. M. van den Berg, Terrens Saaki, Iain W. Manfield, Stefan Kepinski, Irene López-Vidrieo, et al. 2014. “Structural Basis for DNA Binding Specificity by the Auxin-Dependent ARF Transcription Factors.” <em>Cell</em> 156 (3): 577–89. doi:<a href="https://doi.org/10.1016/j.cell.2013.12.027">10.1016/j.cell.2013.12.027</a>.</p>
</div>
<div id="ref-calderonvillalobos_combinatorial_2012">
<p>Calderón Villalobos, Luz Irina A., Sarah Lee, Cesar De Oliveira, Anthony Ivetac, Wolfgang Brandt, Lynne Armitage, Laura B. Sheard, et al. 2012. “A Combinatorial Tir1/AFB-Aux/IAA Co-Receptor System for Differential Sensing of Auxin.” <em>Nat Chem Biol</em> 8 (5): 477–85. doi:<a href="https://doi.org/10.1038/nchembio.926">10.1038/nchembio.926</a>.</p>
</div>
<div id="ref-coll_programmed_2011">
<p>Coll, N. S., P. Epple, and J. L. Dangl. 2011. “Programmed Cell Death in the Plant Immune System.” <em>Cell Death and Differentiation</em> 18 (8): 1247–56. doi:<a href="https://doi.org/10.1038/cdd.2011.37">10.1038/cdd.2011.37</a>.</p>
</div>
<div id="ref-covington_circadian_2007">
<p>Covington, Michael F., and Stacey L. Harmer. 2007. “The Circadian Clock Regulates Auxin Signaling and Responses in Arabidopsis.” <em>PLoS Biology</em> 5 (8): e222. doi:<a href="https://doi.org/10.1371/journal.pbio.0050222">10.1371/journal.pbio.0050222</a>.</p>
</div>
<div id="ref-delker_natural_2010">
<p>Delker, Carolin, Yvonne Pöschl, Anja Raschke, Kristian Ullrich, Stefan Ettingshausen, Valeska Hauptmann, Ivo Grosse, and Marcel Quint. 2010. “Natural Variation of Transcriptional Auxin Response Networks in Arabidopsis Thaliana.” <em>The Plant Cell</em> 22 (7): 2184–2200. doi:<a href="https://doi.org/10.1105/tpc.110.073957">10.1105/tpc.110.073957</a>.</p>
</div>
<div id="ref-dezfulian_oligomerization_2016">
<p>Dezfulian, Mohammad H., Espanta Jalili, Don Karl A. Roberto, Britney L. Moss, Kerry Khoo, Jennifer L. Nemhauser, and William L. Crosby. 2016. “Oligomerization of SCF Tir1 Is Essential for Aux/IAA Degradation and Auxin Signaling in Arabidopsis.” <em>PLOS Genet</em> 12 (9): e1006301. doi:<a href="https://doi.org/10.1371/journal.pgen.1006301">10.1371/journal.pgen.1006301</a>.</p>
</div>
<div id="ref-dharmasiri_plant_2005">
<p>Dharmasiri, Nihal, Sunethra Dharmasiri, Dolf Weijers, Esther Lechner, Masashi Yamada, Lawrence Hobbie, Jasmin S. Ehrismann, Gerd Jürgens, and Mark Estelle. 2005. “Plant Development Is Regulated by a Family of Auxin Receptor F Box Proteins.” <em>Developmental Cell</em> 9 (1): 109–19. doi:<a href="https://doi.org/10.1016/j.devcel.2005.05.014">10.1016/j.devcel.2005.05.014</a>.</p>
</div>
<div id="ref-dreher_arabidopsis_2006">
<p>Dreher, Kate A., Jessica Brown, Robert E. Saw, and Judy Callis. 2006. “The Arabidopsis Aux/IAA Protein Family Has Diversified in Degradation and Auxin Responsiveness.” <em>The Plant Cell</em> 18 (3): 699–714. doi:<a href="https://doi.org/10.1105/tpc.105.039172">10.1105/tpc.105.039172</a>.</p>
</div>
<div id="ref-flores-sandoval_simple_2015">
<p>Flores-Sandoval, Eduardo, D. Magnus Eklund, and John L. Bowman. 2015. “A Simple Auxin Transcriptional Response System Regulates Multiple Morphogenetic Processes in the Liverwort Marchantia Polymorpha.” <em>PLOS Genetics</em> 11 (5): e1005207. doi:<a href="https://doi.org/10.1371/journal.pgen.1005207">10.1371/journal.pgen.1005207</a>.</p>
</div>
<div id="ref-guilfoyle_auxin_2007">
<p>Guilfoyle, Tom J., and Gretchen Hagen. 2007. “Auxin Response Factors.” <em>Current Opinion in Plant Biology</em>, Cell signalling and gene regulationedited by jian-kang zhu and ko shimamoto, 10 (5): 453–60. doi:<a href="https://doi.org/10.1016/j.pbi.2007.08.014">10.1016/j.pbi.2007.08.014</a>.</p>
</div>
<div id="ref-guseman_auxininduced_2015">
<p>Guseman, Jessica M., Antje Hellmuth, Amy Lanctot, Tamar P. Feldman, Britney L. Moss, Eric Klavins, Luz Irina A. Calderón Villalobos, and Jennifer L. Nemhauser. 2015. “Auxin-Induced Degradation Dynamics Set the Pace for Lateral Root Development.” <em>Development</em> 142 (5): 905–9. doi:<a href="https://doi.org/10.1242/dev.117234">10.1242/dev.117234</a>.</p>
</div>
<div id="ref-havens_synthetic_2012">
<p>Havens, Kyle A., Jessica M. Guseman, Seunghee S. Jang, Edith Pierre-Jerome, Nick Bolten, Eric Klavins, and Jennifer L. Nemhauser. 2012. “A Synthetic Approach Reveals Extensive Tunability of Auxin Signaling.” <em>Plant Physiol</em> 160 (1): 135–42. doi:<a href="https://doi.org/10.1104/pp.112.202184">10.1104/pp.112.202184</a>.</p>
</div>
<div id="ref-ito_auxindependent_2016">
<p>Ito, Jun, Hidehiro Fukaki, Makoto Onoda, Lin Li, Chuanyou Li, Masao Tasaka, and Masahiko Furutani. 2016. “Auxin-Dependent Compositional Change in Mediator in Arf7- and Arf19-Mediated Transcription.” <em>Proceedings of the National Academy of Sciences</em>, May, 201600739. doi:<a href="https://doi.org/10.1073/pnas.1600739113">10.1073/pnas.1600739113</a>.</p>
</div>
<div id="ref-kagale_ear_2011">
<p>Kagale, Sateesh, and Kevin Rozwadowski. 2011. “EAR Motif-Mediated Transcriptional Repression in Plants: An Underlying Mechanism for Epigenetic Regulation of Gene Expression.” <em>Epigenetics</em> 6 (2): 141–46.</p>
</div>
<div id="ref-ke_structural_2015">
<p>Ke, Jiyuan, Honglei Ma, Xin Gu, Adam Thelen, Joseph S. Brunzelle, Jiayang Li, H. Eric Xu, and Karsten Melcher. 2015. “Structural Basis for Recognition of Diverse Transcriptional Repressors by the TOPLESS Family of Corepressors.” <em>Science Advances</em> 1 (6): e1500107. doi:<a href="https://doi.org/10.1126/sciadv.1500107">10.1126/sciadv.1500107</a>.</p>
</div>
<div id="ref-kim_two_1996">
<p>Kim, Byung Chul, Moon Soo Soh, Bong Joong Kang, Masaki Furuya, and Hong Gil Nam. 1996. “Two Dominant Photomorphogenic Mutations of Arabidopsis Thaliana Identified as Suppressor Mutations of Hy2.” <em>The Plant Journal</em> 9 (4): 441–56. doi:<a href="https://doi.org/10.1046/j.1365-313X.1996.09040441.x">10.1046/j.1365-313X.1996.09040441.x</a>.</p>
</div>
<div id="ref-kliebenstein_role_2008">
<p>Kliebenstein, Daniel J. 2008. “A Role for Gene Duplication and Natural Variation of Gene Expression in the Evolution of Metabolism.” <em>PLOS ONE</em> 3 (3): e1838. doi:<a href="https://doi.org/10.1371/journal.pone.0001838">10.1371/journal.pone.0001838</a>.</p>
</div>
<div id="ref-koenig_auxin_2009">
<p>Koenig, Daniel, Emmanuelle Bayer, Julie Kang, Cris Kuhlemeier, and Neelima Sinha. 2009. “Auxin Patterns Solanum Lycopersicum Leaf Morphogenesis.” <em>Development (Cambridge, England)</em> 136 (17): 2997–3006. doi:<a href="https://doi.org/10.1242/dev.033811">10.1242/dev.033811</a>.</p>
</div>
<div id="ref-korasick_molecular_2014">
<p>Korasick, David A., Corey S. Westfall, Soon Goo Lee, Max H. Nanao, Renaud Dumas, Gretchen Hagen, Thomas J. Guilfoyle, Joseph M. Jez, and Lucia C. Strader. 2014. “Molecular Basis for AUXIN RESPONSE FACTOR Protein Interaction and the Control of Auxin Response Repression.” <em>Proceedings of the National Academy of Sciences</em> 111 (14): 5427–32. doi:<a href="https://doi.org/10.1073/pnas.1400074111">10.1073/pnas.1400074111</a>.</p>
</div>
<div id="ref-kuck_parametric_2010">
<p>Kück, Patrick, Karen Meusemann, Johannes Dambach, Birthe Thormann, Björn M. von Reumont, Johann W. Wägele, and Bernhard Misof. 2010. “Parametric and Non-Parametric Masking of Randomness in Sequence Alignments Can Be Improved and Leads to Better Resolved Trees.” <em>Frontiers in Zoology</em> 7 (March): 10. doi:<a href="https://doi.org/10.1186/1742-9994-7-10">10.1186/1742-9994-7-10</a>.</p>
</div>
<div id="ref-lee_biological_2016">
<p>Lee, Min-Soo, Ji-Hyun An, and Hyung-Taeg Cho. 2016. “Biological and Molecular Functions of Two EAR Motifs of Arabidopsis Iaa7.” <em>Journal of Plant Biology</em> 59 (1): 24–32. doi:<a href="https://doi.org/10.1007/s12374-016-0453-1">10.1007/s12374-016-0453-1</a>.</p>
</div>
<div id="ref-liu_repression_2007">
<p>Liu, Po-Pu, Taiowa A. Montgomery, Noah Fahlgren, Kristin D. Kasschau, Hiroyuki Nonogaki, and James C. Carrington. 2007. “Repression of AUXIN RESPONSE Factor10 by microRNA160 Is Critical for Seed Germination and Post-Germination Stages.” <em>The Plant Journal: For Cell and Molecular Biology</em> 52 (1): 133–46. doi:<a href="https://doi.org/10.1111/j.1365-313X.2007.03218.x">10.1111/j.1365-313X.2007.03218.x</a>.</p>
</div>
<div id="ref-long_topless_2006">
<p>Long, Jeff A., Carolyn Ohno, Zachery R. Smith, and Elliot M. Meyerowitz. 2006. “TOPLESS Regulates Apical Embryonic Fate in Arabidopsis.” <em>Science</em> 312 (5779): 1520–3. doi:<a href="https://doi.org/10.1126/science.1123841">10.1126/science.1123841</a>.</p>
</div>
<div id="ref-ma_d53_2017">
<p>Ma, Honglei, Jingbo Duan, Jiyuan Ke, Yuanzheng He, Xin Gu, Ting-Hai Xu, Hong Yu, et al. 2017. “A D53 Repression Motif Induces Oligomerization of TOPLESS Corepressors and Promotes Assembly of a Corepressor-Nucleosome Complex.” <em>Science Advances</em> 3 (6). doi:<a href="https://doi.org/10.1126/sciadv.1601217">10.1126/sciadv.1601217</a>.</p>
</div>
<div id="ref-mallory_micrornadirected_2005">
<p>Mallory, Allison C., David P. Bartel, and Bonnie Bartel. 2005. “MicroRNA-Directed Regulation of Arabidopsis AUXIN RESPONSE Factor17 Is Essential for Proper Development and Modulates Expression of Early Auxin Response Genes.” <em>The Plant Cell</em> 17 (5): 1360–75. doi:<a href="https://doi.org/10.1105/tpc.105.031716">10.1105/tpc.105.031716</a>.</p>
</div>
<div id="ref-martin-arevalillo_structure_2017">
<p>Martin-Arevalillo, Raquel, Max H. Nanao, Antoine Larrieu, Thomas Vinos-Poyo, David Mast, Carlos Galvan-Ampudia, Géraldine Brunoud, Teva Vernoux, Renaud Dumas, and François Parcy. 2017. “Structure of the Arabidopsis TOPLESS Corepressor Provides Insight into the Evolution of Transcriptional Repression.” <em>Proceedings of the National Academy of Sciences</em>, July, 201703054. doi:<a href="https://doi.org/10.1073/pnas.1703054114">10.1073/pnas.1703054114</a>.</p>
</div>
<div id="ref-melamed_combining_2015">
<p>Melamed, Daniel, David L. Young, Christina R. Miller, and Stanley Fields. 2015. “Combining Natural Sequence Variation with High Throughput Mutational Data to Reveal Protein Interaction Sites.” <em>PLoS Genet</em> 11 (2): e1004918. doi:<a href="https://doi.org/10.1371/journal.pgen.1004918">10.1371/journal.pgen.1004918</a>.</p>
</div>
<div id="ref-moss_rate_2015">
<p>Moss, Britney L., Haibin Mao, Jessica M. Guseman, Thomas R. Hinds, Antje Hellmuth, Marlies Kovenock, Anisa Noorassa, et al. 2015. “Rate Motifs Tune Auxin/Indole-3-Acetic Acid Degradation Dynamics.” <em>Plant Physiology</em> 169 (1): 803–13. doi:<a href="https://doi.org/10.1104/pp.15.00587">10.1104/pp.15.00587</a>.</p>
</div>
<div id="ref-mutte_origin_2018">
<p>Mutte, Sumanth K., Hirotaka Kato, Carl Rothfels, Michael Melkonian, Gane Ka-Shu Wong, and Dolf Weijers. 2018. “Origin and Evolution of the Nuclear Auxin Response System.” <em>eLife</em> 7 (March): e33399. doi:<a href="https://doi.org/10.7554/eLife.33399">10.7554/eLife.33399</a>.</p>
</div>
<div id="ref-nanao_structural_2014">
<p>Nanao, Max H., Thomas Vinos-Poyo, Géraldine Brunoud, Emmanuel Thévenon, Meryl Mazzoleni, David Mast, Stéphanie Lainé, et al. 2014. “Structural Basis for Oligomerization of Auxin Transcriptional Regulators.” <em>Nature Communications</em> 5: 3617. doi:<a href="https://doi.org/10.1038/ncomms4617">10.1038/ncomms4617</a>.</p>
</div>
<div id="ref-overvoorde_functional_2005">
<p>Overvoorde, Paul J., Yoko Okushima, José M. Alonso, April Chan, Charlie Chang, Joseph R. Ecker, Beth Hughes, et al. 2005. “Functional Genomic Analysis of the AUXIN/INDOLE-3-ACETIC ACID Gene Family Members in Arabidopsis Thaliana.” <em>The Plant Cell</em> 17 (12): 3282–3300. doi:<a href="https://doi.org/10.1105/tpc.105.036723">10.1105/tpc.105.036723</a>.</p>
</div>
<div id="ref-park_distributions_2017">
<p>Park, Briton, Matthew T. Rutter, Charles B. Fenster, V. Vaughan Symonds, Mark C. Ungerer, and Jeffrey P. Townsend. 2017. “Distributions of Mutational Effects and the Estimation of Directional Selection in Divergent Lineages of Arabidopsis Thaliana.” <em>Genetics</em> 206 (4): 2105–17. doi:<a href="https://doi.org/10.1534/genetics.116.199190">10.1534/genetics.116.199190</a>.</p>
</div>
<div id="ref-parry_complex_2009">
<p>Parry, G., L. I. Calderon-Villalobos, M. Prigge, B. Peret, S. Dharmasiri, H. Itoh, E. Lechner, W. M. Gray, M. Bennett, and M. Estelle. 2009. “Complex Regulation of the Tir1/AFB Family of Auxin Receptors.” <em>Proceedings of the National Academy of Sciences</em> 106 (52): 22540–5. doi:<a href="https://doi.org/10.1073/pnas.0911967106">10.1073/pnas.0911967106</a>.</p>
</div>
<div id="ref-pierre-jerome_functional_2016">
<p>Pierre-Jerome, Edith, Britney L. Moss, Amy Lanctot, Amber Hageman, and Jennifer L. Nemhauser. 2016. “Functional Analysis of Molecular Interactions in Synthetic Auxin Response Circuits.” <em>Proceedings of the National Academy of Sciences of the United States of America</em> 113 (40): 11354–9. doi:<a href="https://doi.org/10.1073/pnas.1604379113">10.1073/pnas.1604379113</a>.</p>
</div>
<div id="ref-press_overdue_2014">
<p>Press, Maximilian O., Keisha D. Carlson, and Christine Queitsch. 2014. “The Overdue Promise of Short Tandem Repeat Variation for Heritability.” <em>Trends in Genetics: TIG</em> 30 (11): 504–12. doi:<a href="https://doi.org/10.1016/j.tig.2014.07.008">10.1016/j.tig.2014.07.008</a>.</p>
</div>
<div id="ref-prigge_arabidopsis_2016">
<p>Prigge, Michael J., Kathleen Greenham, Yi Zhang, Aaron Santner, Cristina Castillejo, Andrew M. Mutka, Ronan C. O’Malley, Joseph R. Ecker, Barbara N. Kunkel, and Mark Estelle. 2016. “The Arabidopsis Auxin Receptor F-Box Proteins Afb4 and Afb5 Are Required for Response to the Synthetic Auxin Picloram.” <em>G3: Genes|Genomes|Genetics</em> 6 (5): 1383–90. doi:<a href="https://doi.org/10.1534/g3.115.025585">10.1534/g3.115.025585</a>.</p>
</div>
<div id="ref-remington_contrasting_2004">
<p>Remington, David L., Todd J. Vision, Thomas J. Guilfoyle, and Jason W. Reed. 2004. “Contrasting Modes of Diversification in the Aux/IAA and ARF Gene Families.” <em>Plant Physiology</em> 135 (3): 1738–52. doi:<a href="https://doi.org/10.1104/pp.104.039669">10.1104/pp.104.039669</a>.</p>
</div>
<div id="ref-rhoades_prediction_2002">
<p>Rhoades, Matthew W., Brenda J. Reinhart, Lee P. Lim, Christopher B. Burge, Bonnie Bartel, and David P. Bartel. 2002. “Prediction of Plant microRNA Targets.” <em>Cell</em> 110 (4): 513–20.</p>
</div>
<div id="ref-ronquist_mrbayes_2003">
<p>Ronquist, F., and J. P. Huelsenbeck. 2003. “MrBayes 3: Bayesian Phylogenetic Inference Under Mixed Models.” <em>Bioinformatics</em> 19 (12): 1572–4. doi:<a href="https://doi.org/10.1093/bioinformatics/btg180">10.1093/bioinformatics/btg180</a>.</p>
</div>
<div id="ref-starita_variant_2017">
<p>Starita, Lea M., Nadav Ahituv, Maitreya J. Dunham, Jacob O. Kitzman, Frederick P. Roth, Georg Seelig, Jay Shendure, and Douglas M. Fowler. 2017. “Variant Interpretation: Functional Assays to the Rescue.” <em>The American Journal of Human Genetics</em> 101 (3): 315–25. doi:<a href="https://doi.org/10.1016/j.ajhg.2017.07.014">10.1016/j.ajhg.2017.07.014</a>.</p>
</div>
<div id="ref-szemenyei_topless_2008">
<p>Szemenyei, Heidi, Mike Hannon, and Jeff A. Long. 2008. “TOPLESS Mediates Auxin-Dependent Transcriptional Repression During Arabidopsis Embryogenesis.” <em>Science (New York, N.Y.)</em> 319 (5868): 1384–6. doi:<a href="https://doi.org/10.1126/science.1151461">10.1126/science.1151461</a>.</p>
</div>
<div id="ref-tan_mechanism_2007">
<p>Tan, Xu, Luz Irina A. Calderon-Villalobos, Michal Sharon, Changxue Zheng, Carol V. Robinson, Mark Estelle, and Ning Zheng. 2007. “Mechanism of Auxin Perception by the Tir1 Ubiquitin Ligase.” <em>Nature</em> 446 (7136): 640–5. doi:<a href="https://doi.org/10.1038/nature05731">10.1038/nature05731</a>.</p>
</div>
<div id="ref-vidal_nitrate_2014">
<p>Vidal, Elena A, José M Álvarez, and Rodrigo A Gutiérrez. 2014. “Nitrate Regulation of Afb3 and Nac4 Gene Expression in Arabidopsis Roots Depends on Nrt1.1 Nitrate Transport Function.” <em>Plant Signaling &amp; Behavior</em> 9 (6): e28501. doi:<a href="https://doi.org/10.4161/psb.28501">10.4161/psb.28501</a>.</p>
</div>
<div id="ref-wang_control_2005">
<p>Wang, Jia-Wei, Ling-Jian Wang, Ying-Bo Mao, Wen-Juan Cai, Hong-Wei Xue, and Xiao-Ya Chen. 2005. “Control of Root Cap Formation by MicroRNA-Targeted Auxin Response Factors in Arabidopsis.” <em>The Plant Cell</em> 17 (8): 2204–16. doi:<a href="https://doi.org/10.1105/tpc.105.033076">10.1105/tpc.105.033076</a>.</p>
</div>
<div id="ref-winkler_variation_2017">
<p>Winkler, Martin, Michael Niemeyer, Antje Hellmuth, Philipp Janitza, Gideon Christ, Sophia L. Samodelov, Verona Wilde, et al. 2017. “Variation in Auxin Sensing Guides AUX/IAA Transcriptional Repressor Ubiquitylation and Destruction.” <em>Nature Communications</em> 8 (June): 15706. doi:<a href="https://doi.org/10.1038/ncomms15706">10.1038/ncomms15706</a>.</p>
</div>
<div id="ref-wright_decipher_2015">
<p>Wright, Erik S. 2015. “DECIPHER: Harnessing Local Sequence Context to Improve Protein Multiple Sequence Alignment.” <em>BMC Bioinformatics</em> 16 (October): 322. doi:<a href="https://doi.org/10.1186/s12859-015-0749-z">10.1186/s12859-015-0749-z</a>.</p>
</div>
<div id="ref-wright_insights_2017">
<p>Wright, R. Clay, Mollye L. Zahler, Stacey R. Gerben, and Jennifer L. Nemhauser. 2017. “Insights into the Evolution and Function of Auxin Signaling F-Box Proteins in Arabidopsis Thaliana Through Synthetic Analysis of Natural Variants.” <em>Genetics</em> 207 (2): 583–91. doi:<a href="https://doi.org/10.1534/genetics.117.300092">10.1534/genetics.117.300092</a>.</p>
</div>
<div id="ref-wu_auxinregulated_2015">
<p>Wu, Miin-Feng, Nobutoshi Yamaguchi, Jun Xiao, Bastiaan Bargmann, Mark Estelle, Yi Sang, and Doris Wagner. 2015. “Auxin-Regulated Chromatin Switch Directs Acquisition of Flower Primordium Founder Fate.” <em>eLife</em> 4 (October): e09269. doi:<a href="https://doi.org/10.7554/eLife.09269">10.7554/eLife.09269</a>.</p>
</div>
<div id="ref-yang_iaa1_2004">
<p>Yang, Xiaoqing, Sungsu Lee, Jai-Hyun So, Suni Dharmasiri, Nihal Dharmasiri, Lei Ge, Carolyn Jensen, Roger Hangarter, Lawrence Hobbie, and Mark Estelle. 2004. “The Iaa1 Protein Is Encoded by Axr5 and Is a Substrate of SCF(TIR1).” <em>The Plant Journal: For Cell and Molecular Biology</em> 40 (5): 772–82. doi:<a href="https://doi.org/10.1111/j.1365-313X.2004.02254.x">10.1111/j.1365-313X.2004.02254.x</a>.</p>
</div>
<div id="ref-yu_untethering_2015">
<p>Yu, Hong, Yi Zhang, Britney L. Moss, Bastiaan O. R. Bargmann, Renhou Wang, Michael Prigge, Jennifer L. Nemhauser, and Mark Estelle. 2015. “Untethering the Tir1 Auxin Receptor from the SCF Complex Increases Its Stability and Inhibits Auxin Response.” <em>Nature Plants</em> 1 (3): 14030. doi:<a href="https://doi.org/10.1038/nplants.2014.30">10.1038/nplants.2014.30</a>.</p>
</div>
</div>
</div>
<div class="footnotes">
<hr />
<ol>
<li id="fn1"><p>Corresponding author: <a href="mailto:wrightrc@vt.edu">wrightrc@vt.edu</a><a href="#fnref1">↩</a></p></li>
</ol>
</div>
<!--bookdown:body:end-->
            </section>

          </div>
        </div>
      </div>
<!--bookdown:link_prev-->
<!--bookdown:link_next-->
    </div>
  </div>
<!--bookdown:config-->

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "";
    if (src === "" || src === "true") src = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
