---
title: "A comprehensive analysis of natural sequence variation within the <i>Arabidopsis thaliana</i> nuclear auxin signaling pathway using an accessible web application"
author: 
  affiliation: University of Washington
  name: "R. Clay Wright"

date: "`r Sys.Date()`"

output: 
    html_document:
      toc: true
      fig_caption: yes

keywords: r markdown
fontsize: 11pt
fontfamily: Helvetica

bibliography: ["natural_variation.bib", "TIR1_AFB.bib", "ARF_C.bib", "AuxIAA_citations.bib"]

csl: apa.csl
link-citations: true
---

```{r, echo = FALSE, message=FALSE, warning=FALSE}
knitr::opts_chunk$set(warning = FALSE, collapse = TRUE, comment = "#>", tidy = TRUE, out.width = "90%", fig.height=4, fig.width=7, out.extra='style="margin: auto; display: block; padding-top: 15px;"')
library(r1001genomes)
library(DT)
library(dplyr)

```
<!--This is a CSS (Cascading Style Sheet).
 It controls how panels and images are formatted.></!-->
<style>
  img {
    display: block;
    margin: auto;
    width: 90%;
    height: auto;
  }
</style>

# List of working titles  
* A comprehensive analysis of natural sequence variation within the <i>Arabidopsis thaliana</i> nuclear auxin signaling pathway  
* *Add your suggestions* 

# Abstract  

* Motivating, high level statement  
* Statement defining the cutting edge of your research area  
* A statement of the form: "However, it remains unclear ..." that defines what problem you are  
addressing.  
* A statement of the form:"Here, we ..."  that concisely states your contribution.  
* A series of 3?5 statement about your results.  
* A concluding statement.  

# Intro  

The first genome sequence of *Arabidopsis thaliana* facilitated rapid advancement of plant biology through molecular genetics. Since this initial genome, massive scaling of sequencing technology has allowed a pioneering group to survey the global genomic variation in natural *Arabidopsis thaliana* populations. This valuable population genetics resource has led to several associations of genetic loci with phenotypic traits. Beyond these genome-wide studies for gene- and variant- discovery this valuable dataset has seen little use. This natural genomic variation also provides a set of feasible functional variation at the gene, gene family, and gene network level that may provide insight into the function and evolution of genes, families, and networks. We present a web application and R-package through which plant molecular biologists with little to no bioinformatics experience can make use of this rich dataset of genetic variation to formulate hypotheses as to the sequence/function/phenotype relationships determined by the gene, family, or network of their interest. This candidate gene/family tool also allows family-wise alignments of variants to facilitate de novo functional domain identification and evolutionary analysis, easy access to lists of missense polymorphisms to facilitate biochemical assays of variant effects, as well as lists of accessions containing missense (or any type of) polymorphisms to facilitate genetic segregation analysis for deterimining the phenotypic effects of variants and the study of gene network evolution. Here, we have utilized this tool to analyze the genetic variation in the nuclear auxin signaling pathway, and formulate hypotheses regarding the functional implications of this variation and the evolution of the genes and gene families in this pathway. 

# Related work  
What and how have we, as a field, learned from natural variation?  
[@hollister_genomic_2014; @bielawski_detecting_2013; @kryazhimskiy_population_2008; @schmid_multilocus_2005; @nieduszynski_sequence_2011; @melamed_combining_2015; @weigel_population_2015; @schmitz_patterns_2013; @weigel_natural_2005; @hardtke_case_2010; @koornneef_naturally_2004; @mitchell-olds_genetic_2006; @joly-lopez_developing_2016; @wright_insights_2017a; @park_distributions_2017; @_135_2016a]  
 
# Methods

## Nucleotide Diversity Calculation
Nei  defined the nucleotide diversity statistic in his original paper as:
“…the average number of nucleotide differences per site between two randomly chosen DNA sequences.” (Nei, 1979)
And provided the equation:

$\displaystyle \pi = \sum_{ij}x_{i}x_{j}\pi_{ij}$

Where Where $x_{i}$ is the frequency of the ith sequence in the population and $\pi_{ij}$ is the number of sites that are different between the ith and jth sequence divided by sequence length.

a more general form, that treats each sequence in the population as unique can be written as:

$\displaystyle \pi = \frac{1}{L*n^2} \sum_{i=1}^n \sum_{j=1}^n \sum_{k=1}^L{\pi_{ijk}}
\qquad
\displaystyle \pi_{ijk} = \left\{\begin{array}{l}
1 \quad \textrm{if} \quad N_{ik} \neq N_{jk} \\
0 \quad \textrm{if} \quad N_{ik} = N_{jk}
\end{array}\right.$

where $N_{ik}$ is the nucelotide (A, T, C or G) at position k on the ith sequence of the population.
$L$ is the lenght of the sequence (the same for all sequences in the population, 
assumes only SNPs exist, no indels). n is the total number of sequences in the population.

from this form we can re-arange summations to the form below:

$\displaystyle \pi = \frac{1}{L} \sum_{k=1}^L{\pi_{k}}
\qquad
\pi_{k} = \frac{1}{n^2} \sum_{i=1}^n \sum_{j=1}^n {\pi_{ijk}}$

here $\pi_k$ can be thought of as the site-wise nucleotide diversity at postion k, and is equal to 
the nucleotide diversity of a sequence of lenght 1 at location k. we can calculate $\pi_k$ for each
site, then average those over the sequence lenght to calculate $\pi$, the nucleotide diversity 
of the sequence.

The function `Nucleotide_diversity` in the r1001genomes package calculates $\pi_k$ for each position in the gene or region that contains a variant. note, $\pi_k$ is equal to 0 at all locations without variants.
This is also what is displayed in the Diversity Plot tab of the web app.

## Detailed $\pi_k$ calculaiton simplification.

The formula for $\pi_k$ above requires comparing every sequence to everyother sequence at location k,
however, we know there are only a few variant forms at each individual location.

so we can revert back to using Nei's original forula, modifying it slightly,
replacing $x_i$ with $\frac{n_i}{n}$, $n_i$ being the number of sequences in the population of the form $N_i$ at location k: subscripts i and k are summed only over unique forms [need to clarify this]

$\displaystyle \pi_k = \sum_{ij} \frac{n_i}{n}\frac{n_j}{n} \pi_{ij} = 
\frac{1}{n^2}\sum_{ij} n_i n_j \pi_{ij}$

$\pi_{ij(k)} = \left\{\begin{array}{l}
1 \quad \textrm{if} \quad i \neq j \\
0 \quad \textrm{if} \quad i = j
\end{array}\right.$

we will define $n_{!i} = n - n_i$ as the number of sequences different from i at k.

we can see the summed term will be zero if $i=j$, and $n_i n_j$ if $i\neq j$ so:

$\displaystyle \pi_k = \frac{1}{n^2} \sum_{i} n_i n_{!i}$

next we substitute our definition of $n_{!i}$

$\displaystyle \pi_k = \frac{1}{n^2} \sum_{i} n_i (n - n_{i})$

then distribute and split summation:

$\displaystyle \pi_k = \frac{1}{n^2} (n \sum_{i} n_i - \sum_{i} n_{i}^2)$

finally, summing $\displaystyle \sum_{i}n_i$ is equal to n:

$\displaystyle \pi_k = \frac{1}{n^2} (n^2 - \sum_{i} n_{i}^2)$

this simplified form for $\pi_k$ is used by the app, because the counts of unique 
variants at a single nucleotide location can easily be sumaraized in R.

# Results  
## TIR1/AFB genes  

Auxin acts by binding to receptors that in turn target co-repressors for degradation. Auxin receptors (six in the model plant Arabidopsis thaliana) evolved through gene duplication and diversification early in the history of vascular plants [@Parry_Complex_2009]. The rate of co-repressor degradation is determined by the identity of both the receptor and co-repressor [@Havens_Synthetic_2012], and that this rate sets the pace of development [@Guseman_Auxin-induced_2015]. 

Although it is unknown whether different receptors play different roles at the organismal level, a number of studies have pointed out differences in the ways the six different receptors in A. thaliana differ in their expression. The expression of AFB5 is induced in the presence of auxin, making this a circadian-regulated gene [@Covington_Circadian_2007]. Also, AFB3 is more highly expressed in the roots in the presence of nitrate, allowing increased lateral root formation [@Vidal_Nitrate_2010]. Although TIR1/AFBs are expressed ubiquitously in A. thaliana tissues, TIR1, AFB2, and AFB3 have been shown to accumulate in the shoot and root meristems and leaf tissues, with slightly different expression patterns for TIR1 [@Parry_Complex_2009]. AFB1 has a mutation drastically reducing its ability to assemble into an SCF complex, contributing to its increased accumulation, as well as preventing it from degrading Aux/IAAs[@Yu_Untethering_2015]. Higher order TIR1/AFB mutants show AFB1 to have inconclusive organismal function. However, AFB1 is shown to be able to accumulate ubiquitously in all tissues.AFB4 and AFB5 have been shown to preferentially and functionally bind the synthetic auxin picloram, while others have a preference for naturally occurring auxin indole-3-acetic acid [@Prigge_Arabidopsis_2016]. Interestingly, the strength and rate with which TIR1/AFBs are able to bind and mark Aux/IAAs for degradation are variable [@Havens_Synthetic_2012]. TIR1 and AFB2 have slightly different speed at which they perform this function, with AFB2 being generally faster. 

Examining the natural sequence variation across the AFB revealed that *TIR1* and *AFB1* both had very low nonsynonymous diversity, hinting at their likely functional importance. *AFB3* and *AFB4* had higher nonsynonymous diversity, while their sisters genes, *AFB2* and *AFB5* are more conserved. This matches our current understanding of *AFB3* and *AFB4* as being minor players in the auxin signaling pathway, but brings in to question the inconclusive role of *AFB1* in auxin signaling. Although most important functional regions do not contain variants in AFB1, there is a low diversity mutant, only found in the Can-1 accession, which is in the highly conserved TIR1/AFB oligomerization domain. Because variants eliciting significant changes in function often are not selected for, the fact that this variant is extremely low diversity may suggest that it is associated with a strong auxin signaling phenotype. Characterization of this accession may identify potential *AFB1* phenotypes and may also test the hypothesis that AFB1 functions through oligomerization with other TIR1/AFBs.
  The AFB4 and AFB5 receptors have an N-terminal extension prior to the F-box domains. This extesion had very high nonsynonymous diversity(Fig. 4), suggesting that this extension does not play an important functional role in these proteins. 

Figure 4: Alignment of A. thaliana TIR1/AFB F-box domains showing variants. Cyan marks the Fbox binding domain. Yellow, green, and magenta, mark missense and synonymous, synonymous only, and missense only variant locations respectively. Yellow squared E’s show Cul1-interacting residues, differing in AFB1.


## Aux/IAA genes  

###Introduction
The Aux/IAA proteins (IAAs) mediate both auxin perception and transcriptional co-repression of auxin responsive genes. Auxin perception is carried out along with TIR1/AFB proteins, which together with IAAs act as a coreceptor to bind and perceive intracellular auxin. Interaction of IAAs with auxin and TIR1/AFBs is conferred by the highly conserved 13 amino acid degron motif [@tan_mechanism_2007]. Formation of this auxin coreceptor complex triggers ubiquitination and subsequent degradation of IAAs. This degradation in turn relieves their repression upon ARF transcription factors.Stabilizing mutations within the degron can lead to dominant phenotypes [@berleth_auxin_2004; @yang_iaa1_2004]. The Arabidopsis Aux/IAA gene family has 29 members, most of which possess a canonical degron. Notable exceptions are the auxin-insensitive IAA20 and IAA30 that lack a degron sequence entirely, while the closely related IAA31 retains a semi-functional degron sequence with a very slow degradation rate [@dreher_arabidopsis_2006; @havens_synthetic_2012]. Additionally, IAA degradation rate is influenced by rate motifs found outside the degron, such as the KR motif [@moss_rate_2015; @dreher_arabidopsis_2006]. IAA-mediated repression of ARFs depends upon both an EAR domain, N-terminal of the degron, that recruits co-repressors of the TOPLESS (TPL) family, and a Phox and Bem1 (PB1) domain that enables interaction via heteroligomerization with ARF PB1 domains [@nanao_structural_2014; @kagale_ear_2011]. 

Within the available natural sequence variation, we found that for the canonical Aux/IAAs, no missense mutations were present in the KR motif or residues responsible for PB1-PB1 domain interactions (Supp. Fig. Alignment of canonical IAAs). The majority of nonsynonymous mutations observed were in the variable regions between the conserved domains. However, several missense polymorphisms of interest were identified in EAR domains and degrons of several canonical IAAs. 

###Sister Pair Diversification
Previous research has ~~pointed to~~ *found evidence of* genetic redundancy within closely related pairs or groups of IAA proteins [@overvoorde_functional_2005]. For example, the iaa8-1, iaa9-1 double mutant and the iaa5-1, iaa6-1, iaa19-1 triple mutant have wild-type phenotypes [@overvoorde_functional_2005]. Despite this, most *Arabidopsis* IAA sister pairs have one member with high nonsynonymous diversity while the other sister is more conserved (Figure IAA phylogeny heat map). For example, despite having only minor differences in sequence (61.4% sequence identity), the IAA6/19 sister pair has significant differences in expression patterns, protein abundances and functions, suggesting they have undergone functional specialization since their divergence [@winkler_variation_2017]. In an interspecific comparison between *A. thaliana* abd *A. lyrata*, IAA19 is more conserved, while IAA6 showed regions of increased sequence divergence upstream of the degron and in the N-terminal half of the PB1 domain [@winkler_variation_2017]. Consistent with these results, our analysis of intraspecific natural variation revealed higher conservation for IAA19 (${\pi_N/\pi_S}$ = 0.55) compared to IAA6 (${\pi_N/\pi_S}$ = 2.3) (Figure IAA phylogeny heat map), and also detected high diversity within the same regions of IAA6 as seen in Winkler et al. (Supp Fig IAA6 diversity plot). 

Within the Group B IAA genes, there are four sister pairs--IAA10/IAA11, IAA12/IAA13, IAA18/IAA26, and IAA20/IAA30 (Fig X) [@remington_contrasting_2004]. Interestingly, among these sister pairs only IAA12 and 13 have a noticeable difference in nonsynonymous diversity. The Arabidopsis EFP browser shows that IAA10 and 11 have almost identical expression patterns, with the exception of IAA10 being more highly expressed in the developing seed, and IAA11 being more highly expressed in root tissue during lateral root development. In our analysis, IAA10 and IAA11 both showed functional conservation (${\pi_N/\pi_S}$ of 0.80 and 0.67 respectively) in support of IAA10 and IAA11 playing redundant roles. Accessions and associated variants identified here provide a potentially expedited route to elucidating the extent of redundancy between family members such as IAA10 and 11.
	
###Identification of new Loss-of-Function/Gain-of-Function mutations
To test the extent of redundancy/specialization between sister pairs, the Natural Variation Webtool allows for identifications of accessions that can be potentially used to investigate the phenotypic effects in loss of function variants. For example, it has been shown that IAA10 has a missense mutation in the EAR domain, which converts the second leucine in the LxLxL motif to a serine (L49S), a mutation likely to reduce the ability of the IAA to recruit TOPLESS co-repressor [@lee_biological_2016]. This mutation is found in three accessions in the southern UK. A second potentially stronger mutation, represented in one accession from northeastern Spain is a missense resulting in an early stop codon in the PB1 domain (need Supp. Fig?). Such a mutation is likely to prevent IAA10 from interacting with an ARF, rendering it impotent. These alleles would be strong candidates for loss-of-function mutants, allowing researchers to discover the function of IAA10. In particular, it would make possible a straightforward path for creating an iaa10/11 double mutant using CRISPR/Cas9 directed mutagenesis, allowing for examination of the extent of redundancy between these family members. 

Several examples of possible gain-of-function degron missense mutations were found in IAA2, IAA3, IAA6, IAA7, IAA8, IAA13, IAA15, IAA18, IAA19, IAA27, IAA28, IAA29. Most degron mutations were found in one or two accessions represented in the database (Supplemental Table or Map?). Interestingly, almost all of the IAAs found to have accessions with degron missense mutations have a closely related sister, which may indicate a redundant function within the sister-pairs. IAA6 natural variation has two potentially dominant missense mutations found within the degron (Supp Fig. alignment), one previously believed to lead to the shy1-1 phenotype [@kim_two_1996]. However, this variant is widely represented in 31 accessions across europe (Supplemental Map Fig). This finding supports the proposition that iaa6[C78R] may not be the mutation causing the shy1-1 phenotype but rather represents natural variation in the degron sequence ([note from Mike Prigge in TAIR](https://www.arabidopsis.org/servlets/TairObject?type=notepad&id=11247)). These accessions could be used to confirm whether the shy1-1 phenotype exists within these plants. Additionally, an accession was identified with a V77M mutation in the IAA6 degron that may be useful to explore potential iaa6 gain-of-function phenotypes.

###Identifying novel functional sequences in IAAs
An exciting possible use for the Natural Variation Webtool is to search for conserved, low-diversity sites that may represent possible sites of post-translational modification or protein-protein interaction. For example, recent detailed biochemical analysis identified putative lysine ubiquitination sites in members of the IAA6/19 sister pairs [@winkler_variation_2017]; Supp Fig IAA6/19 alignment). These sites show no natural sequence variation in Arabidopsis, while several other lysines are represented in the natural sequence diversity (Supp Figs alignment and map). In IAA6, there are 17 lysines of which 6 are putative ubiquitination sites, one is in the KR motif, while two others show natural variation. In IAA19, there are 15 total lysines, with 8 as putative ubiquitination sites, 2 in the KR motif and degron, and 1 that exhibits natural variation. Natural variation sequence diversity may be used to narrow down a subset of lysines to assess for functional ubiquitination.

The members of the IAA8/9/27 group have a unique N-terminal extension with a highly conserved region common to orthologs in several other species [@dreher_arabidopsis_2006]. IAA8/9 have  redundant roles in leaf serration [@koenig_auxin_2009], while IAA8 important in lateral roots [@arase_iaa8_2012]. IAA8 has been reported to uniquely localize in the cytosol [@arase_iaa8_2012] where it may interact with LSD1 protein [@coll_programmed_2011]. The N-terminal extension has a conserved region shared between IAA8 and IAA9 with low natural sequence diversity (Supp Fig IAA8/9 alignment) that may represent a site of novel functionality, possibly mediating interaction with LSD1. There are a few accessions with missense alleles in this region in both IAA8 and IAA9 that may be valuable in probing the functionality of this region (Supplemental Fig IAA8/9 map).

Members of the IAA17/7/14/16 group have a second EAR motif (LxLxL) with no observed missense mutations in IAA7/16/17 (Supp Fig IAA7/14/16/17 alignment). This observation supports the experimental evidence for IAA7 that the second EAR motif is functional: it can interact with TPR1 corepressor and has a minor repressive role in auxin-related developmental phenotypes [lee_biological_2016]. For IAA14, we did observe two missense mutations that may be of interest to probe functionality of the IAA14 EAR motif: one in the central “L” (L41F) that is represented in only one accession, and the other in the second “x” (K40N) that is found in 12 accessions around Copenhagen (Supp Fig. IAA14 map). 

## TPL/TPR genes  

  The Auxin signaling pathway utilizes the TOPLESS (TPL) family of Gro/TLE/TUP1 type co-repressor proteins to maintain auxin responsive genes in a transcriptionally-repressed state in the absence of auxin (cite). In Arabidopsis the TPL family includes TOPLESS and four TPL-related proteins (TPR1-4), which are comprised of three structural domains: an N-terminal TPL domain and two beta-propellar domains. TPL is recruited to the AUX/IAA proteins through interaction with a conserved protein motif known as an Ethylene-responsive element binding factor-associated amphiphilic repression (EAR) domain (cite). Canonical EAR domains have the amino acid sequence LxLxL, as is found in domain I of the AUX/IAAs (figure xA, cite). TPL co-repressors bind EAR domains through direct interaction with the TPL N-terminal C-terminal to LisH (CTLH) domain (citations of pre-structure founding papers/reviews). Recent structural analyses of the TPL N-terminal domain have highlighted the precise interaction interface between TPL and AUX/IAA EAR domains, as well as the TPL-TPL dimerization and tetramerization motifs (Avarelilo et al 2017, Ke et al, 2015), and even the residues required for higher-order multimers of TPL tetramers (Ma et al, 2017). The two tandem beta propellar domains of TPL/TPRs have as yet not been crystallized, however numerous high resolution crystal structures of beta propellar domains allow for structural prediction of the TPL c-terminal beta propellar domains and the locations of likely interaction faces (Figure, X). 
  
  The TOPLESS co-repressor family generally exhibits a high level of sequence conservation at the amino acid sequence level across the Arabidopsis accessions with all ${\pi_N/\pi_S}$ values below 1 (Table X). The closely related TPL and TPR1 have the highest ${\pi_N/\pi_S}$ values (TPL-0.425, TPR1-0.739), suggesting that these these two related genes tolerate a higher degree of sequence diversity compared to TPR2-4. The TPL/TPR family demonstrates a high level of sequence conservation across the N-terminal TPL domain, with the lowest of observed ${\pi_N/\pi_S}$ values (Figure X). Indeed, all non-synonymous mutations observed in this region are either in linker domains between alpha helices, or are highly conservative mutations within helices (i.e. Valine to Isoleucine), which would be predicted to exhibit little effect on TPL folding and activity. 
  
  The increased N-terminal conservation underscores the importance of the N-terminal TPL domain. For example, the initial tpl-1 mutation (N176H) in the ninth helix is a dominant gain-of-function allele (Long et al, 2002, PMID12050130), which is capable of binding wild-type TPL protein and inducing protein agreggation (Ma, 2017, pmid28630893). It is therefore understandable that very little diversity is observed in the ninth alpha helix of TPL/TPRs as mutations in this domain could act in a dominant negative fashion, rendering any de novo mutation in this region embryo lethal. However, the entire N-terminal domain exhibits very low rates of nucleotide diversity, suggesting that each domain in this region is critical at the population level. Indeed, the crystal structure of TPL has revealed that the TPL domain is responsible for the multiple roles of TPL/TPR proteins: The LisH domain and CRA domain coordinate TPL dimerization and tetramerization respectively (26601214, 28698367), the CTLH domain is critical for interaction with DNA-binding transcription factors (26601214), and helix 9 within the CRA domain may be critical for the formation of higher-order TPL complexes (multimers of TPL tetramers, 28630893). 
  
  We then compared the frequency of SNPs that cause nonsynonymous mutations across the TPL domains. Compared to the TPL domain, both the proline rich and WD-40 domains exhibit a higher rate of missense mutations accross both of the WD-40 repeat regions, here labeled WD40-1 and WD40-2 respectively. We utilized the Phyre2.0 modeling software to generate a rough model of the TPL/TPR WD-40 repeats stricly for the purpose of mapping non-synonymous mutations onto a visible protein chassis (Figure X). Here we can see both faces of each predicted beta-propellar domains, which are connected by a short linker region. Through utilization of protein modelling, it becomes clear that we observe conservation of amino acid sequence at the center of the beta-propellar domains. These observations could be used as the basis for forward mutagensis strategies to test whether this region of TPL functions similarly to its homolog TLE, which binds its target genes EH1 and the C-terminal WRPW/Y motif (Hairy/Hes/Runx) in the center of the beta-propeller domain (pmid 16762837).  
  
  It is interesting to note that despite the presence of 5 TPL/TPR genes in the genome, which are considered to be redundant (pmid 16763149), very few SNPs are observed that are predicted to result in loss of function for any one TPL/TPR gene. In fact, only one deletion was observed (665delG) that creates a frameshift mutation in TPR4 in a single accession (Aiell-4 in Italy). This mutation occurs in codon 222, just downstream of the TPL N-terminal domain, and would generate an early stop codon 11 codons downstream of the deletion site. It is unclear whether this N-terminal fragment would be expressed at any appreciable level, as it is possible that the generation of an early stop codon would trigger nonsense-mediated decay. However, if expressed, it would be fascinating to examine whether this naturally occuring N-terminal truncation retains activity as a co-repressor. Given the observation that the TPL N-terminal domain is sufficient to bind histones (28630893), and repress transcription via synthetic assays in yeast (24979769), it is possible that this represents a semi-functional TPL variant. There has been no evidence for priotization of individual TPL/TPR family members in the auxin response pathway, but experiments modulating the gene copy number of TPR1 revealed the importance of maintaing the proper dosage of TPL/TPR in order to accurately trigger the salicyclic acid innate immune response (20647385). These results, taken together with the absence of TPL/TPR loss-of-function alleles in the Arabidopsis accessions suggest that perhaps the entire family is being conserved at the nucleotide level in the population to meet a gene dosage requirement, with little opportunity for neo- or sub-functionalization.


## ARF genes  

Auxin response is mediated by the auxin responsive transcription factors, or ARFs. There are 23 ARFs in Arabidopsis that are divided into three phylogenetic classes. Class A ARFs, ARF5, ARF6, ARF7, ARF8, and ARF19, activate transcription. These ARFs have a Q-rich region in the middle of the protein that may mediate activation (Guilefoyle, 2007). It has recently been shown that the middle region of ARF5 interacts with the SWI/SNF chromatin remodeling ATPases BRAMA and SPLAYED, possibly to reduce nucleosome occupancy and allow for the recruitment of transcription machinery (Wu, 2016). Additionally ARF7 can interact with Mediator subunits, directly tethering transcriptional proteins to its binding regions (Ito, 2016). Class B and C ARFs are historically categorized as repressor ARFs, though the mechanism through which they confer repression has not been identified. Their middle regions tend to be proline- and serine-rich. 

ARFs are comprised of three major domains, recent structures of which have informed structure-function analysis of the ARFs (Boer, 2014; Nanao, 2014; Korasick, 2014). These domains are conserved throughout land plants. ARFs share an N-terminal B3 DNA binding domain. Flanking this DNA-binding domain is a dimerization domain, which folds up into a single “taco-shaped” domain to allow for dimerization between ARFs. There is an auxillary domain that immediately follows and interacts with the dimerization domain. The middle region is the most variable between ARFs, as mentioned above, but is characterized by repetitive units of glutamine (class A), serine, or proline residues (classes B and C). On the C-terminal end of the protein a protein-protein interaction domain mediates interactions among ARFs, between ARFs and other transcription factors, and between ARFs and the Aux/IAA repressors. This interaction domain was recently characterized as a Phox and Bem1 (PB1) domain, which is comprised of a positive and negative face with conserved basic and acidic residues, respectively. Theoretically the dipolar nature of the PB1 domain can mediate multimerization by the pairwise interaction of these faces on different proteins, and indeed the ARF7 PB1 domain was crystallized as a multimer. It is unclear whether this multimerization occurs or plays a developmental significant role in vivo. Interfering with ARF dimerization in either the DNA-binding proximal dimerization domain or the PB1 domain decreases the ability of class A ARFs to activate transcription in a heterologous yeast system (Pierre-Jerome, 2016). 

While domain architecture is broadly conserved among the ARFs, there are exceptional cases. Three ARFs do not contain a PB1 domain at all, ARF3, ARF13, and ARF17, and several more have lost the conserved acidic or basic residues in the PB1 domain, suggesting they may be reduced to a single interaction domain. Several ARFs additionally have an expanded conserved region within the DNA-binding domain, of unknown function. The majority of domain variation among ARFs occurs in the large B-class subfamily. The liverwort Marcantia polymorpha has a single representative ARF of each class (Flores-Sandoval, 2015). The much expanded classes of the flowering plants arise from both whole genome and tandem duplication events (Remington, 2004), and may have allowed for the expansion of the quantity and complexity of loci regulated by the ARFs and a consequent expansion in their regulation of developmental processes. Here we examine natural variation in the three ARF classes and how this variation may be informative of conservation and function among the ARFs. 

### Class A ARFs
Amy

Class A ARFs are the most well-studied ARF subfamily--the five family members all act as transcriptional activators and have well-characterized, distinct developmental targets. Class A ARFs all contain the canonical B3 DNA-binding domain and PB1 interaction domain, and all have glutamine-rich middle regions (Guilefoyle, 2007). Analysis of class A ARF nucleotide diversity showed that the majority of these ARFs had low nucleotide diveristy scores, all ${\pi_N/\pi_S}$ values substantially lower than 1 with the exception of ARF19, with a relatively high ${\pi_N/\pi_S}$ value of 1.8. Comparing diversity within sister pairs, the two class A ARF sister pairs had divergent diversity scores, with ARF19 and ARF8 having scores at least triple those of ARF7 and ARF6 respectively. This may suggest that ARF6 and ARF7 have more conserved function whereas ARF19 and ARF8 are more diverse. Overall the diversity of class A ARFs was generally low, especially compared to the class B and C ARFs, suggesting that class A ARFs are central to auxin signal transduction and plant development.

For all class A ARFs, the middle region of the protein was the predominant high diversity region. The DNA-binding domain had very few, low-diversity missense mutations, as did the PB1 domain. Considering the necessity of their conserved functions, the low level of variation in these key DNA and protein-protein interaction domains is expected. 

Within the middle region, ARF7 had several expansions of polyglutamine sequences compared to the other class A ARFs. Polyglutamine regions are known to readily expand and contract throughout evolutionary time due to replication error, and variation in polyQ length can be acted on by natural selection and have phenotypic consequences (Press, 2014). 

A few missense variations in ARF5 highlighted potential functional variation among ecotypes. Two high-diversity changes at the positive face of the PB1 domain, very close to the conserved lysine residue, to phenylalanine were interstingly always found together in all ecotypes in which they occured: Tu-B2-3,Ru-2, Haes-1, Rd-0, 627ME-13Y1, Le-0,Bu-0, Bsch-0. These mutations to a rather bulky side chain may change the folding of the PB1 domain and interfere with interactions on the positive face. A high diversity polymorphism that results in change in the 17th amino acid of ARF5 from a leucine to a methionine is also intriguing. ARF5 has an N-terminal extension of unknown function relative to the other class A ARFs *how many amino acids*. It is possible that this methionine acts as an alternative start site in the accessions in which it is found yielding a truncated ARF5 variant without this N-terminal extension. 


### Class B ARFs
Andrew

ARF23, which is truncated in its DNA-binding region, has a high ${\pi_N/\pi_S}$ value of 4.1, and has many high-diversity valued nonsense variants, as may be expected of a gene that is pseudogenizing. 

ARF3, ARF13, and ARF17 all lack a PB1 C-terminal domain, which suggests they may be unable to interact with Aux/IAA repressors and may act as an auxin-insensitive mechanism of adding competitive inhibition to ARF-mediated transcription. These ARFs span the range of ${\pi_N/\pi_S}$ ratios relative to the other ARFs, ranging from 0.85 to 4.1. ARF13, with the highest ${\pi_N/\pi_S}$ of 4.1 has many nonsense variants of high diversity, perhaps suggesting it is pseudogenizing. 

ARF14, ARF15, ARF16, ARF20, and ARF21 all do not have one or multiple of the conserved acidic residues in the PB1 domain, and themselves form a subclade of the B ARFs. Interestingly, no non-synonymous variation is seen in these amino acid residues of these ARFs, suggesting that the amino acids that are in the regions may still be important for PB1 function. 






### Class C ARFs
Hardik (All figure cite below are in folder "/Dropbox/natural-variation-webtool/manuscript_ARF_C/FIGURES")

Class C ARFs shows low nucleotide diveristy scores, all pi_N/pi_S values substantially lower than 1.
Comparing diversity within coding region of class C ARF, ARF16  has least diversity whereas its family members (ARF10, ARF17) shows  scores at least four times higher [@remington_contrasting_2004].
Structurally, all three members of Class C ARFs contain both canonical B3 DNA-binding domain and whereas only ARF10 and ARF16 contains PB1 interaction domain. DNA binding domain exhibit overall low diversity and presence of largely synonymous changes for all the ARFs. Only ARF10, shows mis-sense variants (p.Ser147Thr/c.439T>A  and p.Ala157Ser/c.469G>T) DNA binding domains with higher diversity score for conservative variation (codon 147) compared to non conservation variation (codon 157) (Fig. ARF10_DBD Variants distribution). Of the PB1 domain containing class C ARFs, ARF16 exhibits several naturally occurring non-synonymous variants in contrast to ARF10 (Fig.  Arf16_PB_variants distribution) and is sporadically distributed in location.

Few members of class B ARFs (ARF 2,3,4) and C ARFs (ARF10,16, 17) distinguish from other ARFs on the basis of its ability to be post transcriptionally regulated by trans-acting small interfering RNA (tasiRNA) and microRNA (miRNA) [@rhoades_prediction_2002; @allen_micrornadirected_2005].  (@AMY Arf 6 and 8 targeted by miRNA 167 - check for conservation).  Interestingly, miRNA binding motif of the class C ARFs show high degree of conservation across the natural variation of 1001 genomes queried here (Fig. ARF10_16_17_Alignment). This is consistent with perturbation studies that has shown drastic developmental phenotype for class C ARFs [@wang_control_2005; @mallory_micrornadirected_2005; @liu_repression_2007]. For the class B ARFs (2,3,4) tasiRNA binding motif is highly conserved and show no variation suggesting strong regulatory constrains at the locus (Fig ARF2_3_4_Alignment). DNA binding domain of ARF4 exhibits non-synonymous variation in 4 accession and would be interesting to carry out phenotypic characterize them for developmental growth defects (Fig ARF4_DBD Variants distribution).

