---
title: "A comprehensive analysis of natural sequence variation within the <i>Arabidopsis thaliana</i> nuclear auxin signaling pathway"
author: 
  affiliation: University of Washington
  name: "R. Clay Wright"

date: "`r Sys.Date()`"

output: 
    html_document:
      toc: true
      fig_caption: yes

keywords: r markdown
fontsize: 11pt
fontfamily: Helvetica

bibliography: ["natural_variation.bib", "TIR1_AFB.bib"]
csl: apa.csl
link-citations: true
---

```{r, echo = FALSE, message=FALSE, warning=FALSE}
knitr::opts_chunk$set(warning = FALSE, collapse = TRUE, comment = "#>", tidy = TRUE, out.width = "90%", fig.height=4, fig.width=7, out.extra='style="margin: auto; display: block; padding-top: 15px;"')
library(r1001genomes)
library(DT)
library(dplyr)

```
<!--This is a CSS (Cascading Style Sheet).
 It controls how panels and images are formatted.></!-->
<style>
  img {
    display: block;
    margin: auto;
    width: 90%;
    height: auto;
  }
</style>

# List of working titles  
* A comprehensive analysis of natural sequence variation within the <i>Arabidopsis thaliana</i> nuclear auxin signaling pathway  
* *Add your suggestions* 

# Abstract  

* Motivating, high level statement  
* Statement defining the cutting edge of your research area  
* A statement of the form: "However, it remains unclear ..." that defines what problem you are  
addressing.  
* A statement of the form:"Here, we ..."  that concisely states your contribution.  
* A series of 3?5 statement about your results.  
* A concluding statement.  


    
# Related work  
What and how have we, as a field, learned from natural variation?  
[@hollister_genomic_2014; @bielawski_detecting_2013; @kryazhimskiy_population_2008; @schmid_multilocus_2005; @nieduszynski_sequence_2011; @melamed_combining_2015; @weigel_population_2015; @schmitz_patterns_2013; @weigel_natural_2005; @hardtke_case_2010; @koornneef_naturally_2004; @mitchell-olds_genetic_2006; @joly-lopez_developing_2016; @wright_insights_2017a; @park_distributions_2017; @_135_2016a]  
 
# Methods

# Results  
## TIR1/AFB genes  
Romi
[@gray_identification_1999; @kepinski_auxininduced_2004; @tan_mechanism_2007; @dezfulian_oligomerization_2016; @wu_evolution_2017b]

## Aux/IAA genes  


### Group A Aux/IAAs
Britney

### Group B Aux/IAAs
Mollye/Alex 

### Group C Aux/IAAs
Mega/Alex

## TPL/TPR genes  
Alex

## ARF genes  

### Class A ARFs
Amy

### Class B ARFs
Andrew

### Class C ARFs
Hardik (All figure cite below are in folder "/Dropbox/natural-variation-webtool/manuscript_ARF_C/FIGURES")

Class C ARFs shows low nucleotide diveristy scores, all pi_N/pi_S values substantially lower than 1.
Comparing diversity within coding region of class C ARF, ARF16  has least diversity whereas its family members (ARF10, ARF17) shows  scores at least four times higher (PMCID: PMC519086).
Structurally, all three members of Class C ARFs contain both canonical B3 DNA-binding domain and whereas only ARF10 and ARF16 contains PB1 interaction domain. DNA binding domain exhibit overall low diversity and presence of largely synonymous changes for all the ARFs. Only ARF10, shows mis-sense variants (p.Ser147Thr/c.439T>A  and p.Ala157Ser/c.469G>T) DNA binding domains with higher diversity score for conservative variation (codon 147) compared to non conservation variation (codon 157) (Fig. ARF10_DBD Variants distribution). Of the PB1 domain containing class C ARFs, ARF16 exhibits several naturally occurring non-synonymous variants in contrast to ARF10 (Fig.  Arf16_PB_variants distribution) and is sporadically distributed in location.

Few members of class B ARFs (ARF 2,3,4) and C ARFs (ARF10,16, 17) distinguish from other ARFs on the basis of its ability to be post transcriptionally regulated by trans-acting small interfering RNA (tasiRNA) and microRNA (miRNA) (PMID: 12202040, PMID: 15851028, ).  (@AMY Arf 6 and 8 targeted by miRNA 167 - check for conservation).  Interestingly, miRNA binding motif of the class C ARFs show high degree of conservation across the natural variation of 1001 genomes queried here (Fig. ARF10_16_17_Alignment). This is consistent with perturbation studies that has shown drastic developmental phenotype for class C ARFs (PMID: 16006581, PMID: 17672844, PMID: 15829600). For the class B ARFs (2,3,4) tasiRNA binding motif is highly conserved and show no variation suggesting strong regulatory constrains at the locus (Fig ARF2_3_4_Alignment). DNA binding domain of ARF4 exhibits non-synonymous variation in 4 accession and would be interesting to carry out phenotypic characterize them for developmental growth defects (Fig ARF4_DBD Variants distribution).






