---
title: "Example gene-wise analysis of the 1001 genomes data set"
author: "R. Clay Wright"
date: "`r Sys.Date()`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Vignette Title}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, echo = FALSE, message=FALSE, warning=FALSE}
knitr::opts_chunk$set(collapse = TRUE, comment = "#>", tidy = TRUE)
library(r1001genomes)
```

## Introduction

This web app and open source software package for the R programming language was built to facilitate exploration of the naturally occuring sequence diversity for your favorite gene or gene family. We hope this product will help plant biologists without bioinformatics experience access the 1001 genomes projectâ€™s rich catalog of natural variation, formulate hypotheses and identify new alleles in their favorite gene family.

The 1001 genomes dataset was collected with the intention of genome scale analyses, such as genome wide association studies, in which a particular phenotype is quantified for all of the accessions and sequence variants that are associated with phenotypic variation are identified. [In recent work](https://doi.org/10.1534/genetics.117.3000), we have utilized this rich dataset in a different manner to measure the effects of coding sequence variation on protein functional variation and phenotypic variation . This work has expanded the sequence-function-phenotype map for the auxin-signaling F-box gene family, and this product will facilitate similar analysis other genes, gene families, and genomic regions.

So pick out your favorite gene, gene family or genomic region and follow the steps in this guide to analyze how it varies in plants collected from all over the world. For this example we will be using the auxin-signaling F-box genes, *TIR1* and *AFB2* which have the TAIR locus IDs [AT3G62980](https://www.arabidopsis.org/servlets/TairObject?id=39952&type=locus) and [AT3G26810](https://www.arabidopsis.org/servlets/TairObject?id=38086&type=locus). Click on one of the previous links to search TAIR for the TAIR IDs of your favorite genes.

***
Before digging into the data, make some hypotheses about what genes, domains and/or postions in your 
favorite gene family might be under different evolutionary pressures. We can make some inferences about the evolutionary pressure acting on particular genes based on the sequence diversity observed in the 1001 genomes dataset. However, evolutionary pressures potentially exert influence across all levels of biology, nucleotides, genes, genomes, phenotypes, organisms and populations. This makes any inferences from study of genetic diversity contingent on potentially unknown effects of epistasis between genes, population structure, environment, and many other effectors. So keep in mind throughout this analysis the limitations of examining a single snapshot in time of a potentially biased collection of frequently self-fertizilizing organisms that were dispersed by human migration. e are making in infering theories about how evolutionary forces are acting on your favorite genes. Taking a gene-centric view of evolution, here are some of the evolutionary pressures, or types of natural selection, that we might observe: 



### Purifying selection  
Most nonsynonymous polymorphisms in a gene result in a loss of function. Therefore if a gene is crucial to the fitness of the organism, we would expect a low level of nonsynonymous polymorphism in this gene. If a gene's sequence is highly conserved (low diversity) it may be under [purifying/negative/stabilizing selection](https://en.wikipedia.org/wiki/Negative_selection_(natural_selection)), meaning that any deleterious polymorphisms that arise in a population of organisms are quickly removed due to the decrease fitness effects. This results in the decreased genetic diversity in genes under purifying selection.  

### Neutral divergence
If evolutionary pressure is very weak on a gene its sequence might change with no effect on organismal fitness. This allows a gene to accumulate many mutations that may persist in the population or be lost at random. Neutral divergence, or genetic drift, therefore results in random fluctuations in allele frequencies, which when combined with an appreciable rate of mutation and a sufficient population size yields increased genetic diversity in the population.  


Beyond purifying selection and drift it is very difficult to make any strong inferences on the population as a whole.  

### Directional selection  
If a polymorphism (or allele) is associated with a phenotype that provides an organims with a fitness advantage this phenotype and associated allele may be experiencing directional or positive selection. Positive/directional selection results in an increase in the prevalence of the advantageous allele within the population over time. Directional selection is one of the most difficult things to infer and is a very active area of research. Directional selection, like stabilizing selection is associated with reduced genetic diversity within the population under selection. However, unlike stabilizing selection the gene under directional selection will potentially have two major alleles that provide adaptive potential to different environments or genetic backgrounds.   

Thing we can potentially infer from this data:
1. Polymorphisms in the dataset are not completely detrimental to fitness, and therefore genes and domains with high diversity are less-likely to have important functions.
2. 




Read the literature about your family. 
Are there any postions that are of vital importance? 
What is each domain in the protein doing?
Might there be 
What structure/function knowledge is there? 

Is there a published protein sequence alignment of your gene family?
What postions in the protein family are highly conserved? 
What postions are highly variable?

Is there a published phylogenetic tree of your gene family? 
What are the most recently duplicated sister gene pairs? 
What do you know about the phenotypes of mutants in these pairs?
Does the genetics suggest these pairs are redundant?

For example 
F-box domain will be more conserved than 
***



 

Got your TAIR loci? Now run the shiny web app!
```{r, eval=FALSE}
run1001genomes()
```

## Download the chromosome positions and gene models 
The first step in retrieving the natural variation in our genes is to retrieve the information about where the genes are in the genome of Col-0, the laboratory reference accession, and how the mRNA and coding sequence is laid out. This information is called the gene model. The app will download this info as soon as you click submit. Copy and paste your TAIR loci in the box, separated by commas, and click submit. 

Do you have a spreasheet of loci and want some quick code to make this list?
```{r}
AFB_loci <- read.csv(system.file("extdata", "AFB_gene_ids.csv", package = "r1001genomes"))
paste(AFB_loci$tair_locus, collapse = ",")
```

Occasionally there are multiple models for a single gene if there are multiple possible transcripts or splice variants. Does your gene have multiple models?


The code the app uses to download this table is the `getGeneInfo` function. This function is just a simplifying wrapper for the `BioMart` function `getBM`. This downloads the gene model from ENSEMBL plants.

```{r}
tair_ids <- c("AT3G62980", "AT3G26810")
geneInfo <- getGeneInfo(genes = tair_ids, firstOnly = TRUE, inputType = "tair_locus", useCache = TRUE)

```

Now we can give this table a look. We can also freshen up the table a bit with `datatable`. 
```{r}
datatable(geneInfo[, -c(5,6,9)], colnames = c("tair locus", "symbol", "transcript", 
                                              "Chr", "transcript \nstart", "transcript \nend", 
                                              "transcript \nlength"), rownames = FALSE, 
          options=list(paging=FALSE, searching=FALSE))
```

## Download the variants in the 1001 genomes dataset for these genes
The 1001 genomes dataset is stored as a VCF (Variant Call Format) file. This is 
a very efficient format for storing such a large dataset (the 1135 genomes are ~17gb), 
and for some analyses its matrix format is very useful. We can download a set of VCF files for 
our favorite genes using the `VCFList` function.

```{r, eval=FALSE}
YFG_VCF <- VCFList(geneInfo = geneInfo, by = "transcript", tidy = FALSE)
```

However, for some filtering, summarizing and plotting operations it is more efficient to 
convert the matrix into a very long data frame. Most of the functions in `r1001genomes` 
operate on a long data frame, often called a tidy data frame. We can automatically covert 
our VCF to a tidy data frame by setting the `tidy` argument to `TRUE`.

```{r, eval = FALSE}
YFG_VCF <- VCFList(geneInfo = geneInfo, by = "transcript", tidy = TRUE)
```

The web app automatically creates a tidy dataframe and calculates the number of unique and nonunique polymorphisms in each region of the gene model as well as calculating the diversity in the protein coding sequence. The diversity is the average number of nucleotide differences per site between all possible pairs of sequences. Diversity is also calculated for only the set of synonymous and nonsynonymous polymorphisms.

```{r, cache=TRUE}
TIR1_AFB2_table <- polymorphTable(geneInfo = geneInfo, strains = strains)

```

Before displaing this table we will again clean it up a bit.

```{r, echo = FALSE}
datatable(TIR1_AFB2_table[,1:8],
                  colnames = c("transcript", "5' UTR", "intron", "3' UTR",
                               "coding \n synonymous", "coding \n missense",
                               "upstream", "coding \n total"),
    options=list(paging=FALSE, searching=FALSE))

```




