---
title: "Example gene-wise analysis of the 1001 genomes data set"
author: "R. Clay Wright"
date: "`r Sys.Date()`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Vignette Title}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, echo = FALSE, message=FALSE, warning=FALSE}
knitr::opts_chunk$set(collapse = TRUE, comment = "#>", tidy = TRUE)
library(r1001genomes)
```

## Introduction

This web app and open source software package for the R programming language was built to facilitate exploration of the naturally occuring sequence diversity for your favorite gene or gene family. We hope this product will help plant biologists without bioinformatics experience access the 1001 genomes projectâ€™s rich catalog of natural variation, formulate hypotheses and identify new alleles in their favorite gene family.

The 1001 genomes dataset was collected with the intention of genome scale analyses, such as genome wide association studies, in which a particular phenotype is quantified for all of the accessions and sequence variants that are associated with phenotypic variation are identified. In my recent work, I have utilized this rich dataset in a different manner to measure the effects of coding sequence variation on protein functional variation and phenotypic variation. This work has expanded the sequence-function-phenotype map for the auxin-signaling F-box gene family, and this product will facilitate similar analysis other genes, gene families, and genomic regions.



So pick out your favorite gene, gene family or genomic region and follow the steps in this guide to analyze its natural variation. For this example we will be using the auxin-signaling F-box genes, *TIR1* and *AFB2* which have the TAIR IDs [AT3G62980](https://www.arabidopsis.org/servlets/TairObject?id=39952&type=locus) and [AT3G26810](https://www.arabidopsis.org/servlets/TairObject?id=38086&type=locus). Click on one of the previous links to search TAIR for the TAIR IDs of your favorite genes. 

## Download the chromosome positions and gene models 
The first step in retrieving the natural variation in our genes is to retrieve the information about where the genes are in the genome of Col-0, the laboratory reference accession, and how the mRNA and coding sequence is laid out. This information is called the gene model. Occasionally there are multiple models for a single gene if there are multiple possible transcripts or splice variants.  

To do this we use the `getGeneInfo` function. This function is just a simplifying wrapper for the `BioMart` function `getBM`.

```{r}
tair_ids <- c("AT3G62980", "AT3G26810")
geneInfo <- getGeneInfo(genes = tair_ids, firstOnly = TRUE, inputType = "tair_locus", useCache = FALSE)

```

Now we can give this table a look.
```{r, echo=FALSE, results='markup'}
knitr::kable(geneInfo)
```

## Download the variants in the 1001 genomes dataset for these genes
The 1001 genomes dataset is stored as a VCF (Variant Call Format) file. 

```{r}
TIR1_AFB2_table <- polymorphTable(geneInfo = geneInfo, strains = strains)
knitr::kable(TIR1_AFB2_table)
```


