---
title: "Example gene-wise analysis of the 1001 genomes data set"
author: "R. Clay Wright"
date: "`r Sys.Date()`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Vignette Title}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, echo = FALSE, message=FALSE, warning=FALSE}
knitr::opts_chunk$set(collapse = TRUE, comment = "#>", tidy = TRUE)
library(r1001genomes)
```

<style>
  .using-app-insert {
     background-color: #bfbfbf;
     padding: 15px;
     border-style: solid;
     border-color:  #8c8c8c;
     border-radius: 7px 7px 0 0;
     border-width: 1px;
  }
  
  .using-R-insert {
     background-color: #e6e6e6;
     padding: 15px;
     border-style: solid;
     border-color:  #8c8c8c;
     border-radius: 0 0 7px 7px;
     border-width: 1px;
     border-top: 0;
  }  
  
  
  .using-app-insert h2, .using-app-insert h3, .using-app-insert h4 {
     border-bottom: 0;
  }
  
  .using-R-insert h2, .using-R-insert h3, .using-R-insert h4 {
     border-bottom: 0;
  }
</style>



## Introduction

This web app and open source software package for the R programming language was built to facilitate exploration of the naturally occuring sequence diversity for your favorite gene or gene family. We hope this product will help plant biologists without bioinformatics experience access the 1001 genomes projectâ€™s rich catalog of natural variation, formulate hypotheses and identify new alleles in their favorite gene family.

The 1001 genomes dataset was collected with the intention of genome scale analyses, such as genome wide association studies, in which a particular phenotype is quantified for all of the accessions and sequence variants that are associated with phenotypic variation are identified. [In recent work](https://doi.org/10.1534/genetics.117.3000), we have utilized this rich dataset in a different manner to measure the effects of coding sequence variation on protein functional variation and phenotypic variation . This work has expanded the sequence-function-phenotype map for the auxin-signaling F-box gene family, and this product will facilitate similar analysis other genes, gene families, and genomic regions.

So pick out your favorite gene, gene family or genomic region and follow the steps in this guide to analyze how it varies in plants collected from all over the world. For this example we will be using the auxin-signaling F-box genes, *TIR1* and *AFB2* which have the TAIR locus IDs [AT3G62980](https://www.arabidopsis.org/servlets/TairObject?id=39952&type=locus) and [AT3G26810](https://www.arabidopsis.org/servlets/TairObject?id=38086&type=locus). Click on one of the previous links to search TAIR for the TAIR IDs of your favorite genes.

***
Before digging into the data, make some hypotheses about what genes, domains and/or postions in your 
favorite gene family might be under different evolutionary pressures. We can make some inferences about the evolutionary pressure acting on particular genes based on the sequence diversity observed in the 1001 genomes dataset. However, evolutionary pressures potentially exert influence across all levels of biology, nucleotides, genes, genomes, phenotypes, organisms and populations. This makes any inference from study of genetic diversity contingent on potentially unknown effects of epistasis between genes, population structure, environment, and many other effectors. So keep in mind throughout this analysis the limitations of examining a single snapshot in time of a potentially biased collection of frequently self-fertizilizing organisms that were dispersed by human migration. Because of these limitations, infering theories about how evolutionary forces are acting on your favorite genes is very uncertain business. However, if we assume a few things we can make some speculations.  

Assumptions:  
1. Polymorphisms in the dataset are not *completely* detrimental to organismal fitness.  
2. Most nonsynonymous polymorphisms in a gene result in a loss of function.  

Now, taking a gene-centric view of evolution, here are some of the evolutionary pressures, or types of natural selection, that we might observe: 


### Purifying selection  
If a gene's sequence is highly conserved (low diversity) it may be under [purifying/negative/stabilizing selection](https://en.wikipedia.org/wiki/Negative_selection_(natural_selection)). 
If a genes' function is crucial to the fitness of the organism, and most polymorphisms result in loss of function, we would expect a low level of nonsynonymous polymorphism in this gene. This is because deleterious polymorphisms that do arise in this gene within the population will be outcompeted by the more-fit parent. This results in the decreased genetic diversity in genes under purifying selection.  

### Neutral divergence
Genes of very high diversity may be under neutral divergence, also called genetic drift.
If evolutionary pressure is very weak on a gene its sequence might change with no effect on organismal fitness. This allows a gene to accumulate many mutations that may persist in the population or be lost at random. Neutral divergence, or genetic drift, therefore results in random fluctuations in allele frequencies, which when combined with an appreciable rate of mutation and a sufficient population size yields increased genetic diversity in the population.   
Beyond purifying selection and drift it is very difficult to make any strong inferences on the population as a whole. However there is one more type of selective pressure, positive or directional selection.
  
### Directional selection  
If a polymorphism (or allele) is associated with a phenotype that provides an organims with a fitness advantage this phenotype and associated allele may be experiencing directional or positive selection. Positive/directional selection results in an increase in the prevalence of the advantageous allele within the population over time. Directional selection is one of the most difficult things to infer and is a very active area of research. Directional selection, like stabilizing selection is associated with reduced genetic diversity within the population under selection. However, unlike stabilizing selection the gene under directional selection will potentially have two major alleles that provide adaptive potential to different environments or genetic backgrounds.   


## Hypothesize
Now go read/collect the literature about your family and make some hypotheses!  
Here are some questions to help you get start:  

What structure/function knowledge is there?  
What is each domain in the protein doing?  
Are there any postions that are of vital importance?  
  
Is there a published protein sequence alignment of your gene family?  
What postions in the protein family are highly conserved?  
What postions are highly variable?  

Is there a published phylogenetic tree of your gene family?  
What are the most recently duplicated sister gene pairs?  
What do you know about the phenotypes of mutants in these pairs?  
Do the genetics suggest these pairs are redundant?  
  
  
## For example
In TIR1 and AFB2 there are 2 domains. The F-box domain interacts with the SCF components[@gray_identification_1999], and the LRR domain binds auxin and Aux/IAA proteins [@kepinski_auxininduced_2004]. The F-box is roughly 40 amino acids and wraps tightly around the SKP1 component of the SCF complex. The LRR is greater than 500 amino acids and binds auxin in a pocket formed by a horseshoe shaped chain of LRRs [@tan_mechanism_2007]. The outside of this horseshoe shape does not have any known interactions other than a dimerization interface [@dezfulian_oligomerization_2016]. Because the F-box domain must interact with the SCF complex for TIR1/AFB2 to perform their known function, we might expect the F-box to be highly conserved. The small size of the F-box also helps strengthen this hypothesis. The LRR domain is much larger and may have differing specificities for Aux/IAAs, so we might expect it to be more diverse. 
We would also expect the residues of the LRR that contact auxin and the Aux/IAA degron to be highly conserved. 

***

<!--=========================================================================-->
<div  class="using-app-insert"><!--panel start--------------------------------->
## Using the App 
### A note on the format of this document

In addition to calling the functions of the r1001genomes package in R directly. there is also a GUI provided in the form of a Shiny app. This app can perform some of the analysis tasks the package is capable of performing, without the need to have knowledge of programming in R. 

In this vignette panels like this one are used to show examples of using the app to achieve similar results as using  the R function calls directly in this vignette. Screenshots will usually be provided to make it clear what section or tab we are talking about

```{r, echo = FALSE, out.width = "90%", out.extra='style="margin: auto; display: block; padding-top: 15px;"' }
knitr::include_graphics("Screenshots/App_Initial.PNG")
knitr::asis_output("
                   ") # This newline required for headings to render after image
```
## Starting the web app

to start the Shiny app, run the following command:
```{r, eval=FALSE}
run1001genomes()
```
The app should appear in a new window, and look similar to the image above.
</div><!--panel end------------------------------------------------------------>
<div  class="using-R-insert"><!--panel start----------------------------------->
## Using R
in these lighter colored panels, the equivalent R code will be displayed.
</div><!--panel end------------------------------------------------------------>
<!--=========================================================================--> 


## Download the chromosome positions and gene models 
The first step in retrieving the natural variation in our genes is to retrieve the information about where the genes are in the genome of Col-0, the laboratory reference accession, and how the mRNA and coding sequence is laid out. This information is called the gene model.

<!--=========================================================================-->
<div  class="using-app-insert"><!--panel start--------------------------------->
## Loading Gene Data in the App
```{r, echo = FALSE, out.width = "90%", out.extra='style="margin: auto; display: block; padding-top: 15px;"' }
knitr::include_graphics("Screenshots/Tab_1_Select_Genes.PNG")
knitr::asis_output("
                   ") # This newline required for headings to render after image
```
### Selecting Genes and Generating the Gene Model Table
At the top of the first tab (the "SNP Stats" tab) is a text input field where you can input a list of TAIR loci.

The app will download the gene model information as soon as you click submit. Copy and paste your TAIR loci in the box, separated by commas, and click submit.

The app may take a little while to process after clicking the submit button on the first tab, especially if you have a long list of genes, but once this process is complete on the first tab, the data for these genes will be already loaded for the subsequent tabs. 

Once the loading process has finished, the first table you should see scrolling down the SNP Stats tab is the Gene Information table. it should look similar to the table produced in the "Using R" section below.
</div><!--panel end------------------------------------------------------------>
<div  class="using-R-insert"><!--panel start----------------------------------->
## Using R
The code the app uses to download this table is the `getGeneInfo` function. This function is just a simplifying wrapper for the `BioMart` function `getBM`. This downloads the gene model from ENSEMBL plants.

```{r}
tair_ids <- c("AT3G62980", "AT3G26810")
geneInfo <- getGeneInfo(genes = tair_ids, firstOnly = TRUE, inputType = "tair_locus", useCache = TRUE)
```

Now we can give this table a look. We can also freshen up the table a bit with `datatable`. 
```{r}
datatable(geneInfo[, -c(5,6,9)], colnames = c("tair locus", "symbol", "transcript", 
                                              "Chr", "transcript \nstart", "transcript \nend", 
                                              "transcript \nlength"), rownames = FALSE, 
          options=list(paging=FALSE, searching=FALSE))
```
</div><!--panel end------------------------------------------------------------>
<!--=========================================================================-->

Do you have a spreasheet of loci and want some quick code to make a text string you can copy and paste into the app gene input box?
```{r}
AFB_loci <- read.csv(system.file("extdata", "AFB_gene_ids.csv", package = "r1001genomes"))
paste(AFB_loci$tair_locus, collapse = ",")
```

Occasionally there are multiple models for a single gene if there are multiple possible transcripts or splice variants. Does your gene have multiple models?


## Download the variants in the 1001 genomes dataset for these genes
The 1001 genomes dataset is stored as a VCF (Variant Call Format) file. This is 
a very efficient format for storing such a large dataset (the 1135 genomes are ~17gb), 
and for some analyses its matrix format is very useful. 

<!--=========================================================================-->
<div  class="using-app-insert"><!--panel start--------------------------------->
## Downloading and processing of VCFs in the App
Upon clicking the "Submit" button on the "SNP Stats" tab, the web app automatically creates a tidy dataframe and calculates the number of unique and nonunique polymorphisms in each region of the gene model as well as calculating the diversity in the protein coding sequence. The diversity is the average number of nucleotide differences per site between all possible pairs of sequences. Diversity is also calculated for only the set of synonymous and nonsynonymous polymorphisms.

The results of these calculations are displayed in the three tables below the Gene Information table.


</div><!--panel end------------------------------------------------------------>
<div  class="using-R-insert"><!--panel start----------------------------------->
## Using R
We can download a set of VCF files for our favorite genes using the `VCFList` function.

```{r, eval=FALSE}
YFG_VCF <- VCFList(geneInfo = geneInfo, by = "transcript", tidy = FALSE)
```

However, for some filtering, summarizing and plotting operations it is more efficient to 
convert the matrix into a very long data frame. Most of the functions in `r1001genomes` 
operate on a long data frame, often called a tidy data frame. We can automatically covert 
our VCF to a tidy data frame by setting the `tidy` argument to `TRUE`.

```{r, eval = FALSE}
YFG_VCF <- VCFList(geneInfo = geneInfo, by = "transcript", tidy = TRUE)
```

To generate the three tables of polymorphism stats the app calls the function `polymorphTable`. in R we can do the same thing.
```{r, cache=TRUE}
TIR1_AFB2_table <- polymorphTable(geneInfo = geneInfo, strains = strains)

```

Before displaing this table we will again clean it up a bit.

```{r, echo = FALSE}
formatRound(datatable(TIR1_AFB2_table[,c(1:5, 7)],
                  colnames = c("transcript", "5' UTR", "intron", "3' UTR",
                               "coding \n synonymous", "coding \n missense",
                               "coding \n total"), escape = FALSE,
    options=list(paging=FALSE, searching=FALSE)), columns = 1:6, digits = 0)
```
The nucleotide diversity for the chosen genes is also calculated in the call to `polymorphTable`.

```{r}
formatRound(datatable(TIR1_AFB2_table[, c(8:12)],
                  #
                  colnames = c("transcript",
                               "&pi;<sub>N</sub>",
                               "&pi;<sub>S</sub>",
                               "&pi;<sub>N</sub>/&pi;<sub>S</sub>",
                               "&pi; coding",
                               "&pi; transcript"), escape = FALSE,
                  options = list(paging=FALSE, searching=FALSE)),
                columns = 1:5, digits = 6)
```




</div><!--panel end------------------------------------------------------------>
<!--=========================================================================-->

### Results
Do these values match your hypotheses? Is there anything unexpected?  

For TIR1 

## Diversity within each gene  

Now we'll move on to the second "Diversity Plot" tab. This tab will calculate the diversity at each position within the coding sequence of each gene. 


<!--=========================================================================-->







